# Category: Settings

## Settings

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

PyMOL is incredibly useful because of its flexibility. Almost any setting can be tweaked to give you the ability to make your scenes appear just as you want. Each command usually comes with one or more settings. For example, the [ray](/index.php/Ray "Ray") command comes with **50** settings! 

  


## Contents

  * 1 List of settings
    * 1.1 A
    * 1.2 B
    * 1.3 C
    * 1.4 D
    * 1.5 E
    * 1.6 F
    * 1.7 G
    * 1.8 H
    * 1.9 I
    * 1.10 J
    * 1.11 K
    * 1.12 L
    * 1.13 M
    * 1.14 N
    * 1.15 O
    * 1.16 P
    * 1.17 Q
    * 1.18 R
    * 1.19 S
    * 1.20 T
    * 1.21 U
    * 1.22 V
    * 1.23 W
    * 1.24 X
    * 1.25 Y
    * 1.26 Z



## List of settings

Below you will find the list of all known settings (as of this writing). Settings are added all the time, so this list should be updated somehow. Also, see all **[documented settings](/index.php/Category:Settings "Category:Settings")**. 

If you feel like documenting them, and what they do, feel free. 

### A

[active_selections](/index.php?title=Active_selections&action=edit&redlink=1 "Active selections \(page does not exist\)") [all_states](/index.php/All_states "All states") [ambient](/index.php/Ambient "Ambient") [angle_color](/index.php?title=Angle_color&action=edit&redlink=1 "Angle color \(page does not exist\)") [angle_label_position](/index.php?title=Angle_label_position&action=edit&redlink=1 "Angle label position \(page does not exist\)") [angle_size](/index.php?title=Angle_size&action=edit&redlink=1 "Angle size \(page does not exist\)") [animation](/index.php?title=Animation&action=edit&redlink=1 "Animation \(page does not exist\)") [animation_duration](/index.php?title=Animation_duration&action=edit&redlink=1 "Animation duration \(page does not exist\)") [antialias](/index.php/Antialias "Antialias") [async_builds](/index.php/Async_builds "Async builds") [ati_bugs](/index.php?title=Ati_bugs&action=edit&redlink=1 "Ati bugs \(page does not exist\)") [atom_name_wildcard](/index.php/Atom_name_wildcard "Atom name wildcard") [auto_classify_atoms](/index.php?title=Auto_classify_atoms&action=edit&redlink=1 "Auto classify atoms \(page does not exist\)") [auto_color](/index.php?title=Auto_color&action=edit&redlink=1 "Auto color \(page does not exist\)") [auto_copy_images](/index.php?title=Auto_copy_images&action=edit&redlink=1 "Auto copy images \(page does not exist\)") [auto_defer_builds](/index.php?title=Auto_defer_builds&action=edit&redlink=1 "Auto defer builds \(page does not exist\)") [auto_dss](/index.php?title=Auto_dss&action=edit&redlink=1 "Auto dss \(page does not exist\)") [auto_hide_selections](/index.php?title=Auto_hide_selections&action=edit&redlink=1 "Auto hide selections \(page does not exist\)") [auto_indicate_flags](/index.php?title=Auto_indicate_flags&action=edit&redlink=1 "Auto indicate flags \(page does not exist\)") [auto_number_selections](/index.php?title=Auto_number_selections&action=edit&redlink=1 "Auto number selections \(page does not exist\)") [auto_remove_hydrogens](/index.php?title=Auto_remove_hydrogens&action=edit&redlink=1 "Auto remove hydrogens \(page does not exist\)") [auto_rename_duplicate_objects](/index.php?title=Auto_rename_duplicate_objects&action=edit&redlink=1 "Auto rename duplicate objects \(page does not exist\)") [auto_sculpt](/index.php?title=Auto_sculpt&action=edit&redlink=1 "Auto sculpt \(page does not exist\)") [auto_show_lines](/index.php?title=Auto_show_lines&action=edit&redlink=1 "Auto show lines \(page does not exist\)") [auto_show_nonbonded](/index.php?title=Auto_show_nonbonded&action=edit&redlink=1 "Auto show nonbonded \(page does not exist\)") [auto_show_selections](/index.php?title=Auto_show_selections&action=edit&redlink=1 "Auto show selections \(page does not exist\)") [auto_show_spheres](/index.php?title=Auto_show_spheres&action=edit&redlink=1 "Auto show spheres \(page does not exist\)") [auto_zoom](/index.php/Auto_zoom "Auto zoom")

### B

[backface_cull](/index.php/Backface_cull "Backface cull") [batch_prefix](/index.php?title=Batch_prefix&action=edit&redlink=1 "Batch prefix \(page does not exist\)") [bg_rgb](/index.php/Bg_rgb "Bg rgb") [bonding_vdw_cutoff](/index.php?title=Bonding_vdw_cutoff&action=edit&redlink=1 "Bonding vdw cutoff \(page does not exist\)") [button_mode](/index.php?title=Button_mode&action=edit&redlink=1 "Button mode \(page does not exist\)") [button_mode_name](/index.php/Button_mode_name "Button mode name")

### C

[cache_display](/index.php?title=Cache_display&action=edit&redlink=1 "Cache display \(page does not exist\)") [cache_frames](/index.php?title=Cache_frames&action=edit&redlink=1 "Cache frames \(page does not exist\)") [cache_max](/index.php?title=Cache_max&action=edit&redlink=1 "Cache max \(page does not exist\)") [cache_memory](/index.php?title=Cache_memory&action=edit&redlink=1 "Cache memory \(page does not exist\)") [cache_mode](/index.php?title=Cache_mode&action=edit&redlink=1 "Cache mode \(page does not exist\)") [cartoon_color](/index.php/Cartoon_color "Cartoon color") [cartoon_cylindrical_helices](/index.php/Cartoon_cylindrical_helices "Cartoon cylindrical helices") [cartoon_debug](/index.php?title=Cartoon_debug&action=edit&redlink=1 "Cartoon debug \(page does not exist\)") [cartoon_discrete_colors](/index.php/Cartoon_discrete_colors "Cartoon discrete colors") [cartoon_dumbbell_length](/index.php/Cartoon_dumbbell_length "Cartoon dumbbell length") [cartoon_dumbbell_radius](/index.php/Cartoon_dumbbell_radius "Cartoon dumbbell radius") [cartoon_dumbbell_width](/index.php/Cartoon_dumbbell_width "Cartoon dumbbell width") [cartoon_fancy_helices](/index.php/Cartoon_fancy_helices "Cartoon fancy helices") [cartoon_fancy_sheets](/index.php/Cartoon_fancy_sheets "Cartoon fancy sheets") [cartoon_flat_cycles](/index.php/Cartoon_flat_cycles "Cartoon flat cycles") [cartoon_flat_sheets](/index.php/Cartoon_flat_sheets "Cartoon flat sheets") [cartoon_helix_radius](/index.php/Cartoon_helix_radius "Cartoon helix radius") [cartoon_highlight_color](/index.php/Cartoon_highlight_color "Cartoon highlight color") [cartoon_ladder_color](/index.php?title=Cartoon_ladder_color&action=edit&redlink=1 "Cartoon ladder color \(page does not exist\)") [cartoon_ladder_mode](/index.php?title=Cartoon_ladder_mode&action=edit&redlink=1 "Cartoon ladder mode \(page does not exist\)") [cartoon_ladder_radius](/index.php?title=Cartoon_ladder_radius&action=edit&redlink=1 "Cartoon ladder radius \(page does not exist\)") [cartoon_loop_cap](/index.php?title=Cartoon_loop_cap&action=edit&redlink=1 "Cartoon loop cap \(page does not exist\)") [cartoon_loop_quality](/index.php?title=Cartoon_loop_quality&action=edit&redlink=1 "Cartoon loop quality \(page does not exist\)") [cartoon_loop_radius](/index.php?title=Cartoon_loop_radius&action=edit&redlink=1 "Cartoon loop radius \(page does not exist\)") [cartoon_nucleic_acid_color](/index.php/Cartoon_nucleic_acid_color "Cartoon nucleic acid color") [cartoon_nucleic_acid_mode](/index.php/Cartoon_nucleic_acid_mode "Cartoon nucleic acid mode") [cartoon_oval_length](/index.php/Cartoon_oval_length "Cartoon oval length") [cartoon_oval_quality](/index.php/Cartoon_oval_quality "Cartoon oval quality") [cartoon_oval_width](/index.php/Cartoon_oval_width "Cartoon oval width") [cartoon_power](/index.php?title=Cartoon_power&action=edit&redlink=1 "Cartoon power \(page does not exist\)") [cartoon_power_b](/index.php?title=Cartoon_power_b&action=edit&redlink=1 "Cartoon power b \(page does not exist\)") [cartoon_putty_quality](/index.php?title=Cartoon_putty_quality&action=edit&redlink=1 "Cartoon putty quality \(page does not exist\)") [cartoon_putty_radius](/index.php/Cartoon_putty_radius "Cartoon putty radius") [cartoon_putty_range](/index.php?title=Cartoon_putty_range&action=edit&redlink=1 "Cartoon putty range \(page does not exist\)") [cartoon_putty_scale_max](/index.php/Cartoon_putty_scale_max "Cartoon putty scale max") [cartoon_putty_scale_min](/index.php/Cartoon_putty_scale_min "Cartoon putty scale min") [cartoon_putty_scale_power](/index.php?title=Cartoon_putty_scale_power&action=edit&redlink=1 "Cartoon putty scale power \(page does not exist\)") [cartoon_putty_transform](/index.php/Cartoon_putty_transform "Cartoon putty transform") [cartoon_rect_length](/index.php/Cartoon_rect_length "Cartoon rect length") [cartoon_rect_width](/index.php/Cartoon_rect_width "Cartoon rect width") [cartoon_refine](/index.php?title=Cartoon_refine&action=edit&redlink=1 "Cartoon refine \(page does not exist\)") [cartoon_refine_normals](/index.php?title=Cartoon_refine_normals&action=edit&redlink=1 "Cartoon refine normals \(page does not exist\)") [cartoon_refine_tips](/index.php?title=Cartoon_refine_tips&action=edit&redlink=1 "Cartoon refine tips \(page does not exist\)") [cartoon_ring_color](/index.php?title=Cartoon_ring_color&action=edit&redlink=1 "Cartoon ring color \(page does not exist\)") [cartoon_ring_finder](/index.php?title=Cartoon_ring_finder&action=edit&redlink=1 "Cartoon ring finder \(page does not exist\)") [cartoon_ring_mode](/index.php/Cartoon_ring_mode "Cartoon ring mode") [cartoon_ring_radius](/index.php?title=Cartoon_ring_radius&action=edit&redlink=1 "Cartoon ring radius \(page does not exist\)") [cartoon_ring_transparency](/index.php/Cartoon_ring_transparency "Cartoon ring transparency") [cartoon_ring_width](/index.php?title=Cartoon_ring_width&action=edit&redlink=1 "Cartoon ring width \(page does not exist\)") [cartoon_round_helices](/index.php/Cartoon_round_helices "Cartoon round helices") [cartoon_sampling](/index.php/Cartoon_sampling "Cartoon sampling") [cartoon_side_chain_helper](/index.php/Cartoon_side_chain_helper "Cartoon side chain helper") [cartoon_smooth_cycles](/index.php?title=Cartoon_smooth_cycles&action=edit&redlink=1 "Cartoon smooth cycles \(page does not exist\)") [cartoon_smooth_first](/index.php?title=Cartoon_smooth_first&action=edit&redlink=1 "Cartoon smooth first \(page does not exist\)") [cartoon_smooth_last](/index.php?title=Cartoon_smooth_last&action=edit&redlink=1 "Cartoon smooth last \(page does not exist\)") [cartoon_smooth_loops](/index.php/Cartoon_smooth_loops "Cartoon smooth loops") [cartoon_throw](/index.php?title=Cartoon_throw&action=edit&redlink=1 "Cartoon throw \(page does not exist\)") [cartoon_trace_atoms](/index.php?title=Cartoon_trace_atoms&action=edit&redlink=1 "Cartoon trace atoms \(page does not exist\)") [cartoon_transparency](/index.php/Cartoon_transparency "Cartoon transparency") [cartoon_tube_cap](/index.php?title=Cartoon_tube_cap&action=edit&redlink=1 "Cartoon tube cap \(page does not exist\)") [cartoon_tube_quality](/index.php?title=Cartoon_tube_quality&action=edit&redlink=1 "Cartoon tube quality \(page does not exist\)") [cartoon_tube_radius](/index.php?title=Cartoon_tube_radius&action=edit&redlink=1 "Cartoon tube radius \(page does not exist\)") [cavity_cull](/index.php/Cavity_cull "Cavity cull") [cgo_dot_radius](/index.php?title=Cgo_dot_radius&action=edit&redlink=1 "Cgo dot radius \(page does not exist\)") [cgo_dot_width](/index.php?title=Cgo_dot_width&action=edit&redlink=1 "Cgo dot width \(page does not exist\)") [cgo_ellipsoid_quality](/index.php?title=Cgo_ellipsoid_quality&action=edit&redlink=1 "Cgo ellipsoid quality \(page does not exist\)") [cgo_line_radius](/index.php?title=Cgo_line_radius&action=edit&redlink=1 "Cgo line radius \(page does not exist\)") [cgo_line_width](/index.php?title=Cgo_line_width&action=edit&redlink=1 "Cgo line width \(page does not exist\)") [cgo_ray_width_scale](/index.php?title=Cgo_ray_width_scale&action=edit&redlink=1 "Cgo ray width scale \(page does not exist\)") [cgo_sphere_quality](/index.php?title=Cgo_sphere_quality&action=edit&redlink=1 "Cgo sphere quality \(page does not exist\)") [cgo_transparency](/index.php/Cgo_transparency "Cgo transparency") [clamp_colors](/index.php?title=Clamp_colors&action=edit&redlink=1 "Clamp colors \(page does not exist\)") [cone_quality](/index.php?title=Cone_quality&action=edit&redlink=1 "Cone quality \(page does not exist\)") [connect_bonded](/index.php?title=Connect_bonded&action=edit&redlink=1 "Connect bonded \(page does not exist\)") [connect_cutoff](/index.php/Connect_cutoff "Connect cutoff") [connect_mode](/index.php/Connect_mode "Connect mode") [coulomb_cutoff](/index.php?title=Coulomb_cutoff&action=edit&redlink=1 "Coulomb cutoff \(page does not exist\)") [coulomb_dielectric](/index.php?title=Coulomb_dielectric&action=edit&redlink=1 "Coulomb dielectric \(page does not exist\)") [coulomb_units_factor](/index.php?title=Coulomb_units_factor&action=edit&redlink=1 "Coulomb units factor \(page does not exist\)") [cull_spheres](/index.php?title=Cull_spheres&action=edit&redlink=1 "Cull spheres \(page does not exist\)") [cylinder_shader_ff_workaround](/index.php/Cylinder_shader_ff_workaround "Cylinder shader ff workaround")

### D

[dash_color](/index.php/Dash_color "Dash color") [dash_gap](/index.php/Dash_gap "Dash gap") [dash_length](/index.php/Dash_length "Dash length") [dash_radius](/index.php/Dash_radius "Dash radius") [dash_round_ends](/index.php/Dash_round_ends "Dash round ends") [dash_width](/index.php/Dash_width "Dash width") [debug_pick](/index.php?title=Debug_pick&action=edit&redlink=1 "Debug pick \(page does not exist\)") [defer_builds_mode](/index.php/Defer_builds_mode "Defer builds mode") [defer_updates](/index.php?title=Defer_updates&action=edit&redlink=1 "Defer updates \(page does not exist\)") [depth_cue](/index.php/Depth_cue "Depth cue") [dihedral_color](/index.php?title=Dihedral_color&action=edit&redlink=1 "Dihedral color \(page does not exist\)") [dihedral_label_position](/index.php?title=Dihedral_label_position&action=edit&redlink=1 "Dihedral label position \(page does not exist\)") [dihedral_size](/index.php?title=Dihedral_size&action=edit&redlink=1 "Dihedral size \(page does not exist\)") [direct](/index.php?title=Direct&action=edit&redlink=1 "Direct \(page does not exist\)") [dist_counter](/index.php?title=Dist_counter&action=edit&redlink=1 "Dist counter \(page does not exist\)") [distance_exclusion](/index.php?title=Distance_exclusion&action=edit&redlink=1 "Distance exclusion \(page does not exist\)") [dot_color](/index.php?title=Dot_color&action=edit&redlink=1 "Dot color \(page does not exist\)") [dot_density](/index.php?title=Dot_density&action=edit&redlink=1 "Dot density \(page does not exist\)") [dot_hydrogens](/index.php?title=Dot_hydrogens&action=edit&redlink=1 "Dot hydrogens \(page does not exist\)") [dot_lighting](/index.php?title=Dot_lighting&action=edit&redlink=1 "Dot lighting \(page does not exist\)") [dot_mode](/index.php?title=Dot_mode&action=edit&redlink=1 "Dot mode \(page does not exist\)") [dot_normals](/index.php?title=Dot_normals&action=edit&redlink=1 "Dot normals \(page does not exist\)") [dot_radius](/index.php/Dot_radius "Dot radius") [dot_solvent](/index.php?title=Dot_solvent&action=edit&redlink=1 "Dot solvent \(page does not exist\)") [dot_width](/index.php/Dot_width "Dot width") [draw_frames](/index.php?title=Draw_frames&action=edit&redlink=1 "Draw frames \(page does not exist\)")

### E

[editor_auto_dihedral](/index.php?title=Editor_auto_dihedral&action=edit&redlink=1 "Editor auto dihedral \(page does not exist\)") [editor_auto_origin](/index.php?title=Editor_auto_origin&action=edit&redlink=1 "Editor auto origin \(page does not exist\)") [editor_label_fragments](/index.php?title=Editor_label_fragments&action=edit&redlink=1 "Editor label fragments \(page does not exist\)") [ellipsoid_color](/index.php/Ellipsoid_color "Ellipsoid color") [ellipsoid_probability](/index.php?title=Ellipsoid_probability&action=edit&redlink=1 "Ellipsoid probability \(page does not exist\)") [ellipsoid_quality](/index.php/Ellipsoid_quality "Ellipsoid quality") [ellipsoid_scale](/index.php/Ellipsoid_scale "Ellipsoid scale") [ellipsoid_transparency](/index.php/Ellipsoid_transparency "Ellipsoid transparency")

### F

[fast_idle](/index.php?title=Fast_idle&action=edit&redlink=1 "Fast idle \(page does not exist\)") [fetch_path](/index.php?title=Fetch_path&action=edit&redlink=1 "Fetch path \(page does not exist\)") [field_of_view](/index.php?title=Field_of_view&action=edit&redlink=1 "Field of view \(page does not exist\)") [fit_iterations](/index.php?title=Fit_iterations&action=edit&redlink=1 "Fit iterations \(page does not exist\)") [fit_tolerance](/index.php?title=Fit_tolerance&action=edit&redlink=1 "Fit tolerance \(page does not exist\)") [float_labels](/index.php?title=Float_labels&action=edit&redlink=1 "Float labels \(page does not exist\)") [fog](/index.php/Fog "Fog") [fog_start](/index.php/Fog_start "Fog start") [frame](/index.php/Frame "Frame") [full_screen](/index.php/Full_screen "Full screen")

### G

[gamma](/index.php?title=Gamma&action=edit&redlink=1 "Gamma \(page does not exist\)") [gaussian_b_adjust](/index.php?title=Gaussian_b_adjust&action=edit&redlink=1 "Gaussian b adjust \(page does not exist\)") [gaussian_b_floor](/index.php?title=Gaussian_b_floor&action=edit&redlink=1 "Gaussian b floor \(page does not exist\)") [gaussian_resolution](/index.php?title=Gaussian_resolution&action=edit&redlink=1 "Gaussian resolution \(page does not exist\)") [geometry_export_mode](/index.php/Geometry_export_mode "Geometry export mode") [gl_ambient](/index.php?title=Gl_ambient&action=edit&redlink=1 "Gl ambient \(page does not exist\)") [gradient_max_length](/index.php?title=Gradient_max_length&action=edit&redlink=1 "Gradient max length \(page does not exist\)") [gradient_min_length](/index.php?title=Gradient_min_length&action=edit&redlink=1 "Gradient min length \(page does not exist\)") [gradient_min_slope](/index.php?title=Gradient_min_slope&action=edit&redlink=1 "Gradient min slope \(page does not exist\)") [gradient_normal_min_dot](/index.php?title=Gradient_normal_min_dot&action=edit&redlink=1 "Gradient normal min dot \(page does not exist\)") [gradient_spacing](/index.php?title=Gradient_spacing&action=edit&redlink=1 "Gradient spacing \(page does not exist\)") [gradient_step_size](/index.php?title=Gradient_step_size&action=edit&redlink=1 "Gradient step size \(page does not exist\)") [gradient_symmetry](/index.php?title=Gradient_symmetry&action=edit&redlink=1 "Gradient symmetry \(page does not exist\)") [grid_max](/index.php/Grid_max "Grid max") [grid_mode](/index.php/Grid_mode "Grid mode") [grid_slot](/index.php/Grid_slot "Grid slot") [group_arrow_prefix](/index.php?title=Group_arrow_prefix&action=edit&redlink=1 "Group arrow prefix \(page does not exist\)") [group_auto_mode](/index.php?title=Group_auto_mode&action=edit&redlink=1 "Group auto mode \(page does not exist\)") [group_full_member_names](/index.php?title=Group_full_member_names&action=edit&redlink=1 "Group full member names \(page does not exist\)")

### H

[h_bond_cone](/index.php?title=H_bond_cone&action=edit&redlink=1 "H bond cone \(page does not exist\)") [h_bond_cutoff_center](/index.php?title=H_bond_cutoff_center&action=edit&redlink=1 "H bond cutoff center \(page does not exist\)") [h_bond_cutoff_edge](/index.php?title=H_bond_cutoff_edge&action=edit&redlink=1 "H bond cutoff edge \(page does not exist\)") [h_bond_exclusion](/index.php?title=H_bond_exclusion&action=edit&redlink=1 "H bond exclusion \(page does not exist\)") [h_bond_from_proton](/index.php?title=H_bond_from_proton&action=edit&redlink=1 "H bond from proton \(page does not exist\)") [h_bond_max_angle](/index.php?title=H_bond_max_angle&action=edit&redlink=1 "H bond max angle \(page does not exist\)") [h_bond_power_a](/index.php?title=H_bond_power_a&action=edit&redlink=1 "H bond power a \(page does not exist\)") [h_bond_power_b](/index.php?title=H_bond_power_b&action=edit&redlink=1 "H bond power b \(page does not exist\)") [half_bonds](/index.php/Half_bonds "Half bonds") [hash_max](/index.php/Hash_max "Hash max") [hide_long_bonds](/index.php?title=Hide_long_bonds&action=edit&redlink=1 "Hide long bonds \(page does not exist\)") [hide_underscore_names](/index.php/Hide_underscore_names "Hide underscore names")

### I

[idle_delay](/index.php?title=Idle_delay&action=edit&redlink=1 "Idle delay \(page does not exist\)") [ignore_case](/index.php/Ignore_case "Ignore case") [ignore_pdb_segi](/index.php?title=Ignore_pdb_segi&action=edit&redlink=1 "Ignore pdb segi \(page does not exist\)") [image_copy_always](/index.php?title=Image_copy_always&action=edit&redlink=1 "Image copy always \(page does not exist\)") [image_dots_per_inch](/index.php?title=Image_dots_per_inch&action=edit&redlink=1 "Image dots per inch \(page does not exist\)") [internal_feedback](/index.php/Internal_feedback "Internal feedback") [internal_gui](/index.php/Internal_gui "Internal gui") [internal_gui_control_size](/index.php/Internal_gui_control_size "Internal gui control size") [internal_gui_mode](/index.php/Internal_gui_mode "Internal gui mode") [internal_gui_width](/index.php/Internal_gui_width "Internal gui width") [internal_prompt](/index.php/Internal_prompt "Internal prompt") [isomesh_auto_state](/index.php?title=Isomesh_auto_state&action=edit&redlink=1 "Isomesh auto state \(page does not exist\)")

### J

### K

### L

[label_angle_digits](/index.php/Label_angle_digits "Label angle digits") [label_color](/index.php/Label_color "Label color") [label_digits](/index.php?title=Label_digits&action=edit&redlink=1 "Label digits \(page does not exist\)") [label_dihedral_digits](/index.php/Label_dihedral_digits "Label dihedral digits") [label_distance_digits](/index.php/Label_distance_digits "Label distance digits") [label_font_id](/index.php/Label_font_id "Label font id") [label_outline_color](/index.php/Label_outline_color "Label outline color") [label_position](/index.php/Label_position "Label position") [label_shadow_mode](/index.php/Label_shadow_mode "Label shadow mode") [label_size](/index.php/Label_size "Label size") [legacy_mouse_zoom](/index.php?title=Legacy_mouse_zoom&action=edit&redlink=1 "Legacy mouse zoom \(page does not exist\)") [legacy_vdw_radii](/index.php?title=Legacy_vdw_radii&action=edit&redlink=1 "Legacy vdw radii \(page does not exist\)") [light](/index.php/Light "Light") [light2](/index.php/Light2 "Light2") [light3](/index.php/Light3 "Light3") [light4](/index.php/Light4 "Light4") [light5](/index.php/Light5 "Light5") [light6](/index.php/Light6 "Light6") [light7](/index.php/Light7 "Light7") [light8](/index.php/Light8 "Light8") [light9](/index.php/Light9 "Light9") [light_count](/index.php/Light_count "Light count") [line_color](/index.php/Line_color "Line color") [line_radius](/index.php?title=Line_radius&action=edit&redlink=1 "Line radius \(page does not exist\)") [line_smooth](/index.php/Line_smooth "Line smooth") [line_stick_helper](/index.php?title=Line_stick_helper&action=edit&redlink=1 "Line stick helper \(page does not exist\)") [line_width](/index.php/Line_width "Line width") [log_box_selections](/index.php?title=Log_box_selections&action=edit&redlink=1 "Log box selections \(page does not exist\)") [log_conformations](/index.php?title=Log_conformations&action=edit&redlink=1 "Log conformations \(page does not exist\)") [logging](/index.php?title=Logging&action=edit&redlink=1 "Logging \(page does not exist\)")

### M

[map_auto_expand_sym](/index.php/Map_auto_expand_sym "Map auto expand sym") [matrix_mode](/index.php/Matrix_mode "Matrix mode") [max_threads](/index.php?title=Max_threads&action=edit&redlink=1 "Max threads \(page does not exist\)") [max_triangles](/index.php?title=Max_triangles&action=edit&redlink=1 "Max triangles \(page does not exist\)") [mesh_carve_cutoff](/index.php?title=Mesh_carve_cutoff&action=edit&redlink=1 "Mesh carve cutoff \(page does not exist\)") [mesh_carve_selection](/index.php?title=Mesh_carve_selection&action=edit&redlink=1 "Mesh carve selection \(page does not exist\)") [mesh_carve_state](/index.php?title=Mesh_carve_state&action=edit&redlink=1 "Mesh carve state \(page does not exist\)") [mesh_clear_cutoff](/index.php?title=Mesh_clear_cutoff&action=edit&redlink=1 "Mesh clear cutoff \(page does not exist\)") [mesh_clear_selection](/index.php?title=Mesh_clear_selection&action=edit&redlink=1 "Mesh clear selection \(page does not exist\)") [mesh_clear_state](/index.php?title=Mesh_clear_state&action=edit&redlink=1 "Mesh clear state \(page does not exist\)") [mesh_color](/index.php/Mesh_color "Mesh color") [mesh_cutoff](/index.php?title=Mesh_cutoff&action=edit&redlink=1 "Mesh cutoff \(page does not exist\)") [mesh_grid_max](/index.php?title=Mesh_grid_max&action=edit&redlink=1 "Mesh grid max \(page does not exist\)") [mesh_lighting](/index.php/Mesh_lighting "Mesh lighting") [mesh_mode](/index.php/Mesh_mode "Mesh mode") [mesh_negative_color](/index.php/Mesh_negative_color "Mesh negative color") [mesh_negative_visible](/index.php/Mesh_negative_visible "Mesh negative visible") [mesh_normals](/index.php?title=Mesh_normals&action=edit&redlink=1 "Mesh normals \(page does not exist\)") [mesh_quality](/index.php/Mesh_quality "Mesh quality") [mesh_radius](/index.php?title=Mesh_radius&action=edit&redlink=1 "Mesh radius \(page does not exist\)") [mesh_skip](/index.php?title=Mesh_skip&action=edit&redlink=1 "Mesh skip \(page does not exist\)") [mesh_solvent](/index.php?title=Mesh_solvent&action=edit&redlink=1 "Mesh solvent \(page does not exist\)") [mesh_type](/index.php/Mesh_type "Mesh type") [mesh_width](/index.php/Mesh_width "Mesh width") [min_mesh_spacing](/index.php/Min_mesh_spacing "Min mesh spacing") [moe_separate_chains](/index.php?title=Moe_separate_chains&action=edit&redlink=1 "Moe separate chains \(page does not exist\)") [mouse_grid](/index.php?title=Mouse_grid&action=edit&redlink=1 "Mouse grid \(page does not exist\)") [mouse_limit](/index.php?title=Mouse_limit&action=edit&redlink=1 "Mouse limit \(page does not exist\)") [mouse_restart_movie_delay](/index.php?title=Mouse_restart_movie_delay&action=edit&redlink=1 "Mouse restart movie delay \(page does not exist\)") [mouse_scale](/index.php?title=Mouse_scale&action=edit&redlink=1 "Mouse scale \(page does not exist\)") [mouse_selection_mode](/index.php?title=Mouse_selection_mode&action=edit&redlink=1 "Mouse selection mode \(page does not exist\)") [mouse_wheel_scale](/index.php?title=Mouse_wheel_scale&action=edit&redlink=1 "Mouse wheel scale \(page does not exist\)") [movie_animate_by_frame](/index.php?title=Movie_animate_by_frame&action=edit&redlink=1 "Movie animate by frame \(page does not exist\)") [movie_delay](/index.php?title=Movie_delay&action=edit&redlink=1 "Movie delay \(page does not exist\)") [movie_fps](/index.php/Movie_fps "Movie fps") [movie_loop](/index.php/Movie_loop "Movie loop") [movie_rock](/index.php?title=Movie_rock&action=edit&redlink=1 "Movie rock \(page does not exist\)") [multiplex](/index.php?title=Multiplex&action=edit&redlink=1 "Multiplex \(page does not exist\)")

### N

[no_idle](/index.php?title=No_idle&action=edit&redlink=1 "No idle \(page does not exist\)") [nonbonded_size](/index.php?title=Nonbonded_size&action=edit&redlink=1 "Nonbonded size \(page does not exist\)") [nonbonded_transparency](/index.php?title=Nonbonded_transparency&action=edit&redlink=1 "Nonbonded transparency \(page does not exist\)") [normal_workaround](/index.php?title=Normal_workaround&action=edit&redlink=1 "Normal workaround \(page does not exist\)") [normalize_ccp4_maps](/index.php/Normalize_ccp4_maps "Normalize ccp4 maps") [normalize_grd_maps](/index.php?title=Normalize_grd_maps&action=edit&redlink=1 "Normalize grd maps \(page does not exist\)") [normalize_o_maps](/index.php?title=Normalize_o_maps&action=edit&redlink=1 "Normalize o maps \(page does not exist\)") [nvidia_bugs](/index.php?title=Nvidia_bugs&action=edit&redlink=1 "Nvidia bugs \(page does not exist\)")

### O

[opaque_background](/index.php/Opaque_background "Opaque background") [orthoscopic](/index.php/Orthoscopic "Orthoscopic") [overlay](/index.php?title=Overlay&action=edit&redlink=1 "Overlay \(page does not exist\)") [overlay_lines](/index.php?title=Overlay_lines&action=edit&redlink=1 "Overlay lines \(page does not exist\)")

### P

[pdb_conect_all](/index.php/Pdb_conect_all "Pdb conect all") [pdb_discrete_chains](/index.php?title=Pdb_discrete_chains&action=edit&redlink=1 "Pdb discrete chains \(page does not exist\)") [pdb_echo_tags](/index.php/Pdb_echo_tags "Pdb echo tags") [pdb_formal_charges](/index.php?title=Pdb_formal_charges&action=edit&redlink=1 "Pdb formal charges \(page does not exist\)") [pdb_hetatm_guess_valences](/index.php?title=Pdb_hetatm_guess_valences&action=edit&redlink=1 "Pdb hetatm guess valences \(page does not exist\)") [pdb_hetatm_sort](/index.php?title=Pdb_hetatm_sort&action=edit&redlink=1 "Pdb hetatm sort \(page does not exist\)") [pdb_honor_model_number](/index.php?title=Pdb_honor_model_number&action=edit&redlink=1 "Pdb honor model number \(page does not exist\)") [pdb_insertions_go_first](/index.php?title=Pdb_insertions_go_first&action=edit&redlink=1 "Pdb insertions go first \(page does not exist\)") [pdb_insure_orthogonal](/index.php?title=Pdb_insure_orthogonal&action=edit&redlink=1 "Pdb insure orthogonal \(page does not exist\)") [pdb_literal_names](/index.php?title=Pdb_literal_names&action=edit&redlink=1 "Pdb literal names \(page does not exist\)") [pdb_no_end_record](/index.php?title=Pdb_no_end_record&action=edit&redlink=1 "Pdb no end record \(page does not exist\)") [pdb_reformat_names_mode](/index.php/Pdb_reformat_names_mode "Pdb reformat names mode") [pdb_retain_ids](/index.php/Pdb_retain_ids "Pdb retain ids") [pdb_standard_order](/index.php?title=Pdb_standard_order&action=edit&redlink=1 "Pdb standard order \(page does not exist\)") [pdb_truncate_residue_name](/index.php?title=Pdb_truncate_residue_name&action=edit&redlink=1 "Pdb truncate residue name \(page does not exist\)") [pdb_unbond_cations](/index.php?title=Pdb_unbond_cations&action=edit&redlink=1 "Pdb unbond cations \(page does not exist\)") [pdb_use_ter_records](/index.php/Pdb_use_ter_records "Pdb use ter records") [pickable](/index.php/Pickable "Pickable") [png_file_gamma](/index.php?title=Png_file_gamma&action=edit&redlink=1 "Png file gamma \(page does not exist\)") [png_screen_gamma](/index.php?title=Png_screen_gamma&action=edit&redlink=1 "Png screen gamma \(page does not exist\)") [power](/index.php?title=Power&action=edit&redlink=1 "Power \(page does not exist\)") [pqr_no_chain_id](/index.php?title=Pqr_no_chain_id&action=edit&redlink=1 "Pqr no chain id \(page does not exist\)") [presentation](/index.php?title=Presentation&action=edit&redlink=1 "Presentation \(page does not exist\)") [presentation_auto_quit](/index.php?title=Presentation_auto_quit&action=edit&redlink=1 "Presentation auto quit \(page does not exist\)") [presentation_auto_start](/index.php?title=Presentation_auto_start&action=edit&redlink=1 "Presentation auto start \(page does not exist\)") [presentation_mode](/index.php?title=Presentation_mode&action=edit&redlink=1 "Presentation mode \(page does not exist\)") [preserve_chempy_ids](/index.php?title=Preserve_chempy_ids&action=edit&redlink=1 "Preserve chempy ids \(page does not exist\)") [pymol_space_max_blue](/index.php?title=Pymol_space_max_blue&action=edit&redlink=1 "Pymol space max blue \(page does not exist\)") [pymol_space_max_green](/index.php?title=Pymol_space_max_green&action=edit&redlink=1 "Pymol space max green \(page does not exist\)") [pymol_space_max_red](/index.php?title=Pymol_space_max_red&action=edit&redlink=1 "Pymol space max red \(page does not exist\)") [pymol_space_min_factor](/index.php?title=Pymol_space_min_factor&action=edit&redlink=1 "Pymol space min factor \(page does not exist\)")

### Q

### R

[raise_exceptions](/index.php?title=Raise_exceptions&action=edit&redlink=1 "Raise exceptions \(page does not exist\)") [ramp_blend_nearby_colors](/index.php?title=Ramp_blend_nearby_colors&action=edit&redlink=1 "Ramp blend nearby colors \(page does not exist\)") [rank_assisted_sorts](/index.php?title=Rank_assisted_sorts&action=edit&redlink=1 "Rank assisted sorts \(page does not exist\)") [ray_blend_blue](/index.php/Ray_blend_blue "Ray blend blue") [ray_blend_colors](/index.php/Ray_blend_colors "Ray blend colors") [ray_blend_green](/index.php/Ray_blend_green "Ray blend green") [ray_blend_red](/index.php/Ray_blend_red "Ray blend red") [ray_clip_shadows](/index.php?title=Ray_clip_shadows&action=edit&redlink=1 "Ray clip shadows \(page does not exist\)") [ray_color_ramps](/index.php?title=Ray_color_ramps&action=edit&redlink=1 "Ray color ramps \(page does not exist\)") [ray_default_renderer](/index.php?title=Ray_default_renderer&action=edit&redlink=1 "Ray default renderer \(page does not exist\)") [ray_direct_shade](/index.php/Ray_direct_shade "Ray direct shade") [ray_hint_camera](/index.php?title=Ray_hint_camera&action=edit&redlink=1 "Ray hint camera \(page does not exist\)") [ray_hint_shadow](/index.php?title=Ray_hint_shadow&action=edit&redlink=1 "Ray hint shadow \(page does not exist\)") [ray_improve_shadows](/index.php?title=Ray_improve_shadows&action=edit&redlink=1 "Ray improve shadows \(page does not exist\)") [ray_interior_color](/index.php?title=Ray_interior_color&action=edit&redlink=1 "Ray interior color \(page does not exist\)") [ray_interior_mode](/index.php?title=Ray_interior_mode&action=edit&redlink=1 "Ray interior mode \(page does not exist\)") [ray_interior_reflect](/index.php?title=Ray_interior_reflect&action=edit&redlink=1 "Ray interior reflect \(page does not exist\)") [ray_interior_shadows](/index.php?title=Ray_interior_shadows&action=edit&redlink=1 "Ray interior shadows \(page does not exist\)") [ray_interior_texture](/index.php?title=Ray_interior_texture&action=edit&redlink=1 "Ray interior texture \(page does not exist\)") [ray_label_specular](/index.php?title=Ray_label_specular&action=edit&redlink=1 "Ray label specular \(page does not exist\)") [ray_legacy_lighting](/index.php?title=Ray_legacy_lighting&action=edit&redlink=1 "Ray legacy lighting \(page does not exist\)") [ray_max_passes](/index.php?title=Ray_max_passes&action=edit&redlink=1 "Ray max passes \(page does not exist\)") [ray_opaque_background](/index.php/Ray_opaque_background "Ray opaque background") [ray_orthoscopic](/index.php/Ray_orthoscopic "Ray orthoscopic") [ray_oversample_cutoff](/index.php?title=Ray_oversample_cutoff&action=edit&redlink=1 "Ray oversample cutoff \(page does not exist\)") [ray_pixel_scale](/index.php?title=Ray_pixel_scale&action=edit&redlink=1 "Ray pixel scale \(page does not exist\)") [ray_scatter](/index.php?title=Ray_scatter&action=edit&redlink=1 "Ray scatter \(page does not exist\)") [ray_shadow](/index.php/Ray_shadow "Ray shadow") [ray_shadow_decay_factor](/index.php?title=Ray_shadow_decay_factor&action=edit&redlink=1 "Ray shadow decay factor \(page does not exist\)") [ray_shadow_decay_range](/index.php?title=Ray_shadow_decay_range&action=edit&redlink=1 "Ray shadow decay range \(page does not exist\)") [ray_shadow_fudge](/index.php?title=Ray_shadow_fudge&action=edit&redlink=1 "Ray shadow fudge \(page does not exist\)") [ray_shadows](/index.php/Ray_shadows "Ray shadows") [ray_spec_local](/index.php?title=Ray_spec_local&action=edit&redlink=1 "Ray spec local \(page does not exist\)") [ray_texture](/index.php/Ray_texture "Ray texture") [ray_texture_settings](/index.php/Ray_texture_settings "Ray texture settings") [ray_trace_color](/index.php?title=Ray_trace_color&action=edit&redlink=1 "Ray trace color \(page does not exist\)") [ray_trace_depth_factor](/index.php?title=Ray_trace_depth_factor&action=edit&redlink=1 "Ray trace depth factor \(page does not exist\)") [ray_trace_disco_factor](/index.php/Ray_trace_disco_factor "Ray trace disco factor") [ray_trace_fog](/index.php/Ray_trace_fog "Ray trace fog") [ray_trace_fog_start](/index.php?title=Ray_trace_fog_start&action=edit&redlink=1 "Ray trace fog start \(page does not exist\)") [ray_trace_frames](/index.php/Ray_trace_frames "Ray trace frames") [ray_trace_gain](/index.php/Ray_trace_gain "Ray trace gain") [ray_trace_mode](/index.php/Ray_trace_mode "Ray trace mode") [ray_trace_persist_cutoff](/index.php?title=Ray_trace_persist_cutoff&action=edit&redlink=1 "Ray trace persist cutoff \(page does not exist\)") [ray_trace_slope_factor](/index.php?title=Ray_trace_slope_factor&action=edit&redlink=1 "Ray trace slope factor \(page does not exist\)") [ray_trace_trans_cutoff](/index.php?title=Ray_trace_trans_cutoff&action=edit&redlink=1 "Ray trace trans cutoff \(page does not exist\)") [ray_transparency_contrast](/index.php?title=Ray_transparency_contrast&action=edit&redlink=1 "Ray transparency contrast \(page does not exist\)") [ray_transparency_oblique](/index.php?title=Ray_transparency_oblique&action=edit&redlink=1 "Ray transparency oblique \(page does not exist\)") [ray_transparency_oblique_power](/index.php?title=Ray_transparency_oblique_power&action=edit&redlink=1 "Ray transparency oblique power \(page does not exist\)") [ray_transparency_shadows](/index.php?title=Ray_transparency_shadows&action=edit&redlink=1 "Ray transparency shadows \(page does not exist\)") [ray_transparency_spec_cut](/index.php?title=Ray_transparency_spec_cut&action=edit&redlink=1 "Ray transparency spec cut \(page does not exist\)") [ray_transparency_specular](/index.php?title=Ray_transparency_specular&action=edit&redlink=1 "Ray transparency specular \(page does not exist\)") [ray_triangle_fudge](/index.php?title=Ray_triangle_fudge&action=edit&redlink=1 "Ray triangle fudge \(page does not exist\)") [reflect](/index.php/Reflect "Reflect") [reflect_power](/index.php?title=Reflect_power&action=edit&redlink=1 "Reflect power \(page does not exist\)") [retain_order](/index.php/Retain_order "Retain order") [ribbon_color](/index.php/Ribbon_color "Ribbon color") [ribbon_nucleic_acid_mode](/index.php?title=Ribbon_nucleic_acid_mode&action=edit&redlink=1 "Ribbon nucleic acid mode \(page does not exist\)") [ribbon_power](/index.php?title=Ribbon_power&action=edit&redlink=1 "Ribbon power \(page does not exist\)") [ribbon_power_b](/index.php?title=Ribbon_power_b&action=edit&redlink=1 "Ribbon power b \(page does not exist\)") [ribbon_radius](/index.php?title=Ribbon_radius&action=edit&redlink=1 "Ribbon radius \(page does not exist\)") [ribbon_sampling](/index.php/Ribbon_sampling "Ribbon sampling") [ribbon_side_chain_helper](/index.php/Ribbon_side_chain_helper "Ribbon side chain helper") [ribbon_smooth](/index.php/Ribbon_smooth "Ribbon smooth") [ribbon_throw](/index.php?title=Ribbon_throw&action=edit&redlink=1 "Ribbon throw \(page does not exist\)") [ribbon_trace_atoms](/index.php?title=Ribbon_trace_atoms&action=edit&redlink=1 "Ribbon trace atoms \(page does not exist\)") [ribbon_width](/index.php/Ribbon_width "Ribbon width") [robust_logs](/index.php?title=Robust_logs&action=edit&redlink=1 "Robust logs \(page does not exist\)") [rock](/index.php/Rock "Rock") [rock_delay](/index.php?title=Rock_delay&action=edit&redlink=1 "Rock delay \(page does not exist\)") [roving_byres](/index.php/Roving_byres "Roving byres") [roving_cartoon](/index.php?title=Roving_cartoon&action=edit&redlink=1 "Roving cartoon \(page does not exist\)") [roving_delay](/index.php/Roving_delay "Roving delay") [roving_detail](/index.php/Roving_detail "Roving detail") [roving_isomesh](/index.php?title=Roving_isomesh&action=edit&redlink=1 "Roving isomesh \(page does not exist\)") [roving_isosurface](/index.php?title=Roving_isosurface&action=edit&redlink=1 "Roving isosurface \(page does not exist\)") [roving_labels](/index.php?title=Roving_labels&action=edit&redlink=1 "Roving labels \(page does not exist\)") [roving_lines](/index.php?title=Roving_lines&action=edit&redlink=1 "Roving lines \(page does not exist\)") [roving_map1_level](/index.php?title=Roving_map1_level&action=edit&redlink=1 "Roving map1 level \(page does not exist\)") [roving_map1_name](/index.php?title=Roving_map1_name&action=edit&redlink=1 "Roving map1 name \(page does not exist\)") [roving_map2_level](/index.php?title=Roving_map2_level&action=edit&redlink=1 "Roving map2 level \(page does not exist\)") [roving_map2_name](/index.php?title=Roving_map2_name&action=edit&redlink=1 "Roving map2 name \(page does not exist\)") [roving_map3_level](/index.php?title=Roving_map3_level&action=edit&redlink=1 "Roving map3 level \(page does not exist\)") [roving_map3_name](/index.php?title=Roving_map3_name&action=edit&redlink=1 "Roving map3 name \(page does not exist\)") [roving_nb_spheres](/index.php/Roving_nb_spheres "Roving nb spheres") [roving_nonbonded](/index.php?title=Roving_nonbonded&action=edit&redlink=1 "Roving nonbonded \(page does not exist\)") [roving_origin](/index.php?title=Roving_origin&action=edit&redlink=1 "Roving origin \(page does not exist\)") [roving_origin_z](/index.php?title=Roving_origin_z&action=edit&redlink=1 "Roving origin z \(page does not exist\)") [roving_origin_z_cushion](/index.php?title=Roving_origin_z_cushion&action=edit&redlink=1 "Roving origin z cushion \(page does not exist\)") [roving_polar_contacts](/index.php?title=Roving_polar_contacts&action=edit&redlink=1 "Roving polar contacts \(page does not exist\)") [roving_polar_cutoff](/index.php?title=Roving_polar_cutoff&action=edit&redlink=1 "Roving polar cutoff \(page does not exist\)") [roving_ribbon](/index.php?title=Roving_ribbon&action=edit&redlink=1 "Roving ribbon \(page does not exist\)") [roving_selection](/index.php?title=Roving_selection&action=edit&redlink=1 "Roving selection \(page does not exist\)") [roving_spheres](/index.php?title=Roving_spheres&action=edit&redlink=1 "Roving spheres \(page does not exist\)") [roving_sticks](/index.php?title=Roving_sticks&action=edit&redlink=1 "Roving sticks \(page does not exist\)")

### S

[scene_animation](/index.php?title=Scene_animation&action=edit&redlink=1 "Scene animation \(page does not exist\)") [scene_animation_duration](/index.php?title=Scene_animation_duration&action=edit&redlink=1 "Scene animation duration \(page does not exist\)") [scene_buttons](/index.php/Scene_buttons "Scene buttons") [scene_buttons_mode](/index.php?title=Scene_buttons_mode&action=edit&redlink=1 "Scene buttons mode \(page does not exist\)") [scene_current_name](/index.php?title=Scene_current_name&action=edit&redlink=1 "Scene current name \(page does not exist\)") [scene_loop](/index.php?title=Scene_loop&action=edit&redlink=1 "Scene loop \(page does not exist\)") [scene_restart_movie_delay](/index.php?title=Scene_restart_movie_delay&action=edit&redlink=1 "Scene restart movie delay \(page does not exist\)") [scenes_changed](/index.php?title=Scenes_changed&action=edit&redlink=1 "Scenes changed \(page does not exist\)") [sculpt_angl_weight](/index.php?title=Sculpt_angl_weight&action=edit&redlink=1 "Sculpt angl weight \(page does not exist\)") [sculpt_auto_center](/index.php?title=Sculpt_auto_center&action=edit&redlink=1 "Sculpt auto center \(page does not exist\)") [sculpt_avd_excl](/index.php?title=Sculpt_avd_excl&action=edit&redlink=1 "Sculpt avd excl \(page does not exist\)") [sculpt_avd_gap](/index.php?title=Sculpt_avd_gap&action=edit&redlink=1 "Sculpt avd gap \(page does not exist\)") [sculpt_avd_range](/index.php?title=Sculpt_avd_range&action=edit&redlink=1 "Sculpt avd range \(page does not exist\)") [sculpt_avd_weight](/index.php?title=Sculpt_avd_weight&action=edit&redlink=1 "Sculpt avd weight \(page does not exist\)") [sculpt_bond_weight](/index.php?title=Sculpt_bond_weight&action=edit&redlink=1 "Sculpt bond weight \(page does not exist\)") [sculpt_field_mask](/index.php/Sculpt_field_mask "Sculpt field mask") [sculpt_hb_overlap](/index.php?title=Sculpt_hb_overlap&action=edit&redlink=1 "Sculpt hb overlap \(page does not exist\)") [sculpt_hb_overlap_base](/index.php?title=Sculpt_hb_overlap_base&action=edit&redlink=1 "Sculpt hb overlap base \(page does not exist\)") [sculpt_line_weight](/index.php?title=Sculpt_line_weight&action=edit&redlink=1 "Sculpt line weight \(page does not exist\)") [sculpt_max_max](/index.php?title=Sculpt_max_max&action=edit&redlink=1 "Sculpt max max \(page does not exist\)") [sculpt_max_min](/index.php?title=Sculpt_max_min&action=edit&redlink=1 "Sculpt max min \(page does not exist\)") [sculpt_max_scale](/index.php?title=Sculpt_max_scale&action=edit&redlink=1 "Sculpt max scale \(page does not exist\)") [sculpt_max_weight](/index.php?title=Sculpt_max_weight&action=edit&redlink=1 "Sculpt max weight \(page does not exist\)") [sculpt_memory](/index.php?title=Sculpt_memory&action=edit&redlink=1 "Sculpt memory \(page does not exist\)") [sculpt_min_max](/index.php?title=Sculpt_min_max&action=edit&redlink=1 "Sculpt min max \(page does not exist\)") [sculpt_min_min](/index.php?title=Sculpt_min_min&action=edit&redlink=1 "Sculpt min min \(page does not exist\)") [sculpt_min_scale](/index.php?title=Sculpt_min_scale&action=edit&redlink=1 "Sculpt min scale \(page does not exist\)") [sculpt_min_weight](/index.php?title=Sculpt_min_weight&action=edit&redlink=1 "Sculpt min weight \(page does not exist\)") [sculpt_nb_interval](/index.php?title=Sculpt_nb_interval&action=edit&redlink=1 "Sculpt nb interval \(page does not exist\)") [sculpt_plan_weight](/index.php?title=Sculpt_plan_weight&action=edit&redlink=1 "Sculpt plan weight \(page does not exist\)") [sculpt_pyra_weight](/index.php?title=Sculpt_pyra_weight&action=edit&redlink=1 "Sculpt pyra weight \(page does not exist\)") [sculpt_tors_tolerance](/index.php?title=Sculpt_tors_tolerance&action=edit&redlink=1 "Sculpt tors tolerance \(page does not exist\)") [sculpt_tors_weight](/index.php?title=Sculpt_tors_weight&action=edit&redlink=1 "Sculpt tors weight \(page does not exist\)") [sculpt_tri_max](/index.php?title=Sculpt_tri_max&action=edit&redlink=1 "Sculpt tri max \(page does not exist\)") [sculpt_tri_min](/index.php?title=Sculpt_tri_min&action=edit&redlink=1 "Sculpt tri min \(page does not exist\)") [sculpt_tri_mode](/index.php?title=Sculpt_tri_mode&action=edit&redlink=1 "Sculpt tri mode \(page does not exist\)") [sculpt_tri_scale](/index.php?title=Sculpt_tri_scale&action=edit&redlink=1 "Sculpt tri scale \(page does not exist\)") [sculpt_tri_weight](/index.php?title=Sculpt_tri_weight&action=edit&redlink=1 "Sculpt tri weight \(page does not exist\)") [sculpt_vdw_scale](/index.php?title=Sculpt_vdw_scale&action=edit&redlink=1 "Sculpt vdw scale \(page does not exist\)") [sculpt_vdw_scale14](/index.php?title=Sculpt_vdw_scale14&action=edit&redlink=1 "Sculpt vdw scale14 \(page does not exist\)") [sculpt_vdw_vis_max](/index.php?title=Sculpt_vdw_vis_max&action=edit&redlink=1 "Sculpt vdw vis max \(page does not exist\)") [sculpt_vdw_vis_mid](/index.php?title=Sculpt_vdw_vis_mid&action=edit&redlink=1 "Sculpt vdw vis mid \(page does not exist\)") [sculpt_vdw_vis_min](/index.php?title=Sculpt_vdw_vis_min&action=edit&redlink=1 "Sculpt vdw vis min \(page does not exist\)") [sculpt_vdw_vis_mode](/index.php?title=Sculpt_vdw_vis_mode&action=edit&redlink=1 "Sculpt vdw vis mode \(page does not exist\)") [sculpt_vdw_weight](/index.php?title=Sculpt_vdw_weight&action=edit&redlink=1 "Sculpt vdw weight \(page does not exist\)") [sculpt_vdw_weight14](/index.php?title=Sculpt_vdw_weight14&action=edit&redlink=1 "Sculpt vdw weight14 \(page does not exist\)") [sculpting](/index.php?title=Sculpting&action=edit&redlink=1 "Sculpting \(page does not exist\)") [sculpting_cycles](/index.php?title=Sculpting_cycles&action=edit&redlink=1 "Sculpting cycles \(page does not exist\)") [sdof_drag_scale](/index.php?title=Sdof_drag_scale&action=edit&redlink=1 "Sdof drag scale \(page does not exist\)") [secondary_structure](/index.php?title=Secondary_structure&action=edit&redlink=1 "Secondary structure \(page does not exist\)") [security](/index.php?title=Security&action=edit&redlink=1 "Security \(page does not exist\)") [sel_counter](/index.php?title=Sel_counter&action=edit&redlink=1 "Sel counter \(page does not exist\)") [selection_overlay](/index.php/Selection_overlay "Selection overlay") [selection_round_points](/index.php/Selection_round_points "Selection round points") [selection_visible_only](/index.php/Selection_visible_only "Selection visible only") [selection_width](/index.php/Selection_width "Selection width") [selection_width_max](/index.php?title=Selection_width_max&action=edit&redlink=1 "Selection width max \(page does not exist\)") [selection_width_scale](/index.php?title=Selection_width_scale&action=edit&redlink=1 "Selection width scale \(page does not exist\)") [seq_view](/index.php/Seq_view "Seq view") [seq_view_alignment](/index.php/Seq_view_alignment "Seq view alignment") [seq_view_color](/index.php/Seq_view_color "Seq view color") [seq_view_discrete_by_state](/index.php/Seq_view_discrete_by_state "Seq view discrete by state") [seq_view_fill_char](/index.php/Seq_view_fill_char "Seq view fill char") [seq_view_fill_color](/index.php?title=Seq_view_fill_color&action=edit&redlink=1 "Seq view fill color \(page does not exist\)") [seq_view_format](/index.php/Seq_view_format "Seq view format") [seq_view_label_color](/index.php/Seq_view_label_color "Seq view label color") [seq_view_label_mode](/index.php/Seq_view_label_mode "Seq view label mode") [seq_view_label_spacing](/index.php/Seq_view_label_spacing "Seq view label spacing") [seq_view_label_start](/index.php?title=Seq_view_label_start&action=edit&redlink=1 "Seq view label start \(page does not exist\)") [seq_view_location](/index.php/Seq_view_location "Seq view location") [seq_view_overlay](/index.php/Seq_view_overlay "Seq view overlay") [seq_view_unaligned_color](/index.php/Seq_view_unaligned_color "Seq view unaligned color") [seq_view_unaligned_mode](/index.php/Seq_view_unaligned_mode "Seq view unaligned mode") [session_cache_optimize](/index.php?title=Session_cache_optimize&action=edit&redlink=1 "Session cache optimize \(page does not exist\)") [session_changed](/index.php?title=Session_changed&action=edit&redlink=1 "Session changed \(page does not exist\)") [session_compression](/index.php?title=Session_compression&action=edit&redlink=1 "Session compression \(page does not exist\)") [session_file](/index.php?title=Session_file&action=edit&redlink=1 "Session file \(page does not exist\)") [session_migration](/index.php?title=Session_migration&action=edit&redlink=1 "Session migration \(page does not exist\)") [session_version_check](/index.php?title=Session_version_check&action=edit&redlink=1 "Session version check \(page does not exist\)") [shininess](/index.php/Shininess "Shininess") [show_alpha_checker](/index.php?title=Show_alpha_checker&action=edit&redlink=1 "Show alpha checker \(page does not exist\)") [show_progress](/index.php?title=Show_progress&action=edit&redlink=1 "Show progress \(page does not exist\)") [simplify_display_lists](/index.php?title=Simplify_display_lists&action=edit&redlink=1 "Simplify display lists \(page does not exist\)") [single_image](/index.php?title=Single_image&action=edit&redlink=1 "Single image \(page does not exist\)") [slice_dynamic_grid](/index.php?title=Slice_dynamic_grid&action=edit&redlink=1 "Slice dynamic grid \(page does not exist\)") [slice_dynamic_grid_resolution](/index.php?title=Slice_dynamic_grid_resolution&action=edit&redlink=1 "Slice dynamic grid resolution \(page does not exist\)") [slice_grid](/index.php?title=Slice_grid&action=edit&redlink=1 "Slice grid \(page does not exist\)") [slice_height_map](/index.php?title=Slice_height_map&action=edit&redlink=1 "Slice height map \(page does not exist\)") [slice_height_scale](/index.php?title=Slice_height_scale&action=edit&redlink=1 "Slice height scale \(page does not exist\)") [slice_track_camera](/index.php?title=Slice_track_camera&action=edit&redlink=1 "Slice track camera \(page does not exist\)") [slow_idle](/index.php?title=Slow_idle&action=edit&redlink=1 "Slow idle \(page does not exist\)") [smooth_color_triangle](/index.php?title=Smooth_color_triangle&action=edit&redlink=1 "Smooth color triangle \(page does not exist\)") [solvent_radius](/index.php/Solvent_radius "Solvent radius") [spec_count](/index.php?title=Spec_count&action=edit&redlink=1 "Spec count \(page does not exist\)") [spec_direct](/index.php/Spec_direct "Spec direct") [spec_direct_power](/index.php/Spec_direct_power "Spec direct power") [spec_power](/index.php/Spec_power "Spec power") [spec_reflect](/index.php/Spec_reflect "Spec reflect") [specular](/index.php/Specular "Specular") [specular_intensity](/index.php/Specular_intensity "Specular intensity") [sphere_color](/index.php/Sphere_color "Sphere color") [sphere_mode](/index.php/Sphere_mode "Sphere mode") [sphere_point_max_size](/index.php/Sphere_point_max_size "Sphere point max size") [sphere_point_size](/index.php/Sphere_point_size "Sphere point size") [sphere_quality](/index.php/Sphere_quality "Sphere quality") [sphere_scale](/index.php/Sphere_scale "Sphere scale") [sphere_solvent](/index.php/Sphere_solvent "Sphere solvent") [sphere_transparency](/index.php/Sphere_transparency "Sphere transparency") [spheroid_fill](/index.php?title=Spheroid_fill&action=edit&redlink=1 "Spheroid fill \(page does not exist\)") [spheroid_scale](/index.php?title=Spheroid_scale&action=edit&redlink=1 "Spheroid scale \(page does not exist\)") [spheroid_smooth](/index.php?title=Spheroid_smooth&action=edit&redlink=1 "Spheroid smooth \(page does not exist\)") [ss_helix_phi_exclude](/index.php?title=Ss_helix_phi_exclude&action=edit&redlink=1 "Ss helix phi exclude \(page does not exist\)") [ss_helix_phi_include](/index.php?title=Ss_helix_phi_include&action=edit&redlink=1 "Ss helix phi include \(page does not exist\)") [ss_helix_phi_target](/index.php?title=Ss_helix_phi_target&action=edit&redlink=1 "Ss helix phi target \(page does not exist\)") [ss_helix_psi_exclude](/index.php?title=Ss_helix_psi_exclude&action=edit&redlink=1 "Ss helix psi exclude \(page does not exist\)") [ss_helix_psi_include](/index.php?title=Ss_helix_psi_include&action=edit&redlink=1 "Ss helix psi include \(page does not exist\)") [ss_helix_psi_target](/index.php?title=Ss_helix_psi_target&action=edit&redlink=1 "Ss helix psi target \(page does not exist\)") [ss_strand_phi_exclude](/index.php?title=Ss_strand_phi_exclude&action=edit&redlink=1 "Ss strand phi exclude \(page does not exist\)") [ss_strand_phi_include](/index.php?title=Ss_strand_phi_include&action=edit&redlink=1 "Ss strand phi include \(page does not exist\)") [ss_strand_phi_target](/index.php?title=Ss_strand_phi_target&action=edit&redlink=1 "Ss strand phi target \(page does not exist\)") [ss_strand_psi_exclude](/index.php?title=Ss_strand_psi_exclude&action=edit&redlink=1 "Ss strand psi exclude \(page does not exist\)") [ss_strand_psi_include](/index.php?title=Ss_strand_psi_include&action=edit&redlink=1 "Ss strand psi include \(page does not exist\)") [ss_strand_psi_target](/index.php?title=Ss_strand_psi_target&action=edit&redlink=1 "Ss strand psi target \(page does not exist\)") [state](/index.php/State "State") [static_singletons](/index.php/Static_singletons "Static singletons") [stereo](/index.php/Stereo "Stereo") [stereo_angle](/index.php/Stereo_angle "Stereo angle") [stereo_double_pump_mono](/index.php?title=Stereo_double_pump_mono&action=edit&redlink=1 "Stereo double pump mono \(page does not exist\)") [stereo_mode](/index.php/Stereo_mode "Stereo mode") [stereo_shift](/index.php?title=Stereo_shift&action=edit&redlink=1 "Stereo shift \(page does not exist\)") [stick_ball](/index.php/Stick_ball "Stick ball") [stick_ball_ratio](/index.php/Stick_ball_ratio "Stick ball ratio") [stick_color](/index.php/Stick_color "Stick color") [stick_fixed_radius](/index.php?title=Stick_fixed_radius&action=edit&redlink=1 "Stick fixed radius \(page does not exist\)") [stick_nub](/index.php/Stick_nub "Stick nub") [stick_overlap](/index.php?title=Stick_overlap&action=edit&redlink=1 "Stick overlap \(page does not exist\)") [stick_quality](/index.php?title=Stick_quality&action=edit&redlink=1 "Stick quality \(page does not exist\)") [stick_radius](/index.php/Stick_radius "Stick radius") [stick_transparency](/index.php/Stick_transparency "Stick transparency") [stick_valence_scale](/index.php?title=Stick_valence_scale&action=edit&redlink=1 "Stick valence scale \(page does not exist\)") [stop_on_exceptions](/index.php?title=Stop_on_exceptions&action=edit&redlink=1 "Stop on exceptions \(page does not exist\)") [suppress_hidden](/index.php?title=Suppress_hidden&action=edit&redlink=1 "Suppress hidden \(page does not exist\)") [surface_best](/index.php?title=Surface_best&action=edit&redlink=1 "Surface best \(page does not exist\)") [surface_carve_cutoff](/index.php/Surface_carve_cutoff "Surface carve cutoff") [surface_carve_normal_cutoff](/index.php/Surface_carve_normal_cutoff "Surface carve normal cutoff") [surface_carve_selection](/index.php/Surface_carve_selection "Surface carve selection") [surface_carve_state](/index.php?title=Surface_carve_state&action=edit&redlink=1 "Surface carve state \(page does not exist\)") [surface_cavity_cutoff](/index.php?title=Surface_cavity_cutoff&action=edit&redlink=1 "Surface cavity cutoff \(page does not exist\)") [surface_cavity_mode](/index.php/Surface_cavity_mode "Surface cavity mode") [surface_cavity_radius](/index.php?title=Surface_cavity_radius&action=edit&redlink=1 "Surface cavity radius \(page does not exist\)") [surface_circumscribe](/index.php?title=Surface_circumscribe&action=edit&redlink=1 "Surface circumscribe \(page does not exist\)") [surface_clear_cutoff](/index.php?title=Surface_clear_cutoff&action=edit&redlink=1 "Surface clear cutoff \(page does not exist\)") [surface_clear_selection](/index.php?title=Surface_clear_selection&action=edit&redlink=1 "Surface clear selection \(page does not exist\)") [surface_clear_state](/index.php?title=Surface_clear_state&action=edit&redlink=1 "Surface clear state \(page does not exist\)") [surface_color](/index.php/Surface_color "Surface color") [surface_debug](/index.php?title=Surface_debug&action=edit&redlink=1 "Surface debug \(page does not exist\)") [surface_miserable](/index.php?title=Surface_miserable&action=edit&redlink=1 "Surface miserable \(page does not exist\)") [surface_mode](/index.php/Surface_mode "Surface mode") [surface_negative_color](/index.php/Surface_negative_color "Surface negative color") [surface_negative_visible](/index.php/Surface_negative_visible "Surface negative visible") [surface_normal](/index.php?title=Surface_normal&action=edit&redlink=1 "Surface normal \(page does not exist\)") [surface_optimize_subsets](/index.php?title=Surface_optimize_subsets&action=edit&redlink=1 "Surface optimize subsets \(page does not exist\)") [surface_poor](/index.php?title=Surface_poor&action=edit&redlink=1 "Surface poor \(page does not exist\)") [surface_proximity](/index.php/Surface_proximity "Surface proximity") [surface_quality](/index.php/Surface_quality "Surface quality") [surface_ramp_above_mode](/index.php/Surface_ramp_above_mode "Surface ramp above mode") [surface_solvent](/index.php/Surface_solvent "Surface solvent") [surface_trim_cutoff](/index.php?title=Surface_trim_cutoff&action=edit&redlink=1 "Surface trim cutoff \(page does not exist\)") [surface_trim_factor](/index.php?title=Surface_trim_factor&action=edit&redlink=1 "Surface trim factor \(page does not exist\)") [surface_type](/index.php/Surface_type "Surface type") [suspend_updates](/index.php/Suspend_updates "Suspend updates") [swap_dsn6_bytes](/index.php?title=Swap_dsn6_bytes&action=edit&redlink=1 "Swap dsn6 bytes \(page does not exist\)") [sweep_angle](/index.php?title=Sweep_angle&action=edit&redlink=1 "Sweep angle \(page does not exist\)") [sweep_mode](/index.php?title=Sweep_mode&action=edit&redlink=1 "Sweep mode \(page does not exist\)") [sweep_phase](/index.php?title=Sweep_phase&action=edit&redlink=1 "Sweep phase \(page does not exist\)") [sweep_speed](/index.php?title=Sweep_speed&action=edit&redlink=1 "Sweep speed \(page does not exist\)")

### T

[test1](/index.php?title=Test1&action=edit&redlink=1 "Test1 \(page does not exist\)") [test2](/index.php?title=Test2&action=edit&redlink=1 "Test2 \(page does not exist\)") [text](/index.php?title=Text&action=edit&redlink=1 "Text \(page does not exist\)") [texture_fonts](/index.php/Texture_fonts "Texture fonts") [trace_atoms_mode](/index.php?title=Trace_atoms_mode&action=edit&redlink=1 "Trace atoms mode \(page does not exist\)") [transparency](/index.php/Transparency "Transparency") [transparency_global_sort](/index.php?title=Transparency_global_sort&action=edit&redlink=1 "Transparency global sort \(page does not exist\)") [transparency_mode](/index.php/Transparency_mode "Transparency mode") [transparency_picking_mode](/index.php?title=Transparency_picking_mode&action=edit&redlink=1 "Transparency picking mode \(page does not exist\)") [triangle_max_passes](/index.php?title=Triangle_max_passes&action=edit&redlink=1 "Triangle max passes \(page does not exist\)") [trim_dots](/index.php?title=Trim_dots&action=edit&redlink=1 "Trim dots \(page does not exist\)") [two_sided_lighting](/index.php/Two_sided_lighting "Two sided lighting")

### U

[use_display_lists](/index.php?title=Use_display_lists&action=edit&redlink=1 "Use display lists \(page does not exist\)")

### V

[valence](/index.php/Valence "Valence") [valence_default](/index.php?title=Valence_default&action=edit&redlink=1 "Valence default \(page does not exist\)") [valence_mode](/index.php/Valence "Valence") [validate_object_names](/index.php?title=Validate_object_names&action=edit&redlink=1 "Validate object names \(page does not exist\)") [valence_size](/index.php/Valence "Valence") [virtual_trackball](/index.php/Virtual_trackball "Virtual trackball")

### W

[wildcard](/index.php?title=Wildcard&action=edit&redlink=1 "Wildcard \(page does not exist\)") [wizard_prompt_mode](/index.php?title=Wizard_prompt_mode&action=edit&redlink=1 "Wizard prompt mode \(page does not exist\)") [wrap_output](/index.php/Wrap_output "Wrap output")

### X

### Y

### Z

Retrieved from "[https://pymolwiki.org/index.php?title=Settings&oldid=11604](https://pymolwiki.org/index.php?title=Settings&oldid=11604)"


---

## All states

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Example
  * 4 User Notes
  * 5 See Also



## Overview

When set "on", this setting causes PyMOL to display all states or in NMR jargon: all the models in the ensemble. The 'default' behavior (OFF) can be overridden by placing the "set all_states, on" statement into your '.pymolrc' file, located in your login directory (under all flavors of unix). 

## Syntax
    
    
    set all_states, on      
    set all_states, off
    

## Example
    
    
    # fetch a PDB and show it in multiple states; this one
    # line does the work of the next 6 lines.
    fetch 1nmr
    
    # this is older code, use the above code which is newer
    import urllib2
    pdbCode = '1BRV'
    pdbUrl = 'http://www.rcsb.org/pdb/downloadFile.do?fileFormat=pdb&compression=NO&structureId='+pdbCode
    pdbFile = urllib2.urlopen(pdbUrl)
    pdbContent = pdbFile.read()
    cmd.read_pdbstr(pdbContent, pdbCode)
    
    set all_states, on
    

This shows the effect of turning on/off the **all_states** setting used with the script above. 

  * [![set all_states, on](/images/7/76/All_states_on.png)](/index.php/File:All_states_on.png "set all_states, on")

set all_states, on 

  * [![set all_states, off](/images/6/6f/All_states_off.png)](/index.php/File:All_states_off.png "set all_states, off")

set all_states, off 




## User Notes

  * There was an error with importing ensembles of ensembles files (complex MOL2 files) before revision 3541. If you experience this problem, update to 3541 or later.



## See Also

[Split_States](/index.php/Split_States "Split States")

Retrieved from "[https://pymolwiki.org/index.php?title=All_states&oldid=5749](https://pymolwiki.org/index.php?title=All_states&oldid=5749)"


---

## Ambient

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Setting Ambient controls the amount of ambient light. 

  


## Syntax
    
    
    # default 0.1
    set ambient, <number>
    

## Example

  * [![Ambient being set from 0 to 1.](/images/2/2e/AmbientScale.gif)](/index.php/File:AmbientScale.gif "Ambient being set from 0 to 1.")

Ambient being set from 0 to 1. 

  * [![ambient -0.4](/images/5/5f/Ambient_minus0.4.png)](/index.php/File:Ambient_minus0.4.png "ambient -0.4")

ambient -0.4 

  * [![ambient 0.1 \(default\)](/images/8/87/Ambient_default.png)](/index.php/File:Ambient_default.png "ambient 0.1 \(default\)")

ambient 0.1 (default) 

  * [![ambient 0.6](/images/a/ad/Ambient_0.6.png)](/index.php/File:Ambient_0.6.png "ambient 0.6")

ambient 0.6 




Retrieved from "[https://pymolwiki.org/index.php?title=Ambient&oldid=6280](https://pymolwiki.org/index.php?title=Ambient&oldid=6280)"


---

## Antialias

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The **antialias** setting in pymol controls the level of antialiasing that pymol does in the **[ray](/index.php/Ray "Ray")** -tracing routine. Higher numbers take longer to ray trace but provide much smoother and better looking images. 

## Examples

Notice the overall **quality** of the images improves as the antialiasing setting is increased. Click each image to see the high-resolution version; the small image doesn't show the differences in AA settings so well. 

  * [![AA set to 0](/images/e/e4/Aa0.png)](/index.php/File:Aa0.png "AA set to 0")

AA set to 0 

  * [![AA set to 1](/images/3/3a/Aa1.png)](/index.php/File:Aa1.png "AA set to 1")

AA set to 1 

  * [![AA set to 2](/images/e/e0/Aa2.png)](/index.php/File:Aa2.png "AA set to 2")

AA set to 2 




## Syntax
    
    
    set antialias,0  # low setting/off
    set antialias,2  # higher setting, better image quality
    

Retrieved from "[https://pymolwiki.org/index.php?title=Antialias&oldid=4275](https://pymolwiki.org/index.php?title=Antialias&oldid=4275)"


---

## Assembly

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When loading **mmCIF** structure files, the assembly setting controls whether PyMOL loads the asymmetric unit (assembly="") or a biological unit (e.g. assembly="1"). Assembly identifiers in mmCIF files can be arbitrary strings, but usually are numeric and most files define at least assembly "1". 

_New in PyMOL 1.8_

Assemblies with more than one symmetry copy are loaded as multi-state objects with [all_states](/index.php/All_states "All states")=on. The assemblies should be equivalent to the **pdb1** files which are available from the [PDB ftp server](ftp://ftp.wwpdb.org/pub/pdb/data/biounit/). 

## Examples

Download the ASU and the first assembly, show them side by side. 
    
    
    set assembly, ""
    fetch 3bw1, asu, async=0
    
    set assembly, 1
    fetch 3bw1, assembly1, async=0
    
    set grid_mode
    

## See Also

  * [all_states](/index.php/All_states "All states")
  * [split_states](/index.php/Split_states "Split states")
  * <http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx.dic/Items/_pdbx_struct_assembly.id.html>
  * [BiologicalUnit/Quat](/index.php/BiologicalUnit/Quat "BiologicalUnit/Quat") and [BiologicalUnit](/index.php/BiologicalUnit "BiologicalUnit"): scripts to parse assembly information from PDB files



Retrieved from "[https://pymolwiki.org/index.php?title=Assembly&oldid=12431](https://pymolwiki.org/index.php?title=Assembly&oldid=12431)"


---

## Atom name wildcard

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## OVERVIEW

This setting controls whether or not PyMOL will respect wildcards ("*"). PyMOL automatically disables the asterisk wildcard for atom names in PDB structures read which contain asterisks in the atom name field (as many NA structures do). 

## USAGE

The following code will restore wildcard usage in PyMOL, if it has been turned off. 
    
    
    # allow wildcards
    unset atom_name_wildcard, object-name
    

**Please note** , however, that PyMOL will then be unable to distinguish **C2** in the base from **C2*** in the sugar. 

If, before running the previous command, you instead issue the following command: 
    
    
    # change all '*'s in names to "'"
    alter all, name=name.replace("*","'")
    

you will be able to distinguish C2 from C2' and still have your atom name wildcards! 

Retrieved from "[https://pymolwiki.org/index.php?title=Atom_name_wildcard&oldid=12682](https://pymolwiki.org/index.php?title=Atom_name_wildcard&oldid=12682)"


---

## Auto show classified

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When loading structure files, the auto_show_classified setting sets up representations based on molecule classification (depends on [auto_classify_atoms](/index.php?title=Auto_classify_atoms&action=edit&redlink=1 "Auto classify atoms \(page does not exist\)")). 

Available from the Tcl/Tk menu: **Setting > Auto Show ... > ...**

_New in PyMOL 1.8.2_

## Example
    
    
    set auto_show_classified
    fetch 1rx1
    

## Values

  * -1: for less than 500k atoms: like 1, else like 3 (default, new in PyMOL 2.0)
  * 0: off (default in PyMOL 1.8)
  * 1: show [polymer](/index.php/Single-word_Selectors "Single-word Selectors") as [cartoon](/index.php/Cartoon "Cartoon"), [organic](/index.php/Single-word_Selectors "Single-word Selectors") as [sticks](/index.php/Sticks "Sticks"), and [inorganic](/index.php/Single-word_Selectors "Single-word Selectors") as [spheres](/index.php/Spheres "Spheres")
  * 2: like 1, but apply [auto_show_lines](/index.php?title=Auto_show_lines&action=edit&redlink=1 "Auto show lines \(page does not exist\)") and [auto_show_nonbonded](/index.php?title=Auto_show_nonbonded&action=edit&redlink=1 "Auto show nonbonded \(page does not exist\)") in addition
  * 3: simplified version, show polymer as [ribbon](/index.php/Ribbon "Ribbon"), organic and inorganic as [lines](/index.php/Lines "Lines") or [nonbonded](/index.php?title=Nonbonded&action=edit&redlink=1 "Nonbonded \(page does not exist\)")



## See Also

  * [show](/index.php/Show "Show")
  * [hide](/index.php/Hide "Hide")
  * [auto_show_lines](/index.php?title=Auto_show_lines&action=edit&redlink=1 "Auto show lines \(page does not exist\)")
  * [auto_show_nonbonded](/index.php?title=Auto_show_nonbonded&action=edit&redlink=1 "Auto show nonbonded \(page does not exist\)")
  * [auto_classify_atoms](/index.php?title=Auto_classify_atoms&action=edit&redlink=1 "Auto classify atoms \(page does not exist\)")



Retrieved from "[https://pymolwiki.org/index.php?title=Auto_show_classified&oldid=12697](https://pymolwiki.org/index.php?title=Auto_show_classified&oldid=12697)"


---

## Auto zoom

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

When set "on", this setting causes PyMOL to "auto_zoom" to any new object upon loading. This is helpful when one wishes to look at a new object immediately upon loading it into your PyMOL session; it can also be vexing in the situation where you have a carefully constructed view that might be "lost" if you do not anticipate that the program will change the view on loading an additional object. If you opt to use "auto_zoom on", it is also wise to get in the habit of frequent session saves and also use of "scenes" and the "get_view" utility, which will save one's current view orientation matrix to the (temporary) memory cache. The 'default' behavior (ON) can be overridden by placing the "set auto_zoom, off" statement into your '.pymolrc' file, located in your login directory (under all flavors of unix). 

## Syntax

set auto_zoom, on 

set auto_zoom, off 

Retrieved from "[https://pymolwiki.org/index.php?title=Auto_zoom&oldid=4269](https://pymolwiki.org/index.php?title=Auto_zoom&oldid=4269)"


---

## Backface cull

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

**backface_cull** controls whether or not backward facing triangles are not filtered out when ray tracing. 

  


## Syntax
    
    
    # on; default
    set backface_cull, 1
    
    # off; allows the visualization of the "other side" (inside) of a surface
    set backface_cull, 0
    

**Note** : To allow or disallow visualization of the inside of a surface, see [two_sided_lighting](/index.php/Two_sided_lighting "Two sided lighting"). 

  


## Examples

  * [![cull_backface ON](/images/d/d1/Cull_backface_on.jpg)](/index.php/File:Cull_backface_on.jpg "cull_backface ON")

cull_backface ON 

  * [![cull_backface OFF](/images/c/c3/Cull_backface_off.jpg)](/index.php/File:Cull_backface_off.jpg "cull_backface OFF")

cull_backface OFF 




Retrieved from "[https://pymolwiki.org/index.php?title=Backface_cull&oldid=4268](https://pymolwiki.org/index.php?title=Backface_cull&oldid=4268)"


---

## Bg rgb

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting is used to set the color of the background. Any color of the spectrum can be rendered. White backgrounds are often desirable for publication images. 

## Syntax

set bg_rgb,[_float1_ ,_float2_ ,_float3_] 

each float must be between 0.0 and 1.0 

float1: red component 

float2: green component 

float3: blue component 

  


example: set bg_rgb,[1,1,1] 

will color the background in white. 

## See Also

  * [Bg Color](/index.php/Bg_Color "Bg Color"), [bg_gradient](/index.php/Bg_gradient "Bg gradient")



Retrieved from "[https://pymolwiki.org/index.php?title=Bg_rgb&oldid=9542](https://pymolwiki.org/index.php?title=Bg_rgb&oldid=9542)"


---

## Bg rgb bottom

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

This value, when [bg_gradient](/index.php/Bg_gradient "Bg gradient") is set, provides the bottom color for the gradient. 

## See Also

[Bg_gradient](/index.php/Bg_gradient "Bg gradient") [bg_rgb_top](/index.php/Bg_rgb_top "Bg rgb top")

Retrieved from "[https://pymolwiki.org/index.php?title=Bg_rgb_bottom&oldid=9019](https://pymolwiki.org/index.php?title=Bg_rgb_bottom&oldid=9019)"


---

## Bg rgb top

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

This value, when [bg_gradient](/index.php/Bg_gradient "Bg gradient") is set, provides the top color for the gradient. 

## See Also

[Bg_gradient](/index.php/Bg_gradient "Bg gradient") [bg_rgb_bottom](/index.php/Bg_rgb_bottom "Bg rgb bottom")

Retrieved from "[https://pymolwiki.org/index.php?title=Bg_rgb_top&oldid=9018](https://pymolwiki.org/index.php?title=Bg_rgb_top&oldid=9018)"


---

## Button mode name

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Button_mode_name is the textual description of the current state of the mouse mode matrix. 

Current options are: 

  * 3-Button Maestro
  * 3-Button Viewing
  * 3-Button Editing
  * 3-Button Motions
  * 3-Button All Modes



Retrieved from "[https://pymolwiki.org/index.php?title=Button_mode_name&oldid=9023](https://pymolwiki.org/index.php?title=Button_mode_name&oldid=9023)"


---

## Cartoon all alt

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The cartoon_all_alt setting activates [cartoon](/index.php/Cartoon "Cartoon") for alternative coordinates. There will be one cartoon geometry per alt-code. The default (**off**) is to only render a single cartoon where only the first alt code is considered. 

_New in PyMOL 1.8.4_

## Example
    
    
    fetch 5a0d, async=0
    spectrum alt
    as cartoon
    
    set cartoon_all_alt
    

[![Cartoon all alt.png](/images/0/04/Cartoon_all_alt.png)](/index.php/File:Cartoon_all_alt.png)

## See Also

  * [cartoon](/index.php/Cartoon "Cartoon")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_all_alt&oldid=12705](https://pymolwiki.org/index.php?title=Cartoon_all_alt&oldid=12705)"


---

## Cartoon color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Related settings
  * 4 See Also



## Overview

set **cartoon_color** allows one to color cartoons independently from the rest of the representations. 

## Syntax
    
    
    set cartoon_color, theColor
    

where _theColor_ can be : 

  * any usual colors (blue, yellow, grey50,...)
  * number-coded colors (1:black, 2:blue, 3:greenish, ...)
  * special code -1 to revert to original chameleon setting (_set cartoon_color,-1_)



## Related settings

[sphere_color](/index.php/Sphere_color "Sphere color")

## See Also

[Color](/index.php/Color "Color")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_color&oldid=5516](https://pymolwiki.org/index.php?title=Cartoon_color&oldid=5516)"


---

## Cartoon cylindrical helices

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The cartoon_cylindrical_helices setting makes [cartoon](/index.php/Cartoon "Cartoon") helices appear as simple cylinders. 

_Changed in PyMOL 2.5: Cylinders are not straight anymore but follow the actual curved shape of a helix. Straight helices are still available as mode 2._

## Contents

  * 1 Modes
  * 2 Example
  * 3 Related Settings
  * 4 See Also



## Modes

  * 0: Off (helices will be rendered according to the [cartoon](/index.php/Cartoon "Cartoon") property or settings like [cartoon_fancy_helices](/index.php/Cartoon_fancy_helices "Cartoon fancy helices"))
  * 1: Curved helices (since PyMOL 2.5)
  * 2: Straight helices



## Example
    
    
    fetch 1bb1
    set cartoon_cylindrical_helices, 1
    as cartoon
    

set cartoon_cylindrical_helices, 1  | set cartoon_cylindrical_helices, 2   
---|---  
[![Cartoon cylindrical helices-1.png](/images/f/f4/Cartoon_cylindrical_helices-1.png)](/index.php/File:Cartoon_cylindrical_helices-1.png) | [![Cartoon cylindrical helices-2.png](/images/2/24/Cartoon_cylindrical_helices-2.png)](/index.php/File:Cartoon_cylindrical_helices-2.png)  
  
## Related Settings

  * [cartoon_smooth_cylinder_window](/index.php?title=Cartoon_smooth_cylinder_window&action=edit&redlink=1 "Cartoon smooth cylinder window \(page does not exist\)"): half-window size for window average smoothing (mode 1)
  * [cartoon_smooth_cylinder_cycles](/index.php?title=Cartoon_smooth_cylinder_cycles&action=edit&redlink=1 "Cartoon smooth cylinder cycles \(page does not exist\)"): number of cycles for window average smoothing (mode 1)
  * [cartoon_loop_quality](/index.php?title=Cartoon_loop_quality&action=edit&redlink=1 "Cartoon loop quality \(page does not exist\)") affects "roundness" of mode 1 curved cylindrical helices (and loops of course).



## See Also

  * [Cartoon Helix Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")
  * [cartoon_smooth_loops](/index.php/Cartoon_smooth_loops "Cartoon smooth loops")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_cylindrical_helices&oldid=13327](https://pymolwiki.org/index.php?title=Cartoon_cylindrical_helices&oldid=13327)"


---

## Cartoon discrete colors

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting toggles whether adjacent colors on a cartoon share a discrete boundary or whether they blend into each other. 

## Settings
    
    
    set cartoon_discrete_colors, on   # sets color blending off
    set cartoon_discrete_colors, off  # allows the blending of adjacent colors
    

## Examples

Open the images to actually see the details! 

  * [![cartoon_discrete_colors ON](/images/a/a9/Cartoon_discrete_colors_on.png)](/index.php/File:Cartoon_discrete_colors_on.png "cartoon_discrete_colors ON")

cartoon_discrete_colors ON 

  * [![cartoon_discrete_colors OFF](/images/4/47/Cartoon_discrete_colors_off.png)](/index.php/File:Cartoon_discrete_colors_off.png "cartoon_discrete_colors OFF")

cartoon_discrete_colors OFF 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_discrete_colors&oldid=5251](https://pymolwiki.org/index.php?title=Cartoon_discrete_colors&oldid=5251)"


---

## Cartoon dumbbell length

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Only relavent when fancy helices is turned on, see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")  
set cartoon_dumbbell_length defines the length between the dumbbell ends when the fancy helices setting has been engaged 

## Syntax
    
    
    set cartoon_dumbbell_length, <number>        #default setting is 1.6
    

## Example

  * [![cartoon_dumbbell_length 0.7](/images/4/40/Cartoon_dumbbell_length_0.7.png)](/index.php/File:Cartoon_dumbbell_length_0.7.png "cartoon_dumbbell_length 0.7")

cartoon_dumbbell_length 0.7 

  * [![cartoon_dumbbell_length 1.6](/images/7/7e/Cartoon_dumbbell_length_1.6.png)](/index.php/File:Cartoon_dumbbell_length_1.6.png "cartoon_dumbbell_length 1.6")

cartoon_dumbbell_length 1.6 

  * [![cartoon_dumbbell_length 2.5](/images/d/dd/Cartoon_dumbbell_length_2.5.png)](/index.php/File:Cartoon_dumbbell_length_2.5.png "cartoon_dumbbell_length 2.5")

cartoon_dumbbell_length 2.5 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_dumbbell_length&oldid=4252](https://pymolwiki.org/index.php?title=Cartoon_dumbbell_length&oldid=4252)"


---

## Cartoon dumbbell radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Only relavent when fancy helices is turned on, see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")  
set cartoon_dumbbell_radius defines the radius of the dumbbell ends when the fancy helices setting has been engaged  
Setting to 0.1 eliminates the dumbbell shape with other settings to defaults, whereas setting <0.1 reveals interior space of the fancy helix 

## Syntax
    
    
    set cartoon_dumbbell_radius, <number>        #default setting is 0.15
    

Adjust this value or that of [cartoon_dumbbell_width](/index.php/Cartoon_dumbbell_width "Cartoon dumbbell width") to remove the bulge at the edges of a fancy helix 

## Example

  * [![cartoon_dumbbell_radius 0](/images/f/f9/Cartoon_dumbbell_radius_0.png)](/index.php/File:Cartoon_dumbbell_radius_0.png "cartoon_dumbbell_radius 0")

cartoon_dumbbell_radius 0 

  * [![cartoon_dumbbell_radius 0.1](/images/5/50/Cartoon_dumbbell_radius_0.1.png)](/index.php/File:Cartoon_dumbbell_radius_0.1.png "cartoon_dumbbell_radius 0.1")

cartoon_dumbbell_radius 0.1 

  * [![cartoon_dumbbell_radius 0.15 \(default\)](/images/7/7e/Cartoon_dumbbell_length_1.6.png)](/index.php/File:Cartoon_dumbbell_length_1.6.png "cartoon_dumbbell_radius 0.15 \(default\)")

cartoon_dumbbell_radius 0.15 (default) 

  * [![cartoon_dumbbell_radius 0.4](/images/6/6c/Cartoon_dumbbell_radius_0.4.png)](/index.php/File:Cartoon_dumbbell_radius_0.4.png "cartoon_dumbbell_radius 0.4")

cartoon_dumbbell_radius 0.4 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_dumbbell_radius&oldid=4253](https://pymolwiki.org/index.php?title=Cartoon_dumbbell_radius&oldid=4253)"


---

## Cartoon dumbbell width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Only relavent when fancy helices is turned on, see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")  
set cartoon_dumbbell_width defines the width of the helix when the fancy helices setting has been engaged  


## Syntax
    
    
    set cartoon_dumbbell_width, <number>        #default setting is 0.1
    

Adjust this value or that of [cartoon_dumbbell_radius](/index.php/Cartoon_dumbbell_radius "Cartoon dumbbell radius") to remove the bulge at the edges of a fancy helix 

## Example

  * [![cartoon_dumbbell_width 0 \(can cause problems, circled in red\)](/images/9/93/Cartoon_dumbbell_width_0.png)](/index.php/File:Cartoon_dumbbell_width_0.png "cartoon_dumbbell_width 0 \(can cause problems, circled in red\)")

cartoon_dumbbell_width 0 (can cause problems, circled in red) 

  * [![cartoon_dumbbell_width 0.1](/images/7/7e/Cartoon_dumbbell_length_1.6.png)](/index.php/File:Cartoon_dumbbell_length_1.6.png "cartoon_dumbbell_width 0.1")

cartoon_dumbbell_width 0.1 

  * [![cartoon_dumbbell_width 0.2](/images/2/29/Cartoon_dumbbell_width_0.2.png)](/index.php/File:Cartoon_dumbbell_width_0.2.png "cartoon_dumbbell_width 0.2")

cartoon_dumbbell_width 0.2 

  * [![cartoon_dumbbell_width 1](/images/c/cf/Cartoon_dumbbell_width_1.png)](/index.php/File:Cartoon_dumbbell_width_1.png "cartoon_dumbbell_width 1")

cartoon_dumbbell_width 1 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_dumbbell_width&oldid=4254](https://pymolwiki.org/index.php?title=Cartoon_dumbbell_width&oldid=4254)"


---

## Cartoon fancy sheets

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This is a display setting. 

This setting turns on and off the arrows at the end of sheets. The arrows point towards the C-terminus of the sheet. The default setting is on. 

## Syntax

Turn on: 
    
    
    set cartoon_fancy_sheets, 1
    

Turn off: 
    
    
    set cartoon_fancy_sheets, 0
    

## Result

[![](/images/1/1b/Fancy_sheet.JPG)](/index.php/File:Fancy_sheet.JPG)

Fancy Sheets

[![](/images/2/2f/Default_sheet.JPG)](/index.php/File:Default_sheet.JPG)

Plain Sheets

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_fancy_sheets&oldid=4256](https://pymolwiki.org/index.php?title=Cartoon_fancy_sheets&oldid=4256)"


---

## Cartoon flat cycles

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set **cartoon_flat_cycles** allows the flatness of the sheets to be defined. The larger the number, the flatter the sheets. 

## Syntax
    
    
    set cartoon_flat_cycles, <integer>          #default setting is 3
    

Or alteratively 
    
    
    set cartoon_flat_cycles, on                 #default is on
    set cartoon_flat_cycles, off
    

## Example

  * [![cartoon_flat_cycles 0 or OFF](/images/4/47/Cartoon_flat_cycles_0.png)](/index.php/File:Cartoon_flat_cycles_0.png "cartoon_flat_cycles 0 or OFF")

cartoon_flat_cycles 0 or OFF 

  * [![cartoon_flat_cycles 3 or ON](/images/c/ce/Cartoon_flat_cycles_3.png)](/index.php/File:Cartoon_flat_cycles_3.png "cartoon_flat_cycles 3 or ON")

cartoon_flat_cycles 3 or ON 

  * [![cartoon_flat_cycles 10](/images/9/90/Cartoon_flat_cycles_10.png)](/index.php/File:Cartoon_flat_cycles_10.png "cartoon_flat_cycles 10")

cartoon_flat_cycles 10 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_flat_cycles&oldid=4257](https://pymolwiki.org/index.php?title=Cartoon_flat_cycles&oldid=4257)"


---

## Cartoon flat sheets

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

**cartoon_flat_sheets** determines how smooth PyMol draws the beta sheets. Flat sheets make for a nice artistic image, however sometime there are problems making sticks appear attached to the sheets. If this problem happens, turn this option off. 

  * [![Cartoon Flat Sheets, ON](/images/f/fb/Cfs_on.png)](/index.php/File:Cfs_on.png "Cartoon Flat Sheets, ON")

Cartoon Flat Sheets, ON 

  * [![Cartoon Flat Sheets, OFF](/images/3/38/Cfs_off.png)](/index.php/File:Cfs_off.png "Cartoon Flat Sheets, OFF")

Cartoon Flat Sheets, OFF 




## See Also

[Cartoon_smooth_loops](/index.php/Cartoon_smooth_loops "Cartoon smooth loops")

Or maybe even better: [Cartoon_side_chain_helper](/index.php/Cartoon_side_chain_helper "Cartoon side chain helper")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_flat_sheets&oldid=7062](https://pymolwiki.org/index.php?title=Cartoon_flat_sheets&oldid=7062)"


---

## Cartoon gap cutoff

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

With cartoon_gap_cutoff > 0, if there are missing residues along the protein backbone (e.g. missing loops), PyMOL will create a dashed cartoon loop segment if the gap (in number of residues) is shorter than the cutoff. 

_New in PyMOL 1.8.2_

## Example
    
    
    set cartoon_gap_cutoff, 10
    
    fetch 2xwu, async=0
    as cartoon
    orient B//152-156/
    

[![Cartoon gap cutoff.png](/images/e/e6/Cartoon_gap_cutoff.png)](/index.php/File:Cartoon_gap_cutoff.png)

## Number of dashes

The number of dashes is directly affected by the [cartoon_sampling](/index.php/Cartoon_sampling "Cartoon sampling") setting: 
    
    
    set cartoon_sampling, 20
    

## See Also

  * [cartoon](/index.php/Cartoon "Cartoon") dash
  * [cartoon_sampling](/index.php/Cartoon_sampling "Cartoon sampling")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_gap_cutoff&oldid=12706](https://pymolwiki.org/index.php?title=Cartoon_gap_cutoff&oldid=12706)"


---

## Cartoon helix radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set cartoon_helix_radius allows the radius to be definied of helices shown in cylinder form (see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")) 

## Syntax
    
    
    set cartoon_helix_radius, <number>          #default radius is 2.0
    

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_helix_radius&oldid=4259](https://pymolwiki.org/index.php?title=Cartoon_helix_radius&oldid=4259)"


---

## Cartoon Helix Settings

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Result
  * 4 Change size of default cartoon helix
  * 5 See also



## Overview

This setting sets the style in which [cartoon](/index.php/Cartoon "Cartoon") helices are rendered. The default is a helix with an ellipsoid cross section. **Fancy** mode helices are ribbons with tubular edges à la [MolScript](/index.php?title=MolScript&action=edit&redlink=1 "MolScript \(page does not exist\)"). Cylindicial mode renders helices as solid cylinders. 

## Syntax

To turn on Fancy Helices: 
    
    
    set cartoon_fancy_helices, 1
    

To turn on Cylindrical Helices: 
    
    
    set cartoon_cylindrical_helices, 1
    

To return to default helices: 
    
    
    set cartoon_fancy_helices, 0
    set cartoon_cylindrical_helices, 0
    

## Result

  * [![Fancy helix](/images/e/ea/Fancy.JPG)](/index.php/File:Fancy.JPG "Fancy helix")

Fancy helix 

  * [![Cylindrical helix](/images/5/58/Cylindre.JPG)](/index.php/File:Cylindre.JPG "Cylindrical helix")

Cylindrical helix 

  * [![Default representation](/images/9/9a/Default.JPG)](/index.php/File:Default.JPG "Default representation")

Default representation 




## Change size of default cartoon helix

To change the size of the cartoon helices when not in the fancy helix mode you change the dimensions of the oval that represents the cross section of the default cartoon representation. 

[![Cartoon oval length.png](/images/1/17/Cartoon_oval_length.png)](/index.php/File:Cartoon_oval_length.png)
    
    
    set cartoon_oval_length , 0.8 # default is 1.20
    

and 
    
    
    set cartoon_oval_width , 0.2  # default is 0.25
    

## See also

[Cartoon_oval_width](/index.php/Cartoon_oval_width "Cartoon oval width"), [Cartoon_oval_length](/index.php/Cartoon_oval_length "Cartoon oval length"), [Cartoon_oval_quality](/index.php/Cartoon_oval_quality "Cartoon oval quality"), [Cartoon_helix_radius](/index.php/Cartoon_helix_radius "Cartoon helix radius"), [Cartoon_dumbbell_radius](/index.php/Cartoon_dumbbell_radius "Cartoon dumbbell radius"), [Cartoon_dumbbell_length](/index.php/Cartoon_dumbbell_length "Cartoon dumbbell length"), [Cartoon_dumbbell_width](/index.php/Cartoon_dumbbell_width "Cartoon dumbbell width"), [Cartoon_sampling](/index.php/Cartoon_sampling "Cartoon sampling"). 

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_Helix_Settings&oldid=12171](https://pymolwiki.org/index.php?title=Cartoon_Helix_Settings&oldid=12171)"


---

## Cartoon highlight color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting allows one to specify a contrasting color for the interior face of helices and the side faces of strands. 

## Settings
    
    
    set cartoon_highlight_color, grey50   # sets cartoon highlight color to middle-grey
    set cartoon_highlight_color, -1      # turns this feature off (default)
    

## Examples

Open the images to actually see the details! 

  * [![cartoon_highlight_colors ON](/images/a/a9/Cartoon_discrete_colors_on.png)](/index.php/File:Cartoon_discrete_colors_on.png "cartoon_highlight_colors ON")

cartoon_highlight_colors ON 

  * [![cartoon_highlight_color OFF](/images/2/24/Cartoon_highlight_color_off.png)](/index.php/File:Cartoon_highlight_color_off.png "cartoon_highlight_color OFF")

cartoon_highlight_color OFF 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_highlight_color&oldid=9551](https://pymolwiki.org/index.php?title=Cartoon_highlight_color&oldid=9551)"


---

## Cartoon nucleic acid color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



## Overview

This setting determines the color of the sugar bb, when in nucleic acid mode. 

## Syntax
    
    
    # set the sugar bb to red
    set cartoon_nucleic_acid_color, red
    

## Examples

  * [![Color Red](/images/6/69/Cna_color2.png)](/index.php/File:Cna_color2.png "Color Red")

Color Red 

  * [![Color Blue](/images/e/e1/Cna_color1.png)](/index.php/File:Cna_color1.png "Color Blue")

Color Blue 




## See Also

[Cartoon#Nucleic_Acid_Representation](/index.php/Cartoon#Nucleic_Acid_Representation "Cartoon")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_nucleic_acid_color&oldid=5252](https://pymolwiki.org/index.php?title=Cartoon_nucleic_acid_color&oldid=5252)"


---

## Cartoon oval length

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Only relevant when oval (default) helices is turned on, see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")  
set cartoon_oval_length defines the length between the edges of the cartoon ribbon  


## Syntax
    
    
    set cartoon_oval_length, <number>        #default setting is ~1.35
    

## Example

  * [![cartoon_oval_length 0.6](/images/a/af/Cartoon_oval_length_0.6.png)](/index.php/File:Cartoon_oval_length_0.6.png "cartoon_oval_length 0.6")

cartoon_oval_length 0.6 

  * [![cartoon_oval_length 1.35 \(~default\)](/images/5/54/Cartoon_oval_width_default.png)](/index.php/File:Cartoon_oval_width_default.png "cartoon_oval_length 1.35 \(~default\)")

cartoon_oval_length 1.35 (~default) 

  * [![cartoon_oval_length 2](/images/7/7c/Cartoon_oval_length_2.png)](/index.php/File:Cartoon_oval_length_2.png "cartoon_oval_length 2")

cartoon_oval_length 2 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_oval_length&oldid=7694](https://pymolwiki.org/index.php?title=Cartoon_oval_length&oldid=7694)"


---

## Cartoon oval quality

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Only relevent when oval (default) helices is turned on, see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings"). set **cartoon_oval_quality** defines the visual quality (the number of points defining the oval) of the helix. The default setting is generally optimal, increasing improves quality marginally, where as quality drops significantly if the value is decreased a lot (see the black edges below). The difference between 100 and greater values is negligible. 

## Syntax
    
    
    set cartoon_oval_quality, <number>        #default setting is 10
    

## Example

  * [![cartoon_oval_quality 3](/images/f/f5/Cartoon_oval_quality_3.png)](/index.php/File:Cartoon_oval_quality_3.png "cartoon_oval_quality 3")

cartoon_oval_quality 3 

  * [![cartoon_oval_quality 10 \(default\)](/images/5/54/Cartoon_oval_width_default.png)](/index.php/File:Cartoon_oval_width_default.png "cartoon_oval_quality 10 \(default\)")

cartoon_oval_quality 10 (default) 

  * [![cartoon_oval_quality 1000](/images/6/65/Cartoon_oval_quality_1000.png)](/index.php/File:Cartoon_oval_quality_1000.png "cartoon_oval_quality 1000")

cartoon_oval_quality 1000 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_oval_quality&oldid=4244](https://pymolwiki.org/index.php?title=Cartoon_oval_quality&oldid=4244)"


---

## Cartoon oval width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Only relevant when oval (default) helices is turned on, see [Cartoon_Helix_Settings](/index.php/Cartoon_Helix_Settings "Cartoon Helix Settings")  
set cartoon_oval_width defines the width of the helix  


## Syntax
    
    
    set cartoon_oval_width, <number>        #default setting is 0.25
    

## Example

  * [![cartoon_oval_width 0.01](/images/b/b3/Cartoon_oval_width_0.01.png)](/index.php/File:Cartoon_oval_width_0.01.png "cartoon_oval_width 0.01")

cartoon_oval_width 0.01 

  * [![cartoon_oval_width 0.2](/images/5/54/Cartoon_oval_width_default.png)](/index.php/File:Cartoon_oval_width_default.png "cartoon_oval_width 0.2")

cartoon_oval_width 0.2 

  * [![cartoon_oval_width 1](/images/3/35/Cartoon_oval_width_1.png)](/index.php/File:Cartoon_oval_width_1.png "cartoon_oval_width 1")

cartoon_oval_width 1 




For editing the "length" of the ribbon see [Cartoon_oval_length](/index.php/Cartoon_oval_length "Cartoon oval length")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_oval_width&oldid=4949](https://pymolwiki.org/index.php?title=Cartoon_oval_width&oldid=4949)"


---

## Cartoon putty radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

cartoon_putty_radius sets the base putty radius. 

# Usage
    
    
    # set a base radius of 1.0 (will be scaled based on B factors)
    set cartoon_putty_radius, 1.0
    

# See Also

  * [putty](/index.php/Cartoon#Sausage_Representation "Cartoon")
  * [Cartoon_putty_scale_max](/index.php/Cartoon_putty_scale_max "Cartoon putty scale max")
  * [cartoon_putty_transform](/index.php/Cartoon_putty_transform "Cartoon putty transform")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_putty_radius&oldid=5733](https://pymolwiki.org/index.php?title=Cartoon_putty_radius&oldid=5733)"


---

## Cartoon putty scale max

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

cartoon_putty_scale_max sets the max scale that PyMOL will use for rendering cartoon putty. 

# Usage
    
    
    # set a max scale
    set cartoon_putty_scale_max, 3
    
    # no limits on scaling
    set cartoon_putty_scale_max, -1
    

  


# See Also

  * [putty](/index.php/Cartoon#Sausage_Representation "Cartoon")
  * [Cartoon_putty_scale_min](/index.php/Cartoon_putty_scale_min "Cartoon putty scale min")
  * [cartoon_putty_transform](/index.php/Cartoon_putty_transform "Cartoon putty transform")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_putty_scale_max&oldid=8241](https://pymolwiki.org/index.php?title=Cartoon_putty_scale_max&oldid=8241)"


---

## Cartoon putty scale min

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

cartoon_putty_scale_min sets the minimum scale that PyMOL will use for rendering cartoon putty. 

# Usage
    
    
    # set a minimum scale
    set cartoon_putty_scale_min, 3
    
    # no limits on scaling
    set cartoon_putty_scale_min, -1
    

  


# See Also

  * [putty](/index.php/Cartoon#Sausage_Representation "Cartoon")
  * [Cartoon_putty_scale_max](/index.php/Cartoon_putty_scale_max "Cartoon putty scale max")
  * [cartoon_putty_transform](/index.php/Cartoon_putty_transform "Cartoon putty transform")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_putty_scale_min&oldid=8240](https://pymolwiki.org/index.php?title=Cartoon_putty_scale_min&oldid=8240)"


---

## Cartoon putty transform

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting determines how PyMOL renders the transformation of original values into putty related settings. 

  


# Usage
    
    
    # normalized nonlinear scaling
    set cartoon_putty_transform, 0
    # relative nonlinear scaling
    set cartoon_putty_transform, 1
    # scaled nonlinear scaling
    set cartoon_putty_transform, 2
    # absolute nonlinear scaling
    set cartoon_putty_transform, 3
    
    # normalized linear scaling
    set cartoon_putty_transform, 4
    # relative linear scaling
    set cartoon_putty_transform, 5
    # scaled linear scaling
    set cartoon_putty_transform, 6
    # absolute linear scaling from the B factor
    set cartoon_putty_transform, 7
    
    # implied RMS scaling
    set cartoon_putty_transform, 8
    

  


# See Also

  * [putty](/index.php/Cartoon#Sausage_Representation "Cartoon")
  * [Cartoon_putty_scale_max](/index.php/Cartoon_putty_scale_max "Cartoon putty scale max")
  * [Cartoon_putty_scale_min](/index.php/Cartoon_putty_scale_min "Cartoon putty scale min")



Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_putty_transform&oldid=8239](https://pymolwiki.org/index.php?title=Cartoon_putty_transform&oldid=8239)"


---

## Cartoon rect length

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set cartoon_rect_length adjusts how broad the cartoon representation of sheets appear 

## Syntax
    
    
    set cartoon_rect_length, <number>                 #default 1.5
    

## Example

  * [![cartoon_rect_length 0.5](/images/e/ef/Cartoon_rect_length_0.5.png)](/index.php/File:Cartoon_rect_length_0.5.png "cartoon_rect_length 0.5")

cartoon_rect_length 0.5 

  * [![cartoon_rect_length 1.5 \(default\)](/images/8/88/Cartoon_rect_width_default.png)](/index.php/File:Cartoon_rect_width_default.png "cartoon_rect_length 1.5 \(default\)")

cartoon_rect_length 1.5 (default) 

  * [![cartoon_rect_length 2.0](/images/c/cb/Cartoon_rect_length_2.0.png)](/index.php/File:Cartoon_rect_length_2.0.png "cartoon_rect_length 2.0")

cartoon_rect_length 2.0 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_rect_length&oldid=4296](https://pymolwiki.org/index.php?title=Cartoon_rect_length&oldid=4296)"


---

## Cartoon rect width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set cartoon_rect_width adjusts thickness of the cartoon representation of sheets 

## Syntax
    
    
    set cartoon_rect_width, <number>                 #default 0.4
    

## Example

  * [![cartoon_rect_width 0.1](/images/a/a6/Cartoon_rect_width_0.1.png)](/index.php/File:Cartoon_rect_width_0.1.png "cartoon_rect_width 0.1")

cartoon_rect_width 0.1 

  * [![cartoon_rect_width 0.4 \(default\)](/images/8/88/Cartoon_rect_width_default.png)](/index.php/File:Cartoon_rect_width_default.png "cartoon_rect_width 0.4 \(default\)")

cartoon_rect_width 0.4 (default) 

  * [![cartoon_rect_width 1.0](/images/8/80/Cartoon_rect_width_1.0.png)](/index.php/File:Cartoon_rect_width_1.0.png "cartoon_rect_width 1.0")

cartoon_rect_width 1.0 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_rect_width&oldid=4295](https://pymolwiki.org/index.php?title=Cartoon_rect_width&oldid=4295)"


---

## Cartoon ring mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



## Overview

PyMOL provides a variety of representations for illustrating nucleic acid structures. This setting, **ring_mode** , affects how PyMol draws the bases in a nucleic acid representation. 

## Syntax
    
    
    # try other natural numbers
    set cartoon_ring_mode, 3
    

Currently, there are 7 modes, numbered **0** through **6**. See the illustrations below. 

## Examples

  * [![Set to 0](/images/0/06/Crm_0.png)](/index.php/File:Crm_0.png "Set to 0")

Set to 0 

  * [![Set to 1](/images/a/ac/Crm_1.png)](/index.php/File:Crm_1.png "Set to 1")

Set to 1 

  * [![Set to 2](/images/e/e0/Crm_2.png)](/index.php/File:Crm_2.png "Set to 2")

Set to 2 

  * [![Set to 3](/images/c/ce/Crm_3.png)](/index.php/File:Crm_3.png "Set to 3")

Set to 3 

  * [![Set to 4](/images/6/65/Crm_4.png)](/index.php/File:Crm_4.png "Set to 4")

Set to 4 

  * [![Set to 6](/images/4/47/Crm_6.png)](/index.php/File:Crm_6.png "Set to 6")

Set to 6 




## See Also

[Cartoon#Nucleic_Acid_Representation](/index.php/Cartoon#Nucleic_Acid_Representation "Cartoon")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_ring_mode&oldid=4294](https://pymolwiki.org/index.php?title=Cartoon_ring_mode&oldid=4294)"


---

## Cartoon ring transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



## Overview

Sets the level of transparency in nucleic acid representation. A value of 0.0 is completely opaque; 1.0 is completely transparent. 

  


## Syntax
    
    
    set cartoon_ring_transparency, 0.5  # valid values reals in the range of 0.0 - 1.0
    

## Examples

  * [![Set to 1.0](/images/2/24/Crt_05.png)](/index.php/File:Crt_05.png "Set to 1.0")

Set to 1.0 

  * [![Set to 0.5](/images/6/65/Crt_1.png)](/index.php/File:Crt_1.png "Set to 0.5")

Set to 0.5 




## See Also

[Nucleic Acid Representations](/index.php/Cartoon#Nucleic_Acid_Representation "Cartoon") and the [Nucleic Acids Category](/index.php/Category:Nucleic_Acids "Category:Nucleic Acids")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_ring_transparency&oldid=6296](https://pymolwiki.org/index.php?title=Cartoon_ring_transparency&oldid=6296)"


---

## Cartoon round helices

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting controls the "roundness" of cartoon helices. The default value is "on"; turning this setting off allows helices to pass more closely through C-alpha positions, however the resulting helices are often aesthetically less appealing. 

## Settings
    
    
    set cartoon_round_helices, on   # default
    set cartoon_round_helices, off  #
    

## Examples

Open the images to actually see the details! 

  * [![cartoon round helices ON \(default\)](/images/9/90/Cartoon_round_helices_on.png)](/index.php/File:Cartoon_round_helices_on.png "cartoon round helices ON \(default\)")

cartoon round helices ON (default) 

  * [![cartoon round helices OFF](/images/3/35/Cartoon_round_helices_off.png)](/index.php/File:Cartoon_round_helices_off.png "cartoon round helices OFF")

cartoon round helices OFF 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_round_helices&oldid=4292](https://pymolwiki.org/index.php?title=Cartoon_round_helices&oldid=4292)"


---

## Cartoon sampling

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting changes the number of segments that makes up a given length of cartoon. The default value is ten; larger values (more segments) provide a smoother curve while requiring greater amounts of memory and slowing [performance](/index.php/Category:Performance "Category:Performance"). 

## Settings
    
    
    set cartoon_sampling, 20   # increases the number of cartoon segments
    

## Examples

Open the images to actually see the details! 

  * [![cartoon sampling 5](/images/1/1c/Cartoon_05.png)](/index.php/File:Cartoon_05.png "cartoon sampling 5")

cartoon sampling 5 

  * [![cartoon sampling 10 \(default\)](/images/b/b0/Cartoon_10.png)](/index.php/File:Cartoon_10.png "cartoon sampling 10 \(default\)")

cartoon sampling 10 (default) 

  * [![cartoon sampling 20](/images/4/48/Cartoon_20.png)](/index.php/File:Cartoon_20.png "cartoon sampling 20")

cartoon sampling 20 

  * [![cartoon sampling 30](/images/e/ec/Cartoon_30.png)](/index.php/File:Cartoon_30.png "cartoon sampling 30")

cartoon sampling 30 




Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_sampling&oldid=5503](https://pymolwiki.org/index.php?title=Cartoon_sampling&oldid=5503)"


---

## Cartoon side chain helper

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Example
  * 4 See Also



## Overview

set **cartoon_side_chain_helper** is an easy way, in cartoon mode, to only show the side chain of a residue. 

For residues represented in [cartoon](/index.php/Cartoon "Cartoon") (but not [ribbon](/index.php/Ribbon "Ribbon")) form, if you show the residue also in [stick](/index.php/Sticks "Sticks") or wireframe **cartoon_side_chain_helper** hides backbone atoms. Showing the sidechain in this method distorts the cartoon shape so it joins with the sidechain, effectively over-riding any [cartoon_flat_cycles](/index.php/Cartoon_flat_cycles "Cartoon flat cycles") setting for that residue. 

## Syntax
    
    
    set cartoon_side_chain_helper, off            
    set cartoon_side_chain_helper, 0              #off
    
    set cartoon_side_chain_helper, on             
    set cartoon_side_chain_helper, 1              #on
    

## Example

  * [![cartoon_side_chain_helper on](/images/b/be/Cartoon_side_chain_helper_on.png)](/index.php/File:Cartoon_side_chain_helper_on.png "cartoon_side_chain_helper on")

cartoon_side_chain_helper on 

  * [![cartoon_side_chain_helper off](/images/a/a6/Cartoon_side_chain_helper_off.png)](/index.php/File:Cartoon_side_chain_helper_off.png "cartoon_side_chain_helper off")

cartoon_side_chain_helper off 




# See Also

[Ribbon_side_chain_helper](/index.php/Ribbon_side_chain_helper "Ribbon side chain helper")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_side_chain_helper&oldid=8514](https://pymolwiki.org/index.php?title=Cartoon_side_chain_helper&oldid=8514)"


---

## Cartoon smooth loops

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
    * 1.1 Sticks and Smooth Loops
  * 2 Syntax
    * 2.1 Additional remarks



## Overview

This is a display setting. When set to off, the cartoon will follow a path that is splined between each alpha carbons. This sometimes yields representations that are pretty frizzy. Turning this setting on will make the cartoon follow a smoother (but less conservative) path so that the representation is easier on the eyes. Use with caution, since false conclusions can be drawn by solely examining a smoothed cartoon trace. It is always best (for non-representation purposes) to also have a line trace so that the backbone is precisely located. 

  * [![set cartoon_smooth_loops,0 : the cartoon passes through each alpha-carbon](/images/e/e5/Frizz.JPG)](/index.php/File:Frizz.JPG "set cartoon_smooth_loops,0 : the cartoon passes through each alpha-carbon")

set cartoon_smooth_loops,0 : the cartoon passes through each alpha-carbon 

  * [![set cartoon_smooth_loops,1 : the cartoon is straighter but less conservative](/images/f/f2/Smooth.JPG)](/index.php/File:Smooth.JPG "set cartoon_smooth_loops,1 : the cartoon is straighter but less conservative")

set cartoon_smooth_loops,1 : the cartoon is straighter but less conservative 




  


**NOTE:** If changing the setting that is set by a _preset_ command (like "Publication"), ie. 
    
    
    set cartoon_smooth_loops,0
    

does not seem to have any effect, then you may need to [unset](/index.php/Unset "Unset") the property on your object, ie 
    
    
    unset cartoon_smooth_loops,obj
    

. 

### Sticks and Smooth Loops

When **cartoon_smooth_loops** is turned **on** and sticks are shown for some residue selection, often times the sticks don't touch the backbone or cartooned elements. To fix this, turn this setting **off** or enable **cartoon_side_chain_helper**. See the images below for an example. 

  * [![Cartoon Smooth Loops ON](/images/e/eb/Csl_on.png)](/index.php/File:Csl_on.png "Cartoon Smooth Loops ON")

Cartoon Smooth Loops ON 

  * [![Cartoon Smooth Loops OFF](/images/e/ec/Csl_off.png)](/index.php/File:Csl_off.png "Cartoon Smooth Loops OFF")

Cartoon Smooth Loops OFF 




  
Also, of note for this problem is [Cartoon_flat_sheets](/index.php/Cartoon_flat_sheets "Cartoon flat sheets"). 

## Syntax

turn on: 
    
    
    set cartoon_smooth_loops, 1
    

turn off: 
    
    
    set cartoon_smooth_loops, 0
    

### Additional remarks

I do not know precisely how the smoothed trace is generated, probably by some splining mode. Could someone more knowledgeable than me edit to be more precise ? 

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_smooth_loops&oldid=4289](https://pymolwiki.org/index.php?title=Cartoon_smooth_loops&oldid=4289)"


---

## Cartoon transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



## Overview

This setting changes the transparency of the cartoon representation. 

As of PyMOL version [2.3](https://pymol.org/dokuwiki/?id=media:new23), cartoon transparency is now an atom-level setting. This allows for a cartoon selection to be transparent without creating an additional object. 

## Syntax
    
    
    set cartoon_transparency, 0.5, <object>  # 50% transparent, the object name is optional
    set cartoon_transparency, 0.5, <sele>  # 50% transparent, sele: the selection name
    

Valid values range from 0.0 (fully opaque) - 1.0 (fully transparent) 

## Examples

  * [![Cartoon Transparency](/images/7/76/Ctt.png)](/index.php/File:Ctt.png "Cartoon Transparency")

Cartoon Transparency 

  * [![Cartoon Atom-level Transparency](/images/4/40/1rmm-atom-cartoon-transparency.png)](/index.php/File:1rmm-atom-cartoon-transparency.png "Cartoon Atom-level Transparency")

Cartoon Atom-level Transparency 




## See Also

[cartoon](/index.php/Cartoon "Cartoon")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_transparency&oldid=12844](https://pymolwiki.org/index.php?title=Cartoon_transparency&oldid=12844)"


---

## Cavity cull

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set cavity_cull allows the user to define how sensitive pymol it to detecting cavities within the interior of a molecule. Smaller cavities are not detected as the cavity cull value is increased. The value _may_ correspond in angstroms a dimension of the cavity approximately. 

## Syntax
    
    
    # default 2
    set cavity_cull, <number>
    

## Example

  * [![cavity_cull 0](/images/3/3d/Cavity_0.jpg)](/index.php/File:Cavity_0.jpg "cavity_cull 0")

cavity_cull 0 

  * [![cavity_cull 2 \(default\)](/images/f/f9/Cavity_default.jpg)](/index.php/File:Cavity_default.jpg "cavity_cull 2 \(default\)")

cavity_cull 2 (default) 

  * [![cavity_cull 20](/images/6/69/Cavity_20.jpg)](/index.php/File:Cavity_20.jpg "cavity_cull 20")

cavity_cull 20 




Retrieved from "[https://pymolwiki.org/index.php?title=Cavity_cull&oldid=6317](https://pymolwiki.org/index.php?title=Cavity_cull&oldid=6317)"


---

## Cell color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Related settings
  * 4 See Also



## Overview

set **cell_color** allows one to color [cells](/index.php/Cell "Cell") independently from the rest of the representations. 

## Syntax
    
    
    set cell_color, theColor
    

where _theColor_ can be : 

  * any usual colors (blue, yellow, grey50,...)
  * number-coded colors (1:black, 2:blue, 3:greenish, ...)
  * special code -1 to revert to original chameleon setting (_set cartoon_color,-1_)



## Related settings

[sphere_color](/index.php/Sphere_color "Sphere color") [cartoon_color](/index.php/Cartoon_color "Cartoon color")

## See Also

[Color](/index.php/Color "Color")

Retrieved from "[https://pymolwiki.org/index.php?title=Cell_color&oldid=13573](https://pymolwiki.org/index.php?title=Cell_color&oldid=13573)"


---

## Cgo transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

This is an object-level setting to control transparency of Compiled Graphics Objects. 

Usage: 
    
    
    # PyMOL command
    set cgo_transparency, value [, object]
    
    # Python API
    cmd.set("cgo_transparency", value [, object])
    

Arguments 

  * value (float) number between 0 and 1 (with 0 meaning the object is opaque, and 1 == invisible).
  * object (str) name of the object to which to apply the setting (default: all CGO objects)



Retrieved from "[https://pymolwiki.org/index.php?title=Cgo_transparency&oldid=13483](https://pymolwiki.org/index.php?title=Cgo_transparency&oldid=13483)"


---

## Chem comp cartn use

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The chem_comp_cartn_use setting selects with coordinate columns to load from an mmCIF chemical component file. 

_New in PyMOL 2.1 and Open-Source revision 4184_

Value  | Column Labels   
---|---  
0  | Use first available (default)   
1  | _chem_comp_atom.pdbx_model_Cartn_{x,y,z}_ideal   
2  | _chem_comp_atom.model_Cartn_{x,y,z}   
4  | _chem_comp_atom.{x,y,z}   
  
## Example
    
    
    set chem_comp_cartn_use, 1
    load <http://files.rcsb.org/ligands/download/HEM.cif>, hem_ideal
    
    set chem_comp_cartn_use, 2
    load <http://files.rcsb.org/ligands/download/HEM.cif>, hem_model
    
    align hem_ideal, hem_model
    

## See Also

  * <http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx.dic/Items/_chem_comp_atom.pdbx_model_Cartn_x_ideal.html>
  * <http://mmcif.wwpdb.org/dictionaries/mmcif_std.dic/Items/_chem_comp_atom.model_Cartn_x.html>



Retrieved from "[https://pymolwiki.org/index.php?title=Chem_comp_cartn_use&oldid=12715](https://pymolwiki.org/index.php?title=Chem_comp_cartn_use&oldid=12715)"


---

## Chromadepth

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The chromadepth setting turns on color by depth, using a rainbow spectrum from red (near) to blue (far). This can produce a pseudo-stereo effect when using [ChromaDepth glasses](http://chromatek.com/). 

_New in Incentive PyMOL 1.7.6_

_New in Incentive PyMOL 1.8.6: Ray tracing support_

_In Open-Source PyMOL since 2.2.0_

## Example
    
    
    fetch 1rx1, async=0
    as surface
    show sticks, organic
    
    set chromadepth
    
    set_view (\
         0.683854997,    0.728400052,   -0.042148408,\
        -0.276858896,    0.312507719,    0.908674002,\
         0.675050080,   -0.609731972,    0.415373385,\
         0.000000000,    0.000000000,  -62.658931732,\
        29.408838272,   54.024459839,   15.061312675,\
        45.707679749,   79.610176086,  -20.000000000 )
    
    # these settings can improve chromadepth experience
    set depth_cue, 0
    set specular, 0
    

## Gallery

  * [![Chroma 3d.png](/images/8/89/Chroma_3d.png)](/index.php/File:Chroma_3d.png)

  * [![Chroma 3d2.png](/images/e/e5/Chroma_3d2.png)](/index.php/File:Chroma_3d2.png)

  * [![Chroma 3d3.png](/images/b/bd/Chroma_3d3.png)](/index.php/File:Chroma_3d3.png)

  * [![Chroma 3d4.png](/images/1/17/Chroma_3d4.png)](/index.php/File:Chroma_3d4.png)

  * [![Chroma 3d5.png](/images/a/a3/Chroma_3d5.png)](/index.php/File:Chroma_3d5.png)




## See Also

  * <http://en.wikipedia.org/wiki/ChromaDepth>
  * [stereo](/index.php/Stereo "Stereo")
  * [depth_cue](/index.php/Depth_cue "Depth cue")



Retrieved from "[https://pymolwiki.org/index.php?title=Chromadepth&oldid=12894](https://pymolwiki.org/index.php?title=Chromadepth&oldid=12894)"


---

## Cif keepinmemory

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The cif_keepinmemory setting is an experimental feature. When loading a CIF file with **cif_keepinmemory** =on, then the parsed file is kept in memory and made accessible like a dictionary from the Python API. 

_New in PyMOL 1.7.8_

## Examples
    
    
    import pymol
    
    pymol.cmd.set("cif_keepinmemory")
    pymol.cmd.fetch("1ubq", type="cif")
    
    citation_titles = pymol.querying.cif_get_array("1ubq", "_citation.title")
    
    import pprint
    pprint.pprint(citation_titles)
    

Output: 
    
    
    ['Structure of ubiquitin refined at 1.8 A resolution.',
     'Comparison of the Three-Dimensional Structures of Human, Yeast, and Oat Ubiquitin',
     'Three-Dimensional Structure of Ubiquitin at 2.8 Angstroms Resolution',
     'Crystallization and Preliminary X-Ray Investigation of Ubiquitin, a Non-Histone Chromosomal Protein',
     'Molecular Conservation of 74 Amino Acid Sequence of Ubiquitin between Cattle and Man']
    

Retrieved from "[https://pymolwiki.org/index.php?title=Cif_keepinmemory&oldid=12900](https://pymolwiki.org/index.php?title=Cif_keepinmemory&oldid=12900)"


---

## Cif use auth

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When loading **mmCIF** structure files, the cif_use_auth setting (default: on) controls whether the **auth_*** fields are used or not. 

_New in PyMOL 1.7.8_

With the [PDBx/mmCIF](http://mmcif.wwpdb.org/) format, the PDB introduced new residue and chain numbering schemes. The numbering which is found in PDB files is available in mmCIF files from the **auth_*** fields. The new numbering, which for example has no insertion codes and strictly increments residue numbers, is available from the **label_*** fields. For compatibility with PDB files, PyMOL by default will read the **auth_*** fields (if available, with fallback to **label_*** fields). 

Note that **_atom_site.label_asym_id** is always loaded into the **segi** field, independant of the **cif_use_auth** setting. 

Note that non-polymer molecules (e.g. water) are not enumerated by **_atom_site.label_seq_id** and will appear as a single residue in PyMOL (sequence viewer, when making selections, etc.) with **cif_use_auth** =off. 

## Currently affected fields

  * _atom_site.auth_atom_id (ID)
  * _atom_site.auth_comp_id (resn)
  * _atom_site.auth_seq_id (resi)
  * _atom_site.auth_asym_id (chain)
  * _atom_site.pdbx_pdb_ins_code (resi suffix)



## Examples

"auth" versus "label" asym_id of 1oky: 
    
    
    PyMOL>set cif_use_auth
    PyMOL>fetch 1oky, authon, async=0
    PyMOL>get_chains authon
     cmd.get_chains:  ['A']
    
    
    
    PyMOL>set cif_use_auth, off
    PyMOL>fetch 1oky, authoff, async=0
    PyMOL>get_chains authoff
     cmd.get_chains:  ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K']
    

Retrieved from "[https://pymolwiki.org/index.php?title=Cif_use_auth&oldid=12147](https://pymolwiki.org/index.php?title=Cif_use_auth&oldid=12147)"


---

## Connect cutoff

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

[![Example](/images/0/05/Connect_cutoff.png)](/index.php/File:Connect_cutoff.png "Example")

connect_cutoff is a global setting for distance-based bonding when loading a file. Default value is 0.35. 

Two atoms with VDW radii vdw1 and vdw2 are connected with a bond, if the following inequality is true: 
    
    
    distance <= connect_cutoff + (vdw1 + vdw2)/2
    

Special cutoff adjustments are made for hydrogens (connect_cutoff - 0.2) and disulfides (connect_cutoff + 0.2). 

## See Also

[load](/index.php/Load "Load"), [connect_mode](/index.php/Connect_mode "Connect mode"), [connect_bonded](/index.php?title=Connect_bonded&action=edit&redlink=1 "Connect bonded \(page does not exist\)")

Retrieved from "[https://pymolwiki.org/index.php?title=Connect_cutoff&oldid=10906](https://pymolwiki.org/index.php?title=Connect_cutoff&oldid=10906)"


---

## Connect mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

Sets how bonds are made when loading a file. 

Values: 

  * 0 = distance-based (excluding HETATM to HETATM) and CONECT records (default)
  * 1 = CONECT records
  * 2 = distance-based, ignores CONECT records
  * 3 = distance-based (including HETATM to HETATM) and CONECT records
  * 4 = for loading mmCIF: use the [chemical components dictionary](http://www.wwpdb.org/data/ccd) to look up bonds (_in PyMOL 1.7.4,**components.cif** needs to be present in the current directory, later versions have a subset of the dictionary bundled with PyMOL, and look up unknown residues from a web service_)



# Syntax
    
    
    # ignore CONECT records
    set connect_mode, 2
    # show current setting
    get connect_mode
    

# See Also

[Load](/index.php/Load "Load"), [Connect_cutoff](/index.php/Connect_cutoff "Connect cutoff"), [Connect_bonded](/index.php?title=Connect_bonded&action=edit&redlink=1 "Connect bonded \(page does not exist\)")

Retrieved from "[https://pymolwiki.org/index.php?title=Connect_mode&oldid=12747](https://pymolwiki.org/index.php?title=Connect_mode&oldid=12747)"


---

## Cylinder shader ff workaround

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

_New in PyMOL 1.5.0_

_Removed in PyMOL 1.7.6_

## OVERVIEW

When cylinders (as in stick representation) are drawn incorrectly, enabling this option might fix the problem. This option is often required in NVIDIA's video cards (GeForce and Quadro). 

  * [![Incorrect rendering in GeForce GTX 580](/images/e/e7/Shader_bug.png)](/index.php/File:Shader_bug.png "Incorrect rendering in GeForce GTX 580")

Incorrect rendering in GeForce GTX 580 

  * [![Setting this option 'on' fixed the problem](/images/5/5d/Shader_bug_fixed.png)](/index.php/File:Shader_bug_fixed.png "Setting this option 'on' fixed the problem")

Setting this option 'on' fixed the problem 




## USAGE
    
    
    set cylinder_shader_ff_workaround, on
    

The default is off. 

Once you found this option necessary, you can add this to [.pymolrc](/index.php/Pymolrc "Pymolrc"). 

Retrieved from "[https://pymolwiki.org/index.php?title=Cylinder_shader_ff_workaround&oldid=12483](https://pymolwiki.org/index.php?title=Cylinder_shader_ff_workaround&oldid=12483)"


---

## Dash Gap

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See also



## Overview

Sets the distance between dashes in labels. 

## Syntax
    
    
    set dash_gap, 0.1500
    

## Examples

  * [![Dash gap small -- dash_gap set to 0.1500](/images/c/cf/Dashes1000.png)](/index.php/File:Dashes1000.png "Dash gap small -- dash_gap set to 0.1500")

Dash gap small -- dash_gap set to 0.1500 

  * [![Dash gap normal -- dash_gap set to 0.3500](/images/8/88/Dashes3500.png)](/index.php/File:Dashes3500.png "Dash gap normal -- dash_gap set to 0.3500")

Dash gap normal -- dash_gap set to 0.3500 




## See also

[Dash Width](/index.php?title=Dash_Width&action=edit&redlink=1 "Dash Width \(page does not exist\)"), [Dash Radius](/index.php/Dash_Radius "Dash Radius"), [Dash Length](/index.php/Dash_Length "Dash Length"), [Dash Round Ends](/index.php/Dash_Round_Ends "Dash Round Ends")

Retrieved from "[https://pymolwiki.org/index.php?title=Dash_Gap&oldid=12032](https://pymolwiki.org/index.php?title=Dash_Gap&oldid=12032)"


---

## Dash Length

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See also



## Overview

Sets the length of dashes in labels. 

## Syntax
    
    
    set dash_length, 0.3500 # make longer dashes
    

## Examples

  * [![Dash Length set to 0.1500](/images/8/88/Dashes3500.png)](/index.php/File:Dashes3500.png "Dash Length set to 0.1500")

Dash Length set to 0.1500 

  * [![Dash Length set to 0.3500](/images/2/2f/Dl3500.png)](/index.php/File:Dl3500.png "Dash Length set to 0.3500")

Dash Length set to 0.3500 




## See also

[Dash Radius](/index.php/Dash_Radius "Dash Radius"), [Dash Width](/index.php?title=Dash_Width&action=edit&redlink=1 "Dash Width \(page does not exist\)"), [Dash Gap](/index.php/Dash_Gap "Dash Gap"), [Dash Round Ends](/index.php/Dash_Round_Ends "Dash Round Ends")

Retrieved from "[https://pymolwiki.org/index.php?title=Dash_Length&oldid=12026](https://pymolwiki.org/index.php?title=Dash_Length&oldid=12026)"


---

## Dash Radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See also



## Overview

Sets the width (or radius) of the dashes when ray-traced. 

## Syntax
    
    
    set dash_radius, 0.500   # fat dashes
    

## Examples

  * [![Normal size dashes](/images/2/2f/Dl3500.png)](/index.php/File:Dl3500.png "Normal size dashes")

Normal size dashes 

  * [![Fat dashes; dash_radius set to 0.100](/images/c/c7/Fat_dashes.png)](/index.php/File:Fat_dashes.png "Fat dashes; dash_radius set to 0.100")

Fat dashes; dash_radius set to 0.100 




## See also

[Dash Width](/index.php?title=Dash_Width&action=edit&redlink=1 "Dash Width \(page does not exist\)"), [Dash Gap](/index.php/Dash_Gap "Dash Gap"), [Dash Length](/index.php/Dash_Length "Dash Length"), [Dash Round Ends](/index.php/Dash_Round_Ends "Dash Round Ends")

Retrieved from "[https://pymolwiki.org/index.php?title=Dash_Radius&oldid=12031](https://pymolwiki.org/index.php?title=Dash_Radius&oldid=12031)"


---

## Dash Round Ends

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See also



## Overview

Sets if dashes are round or not at its ends. 

## Syntax
    
    
    set dash_round_ends, 0   # square ends
    

## Examples

  * [![Rounded dash-ends](/images/c/c7/Fat_dashes.png)](/index.php/File:Fat_dashes.png "Rounded dash-ends")

Rounded dash-ends 

  * [![Squared off dash-ends](/images/d/d0/Dash_square_ends.png)](/index.php/File:Dash_square_ends.png "Squared off dash-ends")

Squared off dash-ends 




## See also

[Dash Width](/index.php?title=Dash_Width&action=edit&redlink=1 "Dash Width \(page does not exist\)"), [Dash Radius](/index.php/Dash_Radius "Dash Radius"), [Dash Gap](/index.php/Dash_Gap "Dash Gap"), [Dash Length](/index.php/Dash_Length "Dash Length")

Retrieved from "[https://pymolwiki.org/index.php?title=Dash_Round_Ends&oldid=12034](https://pymolwiki.org/index.php?title=Dash_Round_Ends&oldid=12034)"


---

## Defer builds mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

The defer_builds_mode setting for improved [performance](/index.php/Category:Performance "Category:Performance") with long trajectories. This now makes it possible to work with files containing thousands of states, and to render impossibly long movies piecewise. This setting, as shown below, stops PyMOL caching the geometry of trajectory data in RAM. 

# Usage
    
    
    # improve PyMOL performance for many-state objects and long movies.
    set defer_builds_mode, 3
    

# See Also

  * [Load_Traj](/index.php/Load_Traj "Load Traj")
  * [async_builds](/index.php/Async_builds "Async builds")



Retrieved from "[https://pymolwiki.org/index.php?title=Defer_builds_mode&oldid=5801](https://pymolwiki.org/index.php?title=Defer_builds_mode&oldid=5801)"


---

## Depth cue

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Example
  * 4 See Also



## Overview

One can adjust the fog that PyMol overlays on objects in the viewer window. This fog helps to assist in emphasizing what is in the foreground and backgound of the image with respect to the camera. 

In mouse mode, 3-Button Viewing: 

Rear clipping plane- hold down the shift key and the right mouse button and drag the mouse to the right, more fog will be added to the background. Drag to the left and more fog will be removed from the background. 

Front clipping plane- hold down the shift key and the right mouse button and drag the mouse toward you. You should see the clipping plane come into view. Adjust the front and rear clipping plane to focus on the area of the molecule you wish to display. 

## Syntax
    
    
    set depth_cue, 1  # turn on depth cueing
    set depth_cue, 0  # turn off depth cueing
    set fog_start, 0.45 #use a number between 0.01 and 0.99 to adjust where the fog will start in the plane of view. Default is 0.45
    

## Example

[![](/images/1/13/DepthCue0.png)](/index.php/File:DepthCue0.png) [](/index.php/File:DepthCue0.png "Enlarge")depth_cue=0 | [![](/images/6/6f/DepthCue1.png)](/index.php/File:DepthCue1.png) [](/index.php/File:DepthCue1.png "Enlarge")depth_cue=1  
---|---  
  
## See Also

[Ray_Trace_Fog](/index.php/Ray_Trace_Fog "Ray Trace Fog"), [Field_Of_View](/index.php/Field_Of_View "Field Of View")

Retrieved from "[https://pymolwiki.org/index.php?title=Depth_cue&oldid=4286](https://pymolwiki.org/index.php?title=Depth_cue&oldid=4286)"


---

## Display scale factor

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The display_scale_factor setting scales the [internal gui](/index.php/Internal_gui "Internal gui"), the [internal prompt](/index.php/Internal_prompt "Internal prompt"), label fonts etc. by a multiple of 1. The setting was introduced to scale by a factor of 2 on high resolution [4K](https://en.wikipedia.org/wiki/4K_resolution) [Retina displays](https://en.wikipedia.org/wiki/Retina_Display). 

_New in Incentive PyMOL 1.8.4_

## Usage

  * **\--retina** [command line option](/index.php/Command_Line_Options "Command Line Options") sets the factor to 2
  * [MacPyMOL](/index.php/MAC_Install#MacPyMOL "MAC Install") menu: **Display > Retina Resolution** sets the factor to 2



## Example
    
    
    set display_scale_factor, 2
    

Retrieved from "[https://pymolwiki.org/index.php?title=Display_scale_factor&oldid=12564](https://pymolwiki.org/index.php?title=Display_scale_factor&oldid=12564)"


---

## Dot radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

the setting dot_radius set the size for dots in ray tracing. 

Default is 0, where it follows [dot_width](/index.php/Dot_width "Dot width")

Retrieved from "[https://pymolwiki.org/index.php?title=Dot_radius&oldid=5550](https://pymolwiki.org/index.php?title=Dot_radius&oldid=5550)"


---

## Dot width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting defines the size of the dots, in pixels, in the dots representation. 

## Syntax

set dot_width=_int_

Where _int_ is any integer, default is 1 pixel. 

  


## Example
    
    
    set dot_width=4
    

Retrieved from "[https://pymolwiki.org/index.php?title=Dot_width&oldid=3383](https://pymolwiki.org/index.php?title=Dot_width&oldid=3383)"


---

## Ellipsoid color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting controls the color of ellipsoids. 

# Syntax
    
    
    # proper settings are colors
    set ellipsoid_color, color
    
    # for example set it to blue
    set ellipsoid_color, blue
    

# See Also

[Color](/index.php/Color "Color"), [The Coloring Category](/index.php/Category:Coloring "Category:Coloring")

Retrieved from "[https://pymolwiki.org/index.php?title=Ellipsoid_color&oldid=6190](https://pymolwiki.org/index.php?title=Ellipsoid_color&oldid=6190)"


---

## Ellipsoid quality

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting determines how much refinement PyMOL uses in rendering ellipsoids. Other [representations](/index.php/Category:Representations "Category:Representations") have similar settings. The most often used is [Surface_quality](/index.php/Surface_quality "Surface quality"). 

This set to 0 is a rough approximation. Higher values, like 1, 2, 3 make truer representations. 

# Syntax
    
    
    # set the quality to some positive integer
    set ellipsoid_quality, int
    
    # for example, turn up the quality for rendering
    set ellipsoid_quailty, 3
    

# See Also

[Ray](/index.php/Ray "Ray"), [Surface_Quality](/index.php?title=Surface_Quality&action=edit&redlink=1 "Surface Quality \(page does not exist\)")

Retrieved from "[https://pymolwiki.org/index.php?title=Ellipsoid_quality&oldid=8362](https://pymolwiki.org/index.php?title=Ellipsoid_quality&oldid=8362)"


---

## Ellipsoid scale

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This sets the relative scale of drawing [Ellipsoids](/index.php/Ellipsoids "Ellipsoids"). Larger numbers make larger ellipsoids. 

# Syntax
    
    
    # set to some real valued size
    set ellipsoid_scale, float
    
    # for example, magnify the ellipses
    # by drawing them 1.4x their normal size
    set ellipsoid_scale, 1.4
    

Retrieved from "[https://pymolwiki.org/index.php?title=Ellipsoid_scale&oldid=6184](https://pymolwiki.org/index.php?title=Ellipsoid_scale&oldid=6184)"


---

## Ellipsoid transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This controls the level of [Transparency](/index.php/Transparency "Transparency") with which PyMOL renders ellipsiods. Setting this to 0 means full opacity (no transparency); setting this to 1.0 means fully transparent. 

# Syntax
    
    
    # set to a float in the range [0,1]
    set ellipsoid_transparency, float
    
    # for example, set it to 0.35
    set ellipsoid_transparency, 0.35
    

# See Also

[Category:Representations](/index.php/Category:Representations "Category:Representations"), [Transparency](/index.php/Transparency "Transparency"), [Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Ellipsoid_transparency&oldid=6183](https://pymolwiki.org/index.php?title=Ellipsoid_transparency&oldid=6183)"


---

## Fetch Host

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The **fetch_host** setting controls from which PDB server you download files when using the [fetch](/index.php/Fetch "Fetch") command. You options are: 

  * pdb
  * pdbe
  * pdbj
  * mirror URL with "divided" layout _(new in PyMOL 1.6)_



for the standard RCSB (pdb), pdb mirror in europe, and the pdb mirror in Japan. 

_New in PyMOL 1.3_

## Details

The exact URL (or list of URLs to try) is stored in `pymol.importing.hostPaths[type]`. The **fetch_host** setting only applies to the incomplete (not starting with `scheme://`) URL entries there. 

## Usage
    
    
    set fetch_host, pdb
    set fetch_host, pdbe
    set fetch_host, pdbj
    set fetch_host, <http://ftp.wwpdb.org/pub/pdb>
    set fetch_host, file:///local/mirror/pub/pdb
    
    
    
    # Paths which are appended to fetch_host
    import pymol
    pymol.importing.hostPaths["cif"] = "/data/structures/divided/mmCIF/{mid}/{code}.cif.gz"
    pymol.importing.hostPaths["pdb"] = "/data/structures/divided/pdb/{mid}/pdb{code}.ent.gz"
    
    
    
    # Don't use fetch_host, use a custom URL
    import pymol
    pymol.importing.hostPaths["cif"] = "http://files.rcsb.org/download/{code}.{type}.gz"
    pymol.importing.hostPaths["pdb"] = "http://files.rcsb.org/download/{code}.{type}.gz"
    

## See Also

[fetch](/index.php/Fetch "Fetch"), [fetch_path](/index.php/Fetch_Path "Fetch Path")

Retrieved from "[https://pymolwiki.org/index.php?title=Fetch_Host&oldid=12694](https://pymolwiki.org/index.php?title=Fetch_Host&oldid=12694)"


---

## Fetch Path

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Fetch_Path sets the default path that PyMOL uses to load files from _before_ it tries to download them from the PDB. 

## Contents

  * 1 Details
  * 2 Examples
  * 3 Hints
  * 4 See Also



# Details

If you have a local copy of the PDB on your machine, say in directory **/spc/pdb** then, once you set Fetch_Path to that directory the [fetch](/index.php/Fetch "Fetch") command will look in **/spc/pdb** before going to the PDB for your file. Also, the files fetched from the PDB are stored here once downloaded. PyMOL will only look for files starting with the pdb code in lower case. 

# Examples

On Linux or MacOS: 
    
    
    set fetch_path, /spc/pdb
    

On Windows: 
    
    
    set fetch_path, D:\mypdbs
    

Using `~/fetch_path`: 
    
    
    cmd.set('fetch_path', cmd.exp_path('~/fetch_path'), quiet=0)
    

# Hints

Put this in your [pymolrc](/index.php/Pymolrc "Pymolrc") to make it permanent. 

# See Also

  * [fetch](/index.php/Fetch "Fetch")
  * [pymolrc](/index.php/Pymolrc "Pymolrc")



Retrieved from "[https://pymolwiki.org/index.php?title=Fetch_Path&oldid=12484](https://pymolwiki.org/index.php?title=Fetch_Path&oldid=12484)"


---

## Fetch type default

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When downloading structures from the wwPDB with the [fetch](/index.php/Fetch "Fetch") command, the fetch_type_default setting defines what file format will be downloaded. The setting can be overruled with the **type** argument. 

_New in PyMOL 1.8.6_

## Values

  * **cif** (default)
  * **mmtf**
  * **pdb**
  * **pdb1**



## Examples
    
    
    set fetch_type_default, mmtf
    fetch 1rx1
    

## See Also

  * [assembly](/index.php/Assembly "Assembly")
  * [fetch](/index.php/Fetch "Fetch")
  * [fetch_path](/index.php/Fetch_Path "Fetch Path")
  * [pymolrc](/index.php/Pymolrc "Pymolrc")



Retrieved from "[https://pymolwiki.org/index.php?title=Fetch_type_default&oldid=12550](https://pymolwiki.org/index.php?title=Fetch_type_default&oldid=12550)"


---

## Field Of View

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Examples
  * 3 Syntax
  * 4 See Also



## Overview

**field_of_view** sets the amount of perspective to apply to the scene. Typical values range from 40-60. Turning off **field_of_view** amounts to the same thing as setting [orthoscopic](/index.php/Orthoscopic "Orthoscopic") to **1**. The higher the value of **field_of_view** the more amplified is the effect of the perspective. 

## Examples

  * [![No field of View](/images/8/83/No_fov.png)](/index.php/File:No_fov.png "No field of View")

No field of View 

  * [![Field of View set to 60](/images/1/14/Fov60.png)](/index.php/File:Fov60.png "Field of View set to 60")

Field of View set to 60 




## Syntax
    
    
    set field_of_view, 60
    

## See Also

[orthoscopic](/index.php/Orthoscopic "Orthoscopic") [Ray_orthoscopic](/index.php/Ray_orthoscopic "Ray orthoscopic")

Retrieved from "[https://pymolwiki.org/index.php?title=Field_Of_View&oldid=8552](https://pymolwiki.org/index.php?title=Field_Of_View&oldid=8552)"


---

## Fog

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set fog allows Depth Cue (or fog!) to be turned on or off in the interactive viewer, not for ray traced images, for this see [Ray_Trace_Fog](/index.php/Ray_Trace_Fog "Ray Trace Fog")

## Syntax
    
    
    # turn off the fog
    set fog, off
    set fog, 0
    
    # turn on the fog
    set fog, on
    set fog, 1
    

## Example

  * [![fog on](/images/f/fd/Fog_on.png)](/index.php/File:Fog_on.png "fog on")

fog on 

  * [![fog off](/images/1/17/Fog_off.png)](/index.php/File:Fog_off.png "fog off")

fog off 




Retrieved from "[https://pymolwiki.org/index.php?title=Fog&oldid=5518](https://pymolwiki.org/index.php?title=Fog&oldid=5518)"


---

## Fog start

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set fog_start adjusts the gradient strength of Depth Cue/fog.  
A low value gives a more generally cloudy image with a gradual introduction of fog. A larger value gives a sharper introduction of fog into the image. 

## Syntax
    
    
    set fog_start, <number>      #default 0.5
    

## Example

  * [![fog_start -0.5](/images/f/fb/Fog_start_minus0.5.png)](/index.php/File:Fog_start_minus0.5.png "fog_start -0.5")

fog_start -0.5 

  * [![fog_start 0.5 \(default\)](/images/c/c8/Fog_start_default.png)](/index.php/File:Fog_start_default.png "fog_start 0.5 \(default\)")

fog_start 0.5 (default) 

  * [![fog_start 1](/images/d/d5/Fog_start_1.png)](/index.php/File:Fog_start_1.png "fog_start 1")

fog_start 1 




Retrieved from "[https://pymolwiki.org/index.php?title=Fog_start&oldid=4282](https://pymolwiki.org/index.php?title=Fog_start&oldid=4282)"


---

## Grid mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Example
  * 4 See Also



## Overview

**grid_mode** partitions the screen into a grid and displays each molecule in one grid location. Each molecule rotates, zooms, etc in that grid. A possibly very useful option for PyMOL. Each grid area is assigned a [grid_slot](/index.php/Grid_slot "Grid slot") number. This allows you to assign objects to certain grids; very helpful. This can be useful when comparing homologous structures and you want to view and rotate the aligned structures side by side. This is similar to a multiple split screen view of the proteins. 

**Hint:** When using grid_mode with many molecules, it's sometimes good to align their centers of mass. This puts them all squarely in the middle of their grid element. The **alignto** command from [cealign](/index.php/Cealign "Cealign") can do this for you. 

**Hint:** If you don't want to align the molecules (it's not really necessary for grid_mode) you can just center them on some coordinate. See [COM](/index.php/COM "COM"). 

**Hint:** For most uses, instead of using the above ways to calculating centering, you may be fine just executing the 'reset' command on the command line after running this mode to get full molecules in view of window and centered. 

## Syntax
    
    
    # turn on grid mode
    set grid_mode,1
    
    # turn off grid mode
    set grid_mode,0
    

  


## Example

  * [![grid_mode off. Four molecules shown loaded and aligned.](/images/d/dd/Gm0.png)](/index.php/File:Gm0.png "grid_mode off. Four molecules shown loaded and aligned.")

grid_mode off. Four molecules shown loaded and aligned. 

  * [![grid_mode on. Each molecule is in its own window.](/images/6/63/Gm1.png)](/index.php/File:Gm1.png "grid_mode on. Each molecule is in its own window.")

grid_mode on. Each molecule is in its own window. 

  * [![grid_mode on. Lots of molecules shown.](/images/2/28/Gm2.png)](/index.php/File:Gm2.png "grid_mode on. Lots of molecules shown.")

grid_mode on. Lots of molecules shown. 

  * [![grid_mode on. 159 proteins shown!](/images/3/32/Wow_grid_mode.png)](/index.php/File:Wow_grid_mode.png "grid_mode on. 159 proteins shown!")

grid_mode on. 159 proteins shown! 




## See Also

[grid_slot](/index.php/Grid_slot "Grid slot")

Retrieved from "[https://pymolwiki.org/index.php?title=Grid_mode&oldid=13263](https://pymolwiki.org/index.php?title=Grid_mode&oldid=13263)"


---

## Grid slot

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Sets the grid slot location for a given object. 

## Syntax
    
    
    set grid_slot, int, obj
    

where **int** is the number of the grid slot and **obj** is the object to put there. 

Additionally, multiple objects can be assigned to the same grid slot by simply specifying different objects to the same grid slot. For example, if you have four objects loaded into PyMOL (A, B, C, D) and you want to show objects A and B in one slot while C and D are in another. Object E is shown in all slots by setting slot to -2: 
    
    
    set grid_slot, 1, A
    set grid_slot, 1, B
    set grid_slot, 2, C
    set grid_slot, 2, D
    set grid_slot, -2, E
    

One use for this representation is in the context of structure prediction/validation (CASP) where you may want to show how your structure looks with respect to the native strucutre versus how somebody else's structure looks with respect to the native. Using grid_slot (as shown) will allow for a clearer, side-by-side comparison rather than the traditional superimposition of all three structures. 

## See Also

[grid_mode](/index.php/Grid_mode "Grid mode")

Retrieved from "[https://pymolwiki.org/index.php?title=Grid_slot&oldid=6432](https://pymolwiki.org/index.php?title=Grid_slot&oldid=6432)"


---

## Half bonds

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The half_bonds setting determines whether a half bond will be drawn between visible and hidden atoms. 

[![](/images/7/70/Half-yes.png)](/index.php/File:Half-yes.png) Stick representation of a glycine showing half bond. |  [![](/images/d/d3/Halfbond-off.jpg)](/index.php/File:Halfbond-off.jpg) Stick representation of a glycine without half bond (default).  
---|---  
  
## Syntax

set half_bonds, _boolean_

Where _boolean_ may be one of: 

0 0.0 off false 

or: 

1 1.0 on true 

## Example

To turn on half bonds: 
    
    
    set half_bonds, 1
    

To turn off half bonds (default): 
    
    
    set half_bonds, off
    

Retrieved from "[https://pymolwiki.org/index.php?title=Half_bonds&oldid=3387](https://pymolwiki.org/index.php?title=Half_bonds&oldid=3387)"


---

## Hash max

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Example
  * 4 See Also



## Overview

hash_max sets how much memory PyMOL uses when ray tracing. Higher values will enable PyMOL to ray trace more quickly, provided you can secure the necessary memory. Thus for large scenes, it can be useful to increase this value. Simpler scenes probably don't need it. If **hash_max** is set too high, then PyMOL can (and will) crash when it attempts to use more memory than available. Likewise, **hash_max** can be used to limit the use of memory to avoid crashes at the expense of increased computing time. 

## Syntax
    
    
    set hash_max, int
    

where **int** is a positive integer. The default value is 100. 

## Example
    
    
    set hash_max, 200
    

## See Also

[ray](/index.php/Ray "Ray"), [png](/index.php/Png "Png"), [draw](/index.php/Draw "Draw")

Retrieved from "[https://pymolwiki.org/index.php?title=Hash_max&oldid=11525](https://pymolwiki.org/index.php?title=Hash_max&oldid=11525)"


---

## Hide underscore names

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

PyMOL uses names prefixed with "_" (underscore) as special internal objects and selections. Setting "hide_underscore_names" in PyMOL will hide these objects or selections from the control panel. Unsetting it will show them all. 

Retrieved from "[https://pymolwiki.org/index.php?title=Hide_underscore_names&oldid=9031](https://pymolwiki.org/index.php?title=Hide_underscore_names&oldid=9031)"


---

## Ignore case

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The ignore_case setting (default: **on** , _except in PyMOL 1.8.0.0 - 1.8.0.4_) controls whether PyMOL does case sensitive matching of atomic identifiers and selection operators in the selection language. Most notably, it affects whether chain identifiers are matched case sensitive, which becomes relevant when using upper and lower case chain identifiers in a structure with more than 26 chains. 

The default value was changed to **off** in PyMOL 1.8.0.0. However, due to undesired side effects, the **on** default was restored in 1.8.0.5. 

The next PyMOL version (expect 1.8.2) will introduce a new [ignore_case_chain](/index.php/Ignore_case_chain "Ignore case chain") setting to address the issue of mixed case chain identifiers. 

See also: <https://sourceforge.net/p/pymol/mailman/message/34815599/>

## Example

Load 1a00 which has chains A, B, C, D 
    
    
    PyMOL>fetch 1a00, async=0
    

1) case insensitive selection language (default for PyMOL <= 1.7.6 and >= 1.8.0.5) 
    
    
    PyMOL>set ignore_case
     Setting: ignore_case set to on.
    PyMOL>count_atoms chain A
     count_atoms: 1164 atoms
    PyMOL>count_atoms chain a
     count_atoms: 1164 atoms
    

2) case sensitive selection language (default for PyMOL 1.8.0.0 - 1.8.0.4) 
    
    
    PyMOL>set ignore_case, off
     Setting: ignore_case set to off.
    PyMOL>count_atoms chain A
     count_atoms: 1164 atoms
    PyMOL>count_atoms chain a
     count_atoms: 0 atoms
    

## Best Practice for Scripting

When writing scripts or plugins, all selection expressions should be strict about case (e.g. "name CA" and not "name ca") to not depend on the users setting of ignore_case. 

Retrieved from "[https://pymolwiki.org/index.php?title=Ignore_case&oldid=12289](https://pymolwiki.org/index.php?title=Ignore_case&oldid=12289)"


---

## Ignore case chain

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The ignore_case_chain setting (default: **off**) controls whether the **chain** and **segi** [selection operators](/index.php/Selection_Algebra "Selection Algebra") are case sensitive. The default is to be case sensitive, so "chain A" is different from "chain a". 

_New In PyMOL 1.8.2 (see[ignore_case](/index.php/Ignore_case "Ignore case") for previous versions)_

## Example

The PDB structure **3tcx** has 28 chains, including "A" and "a". 
    
    
    PyMOL>fetch 3tcx, async=0
    PyMOL>count_atoms chain A
     count_atoms: 641 atoms
    PyMOL>set ignore_case_chain, on
    PyMOL>count_atoms chain A
     count_atoms: 1282 atoms
    

## See Also

  * [ignore_case](/index.php/Ignore_case "Ignore case")
  * [Selection Algebra](/index.php/Selection_Algebra "Selection Algebra")



Retrieved from "[https://pymolwiki.org/index.php?title=Ignore_case_chain&oldid=12695](https://pymolwiki.org/index.php?title=Ignore_case_chain&oldid=12695)"


---

## Internal feedback

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

The Internal_feedback is the openGL-based command line area. Setting this to 0 removes the area; setting it to 1 makes one line visible (so you can see what you're typing); or you can set it to a higher value and see more lines internally (see images below). 

  * [![Internal_feedback set to 1](/images/f/f3/InternalFeedback1.png)](/index.php/File:InternalFeedback1.png "Internal_feedback set to 1")

Internal_feedback set to 1 

  * [![Internal_feedback set to 10.](/images/2/28/InternalFeedback10.png)](/index.php/File:InternalFeedback10.png "Internal_feedback set to 10.")

Internal_feedback set to 10. 




# Syntax
    
    
    # set the number of lines shown to 'int', where int is
    # an integer >= 0.
    set internal_feedback, int
    
    # For example, show 10 lines of internal feedback
    set internal_feedback, 10
    

# See Also

[GUI Category](/index.php/Category:GUI "Category:GUI"), [Internal_gui](/index.php/Internal_gui "Internal gui")

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_feedback&oldid=6162](https://pymolwiki.org/index.php?title=Internal_feedback&oldid=6162)"


---

## Internal gui

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

The internal GUI is the openGL-based user interface (that is--not the TK-window). Here you can enable/disable proteins from view, color then, change their representation--a whole host of things. If you want a nice clean presentation mode, you can turn off the internal GUI using this setting. 

  * [![Internal GUI on](/images/9/98/Ig_on.png)](/index.php/File:Ig_on.png "Internal GUI on")

Internal GUI on 

  * [![Internal GUI off](/images/f/f3/Ig_off.png)](/index.php/File:Ig_off.png "Internal GUI off")

Internal GUI off 




# Syntax
    
    
    # turn the internal gui off
    set internal_gui, off
    
    # turn it back on
    set internal_gui, on
    

# See Also

[GUI Category](/index.php/Category:GUI "Category:GUI") , [Group](/index.php/Group "Group")

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_gui&oldid=10532](https://pymolwiki.org/index.php?title=Internal_gui&oldid=10532)"


---

## Internal Gui

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The **internal GUI** is the GUI that is natively built into PyMOL. 

## Settings

Hide the internal gui with, 
    
    
    set internal_gui, 0
    

or show it with 
    
    
    set internal_gui, 1
    

### Size

There's a small vertical bar on the bottom left of the VCR-style controls -- drag this left or right to resize the internal gui. This may allow you to resize the internal gui to see full names longer than 16 characters. 

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_Gui&oldid=4323](https://pymolwiki.org/index.php?title=Internal_Gui&oldid=4323)"


---

## Internal gui control size

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This sets how large the internal gui controls are. An internal GUI control is the button you click on to enable/disable a protein, select colors, etc. 

  * [![Set to default, 18.](/images/c/cd/InteranlGuiControlSize_Default.png)](/index.php/File:InteranlGuiControlSize_Default.png "Set to default, 18.")

Set to default, 18. 

  * [![Set to 30](/images/4/46/Igcs_df30.png)](/index.php/File:Igcs_df30.png "Set to 30")

Set to 30 




# Syntax
    
    
    # set the size to some positive integer
    set internal_gui_control_size, int
    
    # for example,
    set internal_gui_control_size, 30
    

# See Also

[Get](/index.php/Get "Get"), [Internal_gui](/index.php/Internal_gui "Internal gui")

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_gui_control_size&oldid=6164](https://pymolwiki.org/index.php?title=Internal_gui_control_size&oldid=6164)"


---

## Internal gui mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This sets how PyMOL renders the internal GUI & controls. The default setting, 0. 

  * [![internal_gui_mode, set to 0](/images/0/05/Ig_mode0.png)](/index.php/File:Ig_mode0.png "internal_gui_mode, set to 0")

internal_gui_mode, set to 0 

  * [![internal_gui_mode, set to 1. Notice the coloring under the controls.](/images/6/65/Ig_mode1.png)](/index.php/File:Ig_mode1.png "internal_gui_mode, set to 1. Notice the coloring under the controls.")

internal_gui_mode, set to 1. Notice the coloring under the controls. 




# Syntax
    
    
    # set to 0 or 1
    set internal_gui_mode, int
    
    # for example, put it in mode 1
    set internal_gui_mode, 1
    

# See Also

[GUI Category](/index.php/Category:GUI "Category:GUI"), [Internal_gui](/index.php/Internal_gui "Internal gui")

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_gui_mode&oldid=6165](https://pymolwiki.org/index.php?title=Internal_gui_mode&oldid=6165)"


---

## Internal gui width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting determines with width of the openGL-based internal GUI in pixels. If you have objects with very long names, this might come in handy so you can see more of the names. 

In the following images, notice the size of the GUI controls on the right: 

  * [![internal_gui_width set to the default, 220.](/images/0/0d/Igwidth220.png)](/index.php/File:Igwidth220.png "internal_gui_width set to the default, 220.")

internal_gui_width set to the default, 220. 

  * [![internal_gui_width set to 50.](/images/3/3a/Igwidth50.png)](/index.php/File:Igwidth50.png "internal_gui_width set to 50.")

internal_gui_width set to 50. 

  * [![internal_gui_width set to the default, 500.](/images/a/a9/Igwidth500.png)](/index.php/File:Igwidth500.png "internal_gui_width set to the default, 500.")

internal_gui_width set to the default, 500. 




# Syntax
    
    
    # set to a positive integer width (in pixels)
    set internal_gui_width, int
    
    # for example; set the width to 446 pixels.
    set internal_gui_width, 446
    

# See Also

[Get](/index.php/Get "Get"), [Set](/index.php/Set "Set"), [GUI Category](/index.php/Category:GUI "Category:GUI"),[Internal_gui](/index.php/Internal_gui "Internal gui")

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_gui_width&oldid=6166](https://pymolwiki.org/index.php?title=Internal_gui_width&oldid=6166)"


---

## Internal prompt

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Look even nerdier in front of your friends ! set internal_prompt,f ! 

internal_prompt turns the internal 'PyMOL>' prompt on or off. The typed commands are not seen anymore, unless you use the esc toggle, which allows you as usual to see the console dialogue. The last line of verbose is still visible where the command line used to be. 

## Syntax

set internal_prompt,_boolean_

where _boolean_ can be either of true, false, t, f, 1 or 0. 

## Feature request

I wish it was possible to replace the usual prompt by any given string, and especially by the current path like it is possible in other consoles. 

Retrieved from "[https://pymolwiki.org/index.php?title=Internal_prompt&oldid=4322](https://pymolwiki.org/index.php?title=Internal_prompt&oldid=4322)"


---

## Label anchor

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Label_anchor is a new setting that controls onto which atoms residue labels will go. Selecting L > Label > Residues creates labels for your object or selection and they default to the alpha carbon. You can set Label_anchor to 'CB' or any other atom type to label on that atom instead of the alpha carbon. 

Note: If you set this to an atom that doesn't exist in a residue, that label will be blank. Eg, setting this to "CB" will _not label_ glycine residues. 

# Usage
    
    
    # default labels will be on beta carbons now
    set label_anchor, "CB"
    

Retrieved from "[https://pymolwiki.org/index.php?title=Label_anchor&oldid=8063](https://pymolwiki.org/index.php?title=Label_anchor&oldid=8063)"


---

## Label angle digits

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

label_angle_digits sets the precision of angle labels 

# Usage
    
    
    # set angle labels to display 2 decimals places to the right of the period
    set label_angle_digits, 2
    

# See Also

  * [label_distance_digits](/index.php/Label_distance_digits "Label distance digits")
  * [label_dihedral_digits](/index.php/Label_dihedral_digits "Label dihedral digits")



Retrieved from "[https://pymolwiki.org/index.php?title=Label_angle_digits&oldid=8422](https://pymolwiki.org/index.php?title=Label_angle_digits&oldid=8422)"


---

## Label color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

[![](/images/b/b0/Lce.png)](/index.php/File:Lce.png)

[](/index.php/File:Lce.png "Enlarge")

Label colors and sized changed, per object.

  


## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Comments
  * 4 See Also



## Overview

Sets the color that PyMol uses to draws/renders labels. This can be set for all objects/selections or for one in particular. 

## Syntax
    
    
    # set object's color to colorName
    set label_color, colorName, object
    
    # example showing two different objects
    # each with their own coloring.
    pseudoatom foo
    label foo, "foo"
    pseudoatom another
    label another, "Another label"
    set label_color, green, foo
    set label_color, lightpink, another
    translate [0, -10, 0], object=another
    set label_size, -2
    zoom foo or another, 10
    

# User Comments

If the coloring of the labels is not _exactly_ the same as you'd expect (say black turns out grey, or red turns out pink), then try the following settings: 
    
    
    unset depth_cue
    unset ray_label_specular
    

# See Also

[Color_Values](/index.php/Color_Values "Color Values"), [Category:Coloring](/index.php/Category:Coloring "Category:Coloring")

Retrieved from "[https://pymolwiki.org/index.php?title=Label_color&oldid=6913](https://pymolwiki.org/index.php?title=Label_color&oldid=6913)"


---

## Label dihedral digits

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

label_dihedral_digits sets the precision of dihedral labels 

# Usage
    
    
    # set dihedral labels to display 2 decimals places to the right of the period
    set label_dihedral_digits, 2
    

# See Also

  * [label_angle_digits](/index.php/Label_angle_digits "Label angle digits")
  * [label_distancel_digits](/index.php?title=Label_distancel_digits&action=edit&redlink=1 "Label distancel digits \(page does not exist\)")



Retrieved from "[https://pymolwiki.org/index.php?title=Label_dihedral_digits&oldid=8423](https://pymolwiki.org/index.php?title=Label_dihedral_digits&oldid=8423)"


---

## Label distance digits

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

label_distance_digits sets the precision of distance labels 

# Usage
    
    
    # set distance labels to display 2 decimals places to the right of the period
    set label_distance_digits, 2
    

# See Also

  * [label_angle_digits](/index.php/Label_angle_digits "Label angle digits")
  * [label_dihedral_digits](/index.php/Label_dihedral_digits "Label dihedral digits")



Retrieved from "[https://pymolwiki.org/index.php?title=Label_distance_digits&oldid=8420](https://pymolwiki.org/index.php?title=Label_distance_digits&oldid=8420)"


---

## Label font id

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The label_font_id setting sets the typeface (font family) for atom [labels](/index.php/Label "Label"). 

## Contents

  * 1 Syntax
  * 2 Available Font Families
  * 3 Special Characters
  * 4 Example 1
  * 5 Example 2
  * 6 See Also



## Syntax
    
    
    # use "Serif Bold" font family
    set label_font_id, 10
    

## Available Font Families

_Font IDs 0-4 were fixed size GLUT fonts, their support was dropped in PyMOL 1.6._

Name | label_font_id   
---|---  
Sans | 5   
Sans Oblique | 6   
Sans Bold | 7   
Sans Bold Oblique | 8   
Serif | 9   
Serif Oblique | 17   
Serif Bold | 10   
Serif Bold Oblique | 18   
Mono | 11   
Mono Oblique | 12   
Mono Bold | 13   
Mono Bold Oblique | 14   
Gentium Roman | 15   
Gentium Italic | 16   
  
## Special Characters

Several [Unicode](http://www.unicode.org/) characters are supported. They can be entered with [unicode literals](https://docs.python.org/2/howto/unicode.html#unicode-literals-in-python-source-code) as 4-digit hexadecimal escape sequences. 

**Example characters** (find the code for your character at [Unicode Charts](http://www.unicode.org/charts)): 

Code | Character | Name   
---|---|---  
`u"\u03b1"` | α | Alpha   
`u"\u03b2"` | β | Beta   
`u"\u00c5"` | Å | Ångström   
`u"\u00b1"` | ± | plus/minus   
`u"\u00b2"` | ² | superscript 2   
  
## Example 1

[![](/images/e/e0/New_fonts.jpeg)](/index.php/File:New_fonts.jpeg)

[](/index.php/File:New_fonts.jpeg "Enlarge")

The alpha and beta are Unicode characters.
    
    
    # if Python is configured with utf-8 default encoding (all incentive builds)
    label  5/CA, u"\u03b1-Helix"
    label 10/CA, u"\u03b2-Sheet"
    
    # if Python is configured differently, explicit utf-8 encoding is necessary
    label  5/CA, u"\u03b1-Helix".encode("utf-8")
    label 10/CA, u"\u03b2-Sheet".encode("utf-8")
    
    # italic
    set label_font_id, 16
    
    # make bigger
    set label_size, 50
    
    # cast shadows in ray tracing
    set label_shadow_mode, 3
    

## Example 2

[![](/images/2/2f/Font_ex.png)](/index.php/File:Font_ex.png)

[](/index.php/File:Font_ex.png "Enlarge")

Notice the Angstrom and superscript 2 characters. You can add other characters as well.
    
    
    # label residue 30 with "4.1 Ang^2 +/- 0.65 Ang^2
    label i. 30, "4.1" + u"\u00c5\u00b2  \u00b1 0.65 \u00c5\u00b2 "
    

## See Also

  * [label](/index.php/Label "Label")
  * [label_size](/index.php/Label_size "Label size")
  * [label_shadow_mode](/index.php/Label_shadow_mode "Label shadow mode")



Retrieved from "[https://pymolwiki.org/index.php?title=Label_font_id&oldid=12622](https://pymolwiki.org/index.php?title=Label_font_id&oldid=12622)"


---

## Label outline color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Sets which color PyMol uses to outlines the labels. This helps offset labels from their backgrounds. 

## Syntax
    
    
    set label_outline_color, red  # will draw red outline around the labels
    

Retrieved from "[https://pymolwiki.org/index.php?title=Label_outline_color&oldid=6222](https://pymolwiki.org/index.php?title=Label_outline_color&oldid=6222)"


---

## Label position

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

label_position sets the offset of the label relative to the atomic position. 

example: 
    
    
    set label_position,(3,2,1)
    

Offsets the labels 3 Å in X, 2 in Y and 1 in Z, relative to the [Viewport](/index.php/Viewport "Viewport"). Can be useful if [spheres](/index.php/Spheres "Spheres") needs [labeling](/index.php/Label "Label")

Retrieved from "[https://pymolwiki.org/index.php?title=Label_position&oldid=8315](https://pymolwiki.org/index.php?title=Label_position&oldid=8315)"


---

## Label shadow mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Sets whether or not PyMol renders shadows casted onto labels and whether or not the labels themselves cast shadows. 

Values 

  * 0, no shadows
  * 1, object shadows are projected onto the labels, but the labels don't cast shadows
  * 2, object shadows are projected onto the labels, AND the lables DO cast shadows
  * 3, object shadows are NOT projected onto the labels, BUT the lables DO cast shadows.



## Syntax
    
    
    set label_shadow_mode, 0  # turn off shadows projected onto lables, turn off shadows cast from labels
    set label_shadow_mode, 1  # turn on shadows projected onto labels, turn off shadows cast from labels
    set label_shadow_mode, 2  # turn off shadows projected onto lables, turn on shadows cast from labels
    set label_shadow_mode, 3  # turn on shadows projected onto lables, turn on shadows cast from labels
    

Retrieved from "[https://pymolwiki.org/index.php?title=Label_shadow_mode&oldid=4317](https://pymolwiki.org/index.php?title=Label_shadow_mode&oldid=4317)"


---

## Label size

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Sets how large the labels are rendered. You can use positive numbers 2, 3, 4, etc for point sizes, or negative numbers for Angstrom-based sizes. Default is 14 points. Labels in Angstrom-size scale with the distance from the front plane, labels in point-size don't. This automatic scaling works with the [draw](/index.php/Draw "Draw") command but not with the [ray](/index.php/Ray "Ray") command. 

## Syntax
    
    
    # set the label size to 10pt
    set label_size, 10
    
    # set the label size to 1.5 Ang. -- large!
    set label_size, -1.5
    

Retrieved from "[https://pymolwiki.org/index.php?title=Label_size&oldid=13561](https://pymolwiki.org/index.php?title=Label_size&oldid=13561)"


---

## Light

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light count

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 PyMol API
  * 4 Example
  * 5 See Also



## Overview

set light_count defines the number of light sources.  
Setting to 0 or 1 removes any directional light source resulting in no shadows. Light_count does affect [ambient](/index.php/Ambient "Ambient") lighting. 

## Syntax
    
    
    set light_count, <integer>                          #default 2
    

## PyMol API
    
    
    cmd.set(light_count,int sources)
    

  


## Example

  * [![light_count 0](/images/f/ff/Light_count_0.png)](/index.php/File:Light_count_0.png "light_count 0")

light_count 0 

  * [![light_count 2 \(default\)](/images/b/b2/Light_count_2.png)](/index.php/File:Light_count_2.png "light_count 2 \(default\)")

light_count 2 (default) 

  * [![light_count 10](/images/3/35/Light_count_10.png)](/index.php/File:Light_count_10.png "light_count 10")

light_count 10 




# See Also

[Light](/index.php/Light "Light")

Retrieved from "[https://pymolwiki.org/index.php?title=Light_count&oldid=12005](https://pymolwiki.org/index.php?title=Light_count&oldid=12005)"


---

## Light2

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light3

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light4

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light5

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light6

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light7

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light8

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Light9

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 User Hint
    * 3.1 The Code
  * 4 See Also



# Overview

Lighting is important for high-quality shots. PyMOL supports of up to 10 virtual lights. You can turn the lights on/off and also position them where you want (behind the camera). 

Light2..Light9 control where the other lights go. 

  * [![Normal lighting](/images/7/76/L1.png)](/index.php/File:L1.png "Normal lighting")

Normal lighting 

  * [![Light1 moved a bit. Notice how the shadows have changed.](/images/9/93/L2.png)](/index.php/File:L2.png "Light1 moved a bit. Notice how the shadows have changed.")

Light1 moved a bit. Notice how the shadows have changed. 




# Syntax
    
    
    # set the light to some position.  The 'position'
    # must be a vector specifying the XYZ location
    # to put the light.
    set light, position
    
    # for example
    set light, [ -0.55, -0.70, 0.15 ]
    

# User Hint

  * [![Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.](/images/1/1b/Ll.gif)](/index.php/File:Ll.gif "Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges.")

Moving lights make for a cool effect; you can make this look better by smoothing out the distances and ranges. 




: 

One neat trick, for rendering a "sunset" on your protein is to turn off all the lights, then render the scene as you move the light across the scene. The shadows move across the protein based on the light position and it looks like the sun is setting. 

  


## The Code

Here's the code for the animated GIF shown above. 
    
    
    python
    cmd.set("light", ll)
    for x in range(10):
            l = [ -0.4 + 2*float(x/10.), -0.4, -1  ]
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("lll" + str(x) + ".png" )
    for x in range(10):
            l[0] -= 2*float(x/10.)
            print l
            cmd.set("light", l)
            cmd.ray()
            cmd.png("llll" + str(x) + ".png" )
    python end
    

Then, in the shell do 
    
    
    convert lll* llll* light_movie.gif
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Light&oldid=6152](https://pymolwiki.org/index.php?title=Light&oldid=6152)"


---

## Line color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

Line_color controls per atom/bond coloring in objects or selections. 

[![](/images/b/ba/Lw2.png)](/index.php/File:Lw2.png)

[](/index.php/File:Lw2.png "Enlarge")

Line_color changed using set_bond. (Also, line_width was changed using set_bond)

# Syntax
    
    
    # set the color of the lines to colorName
    set line_color, colorName
    
    # set per atom/bond line colors (see examples)
    set_bond line_color, colorName, objSel
    
    # example
    fetch 1te1
    as lines
    orient
    # draw all lines red
    set line_color, red
    # draw just chain A, blue
    set_bond line_color, marine, blue
    # color the lysines magnesium!
    set_bond line_color, magnesium, resn lys
    

# See Also

  * [Set](/index.php/Set "Set")
  * [Set_bond](/index.php/Set_bond "Set bond")
  * [Lines](/index.php/Lines "Lines")
  * [Line_width](/index.php/Line_width "Line width")



Retrieved from "[https://pymolwiki.org/index.php?title=Line_color&oldid=7012](https://pymolwiki.org/index.php?title=Line_color&oldid=7012)"


---

## Line smooth

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This is a display setting. 

The **line_smooth setting** determines whether lines are **[antialias](/index.php/Antialias "Antialias")** ed. The default is to use antialiased lines. 

## Syntax

To turn off antialiasing: 
    
    
    set line_smooth, 0
    

To turn on antialising (default): 
    
    
    set line_smooth, 1
    

From python: 
    
    
    cmd.set('line_smooth', '0')
    

or 
    
    
    cmd.set('line_smooth', '1')
    

Retrieved from "[https://pymolwiki.org/index.php?title=Line_smooth&oldid=5284](https://pymolwiki.org/index.php?title=Line_smooth&oldid=5284)"


---

## Line width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

Controls the width that PyMOL draws lines. 

  * [![Line width set to default](/images/3/31/Lw1.png)](/index.php/File:Lw1.png "Line width set to default")

Line width set to default 

  * [![Line width set to 10 on residues 1-10 and then colored those residues red.](/images/b/ba/Lw2.png)](/index.php/File:Lw2.png "Line width set to 10 on residues 1-10 and then colored those residues red.")

Line width set to 10 on residues 1-10 and then colored those residues red. 




# Sytnax
    
    
    # set line width
    set line_width, value
    
    # set line_width to value for object or selection, objSel
    set_bond line_width, value, objSel
    
    # Example
    load $TUT/1hpv.pdb
    as lines
    set line_width, 10
    
    # or on a bond-by-bond basis,
    set_bond line_width, 7, i. 20-30
    

# See Also

  * [Set](/index.php/Set "Set")
  * [Set_bond](/index.php/Set_bond "Set bond")



Retrieved from "[https://pymolwiki.org/index.php?title=Line_width&oldid=7011](https://pymolwiki.org/index.php?title=Line_width&oldid=7011)"


---

## Map auto expand sym

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 OVERVIEW
  * 2 USAGE
  * 3 Note
  * 4 See Also



## OVERVIEW

When **map_auto_expand_sym** is ON, symmetry operations will be applied to expand it beyond the precalculated volume when necessary. 

## USAGE
    
    
    set map_auto_expand_sym, on
    

The default is ON. 

## Note

[isomesh](/index.php/Isomesh "Isomesh"), [isosurface](/index.php/Isosurface "Isosurface"), etc use symmetry information (lattice constants, space group) of the model specified by _selection_ argument if available, or (new in 1.7) from the map object. 

## See Also

  * [isomesh](/index.php/Isomesh "Isomesh")
  * [isosurface](/index.php/Isosurface "Isosurface")



Retrieved from "[https://pymolwiki.org/index.php?title=Map_auto_expand_sym&oldid=11506](https://pymolwiki.org/index.php?title=Map_auto_expand_sym&oldid=11506)"


---

## Max ups

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Setting max_ups (updates per second) sets how quickly PyMOL redraws itself. Set to 0, PyMOL draws as quickly as possible; set to 10, PyMOL would redraw the screen 10 times a second, max. 

Retrieved from "[https://pymolwiki.org/index.php?title=Max_ups&oldid=9038](https://pymolwiki.org/index.php?title=Max_ups&oldid=9038)"


---

## Mesh color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set mesh_color simply sets the colour of the mesh 

## Syntax
    
    
    set mesh_color, <colour>            #default is what ever the protein already is
    

## Example
    
    
    set mesh_color, blue
    

Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_color&oldid=5244](https://pymolwiki.org/index.php?title=Mesh_color&oldid=5244)"


---

## Mesh lighting

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The **mesh_lighting** setting turns lighting on or off for the mesh representation, in PyMOL. Actually, this doesn't make much difference with the default mesh settings. 

## Syntax
    
    
    # on 
    set mesh_lighting, on
    set mesh_lighting, 1 
    
    # off
    set mesh_lighting, off
    set mesh_lighting, 0
    

Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_lighting&oldid=4305](https://pymolwiki.org/index.php?title=Mesh_lighting&oldid=4305)"


---

## Mesh mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set mesh_mode changes the mesh drawing mode for non-protein objects. 

mesh_mode 1 allows you to show meshes for hetero atom and solvent atom objects. 

## Syntax
    
    
    set mesh_mode, <integer>                          #default 0
    

Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_mode&oldid=4306](https://pymolwiki.org/index.php?title=Mesh_mode&oldid=4306)"


---

## Mesh negative visible

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

If the **mesh_negative_visible** setting is **on** , then an [isomesh](/index.php/Isomesh "Isomesh") at iso level 3.0 will in addition show the -3.0 contour. The color of the negative contour is defined by the **mesh_negative_color** setting. The color of the positive contour is defined with the [color](/index.php/Color "Color") command, or with the [mesh_color](/index.php/Mesh_color "Mesh color") setting. 

The **surface_negative_visible** and **surface_negative_color** settings control the same behavior for [isosurface](/index.php/Isosurface "Isosurface") objects. 

Since PyMOL 1.8.4 the negative color can be controlled from the color menu ("C > negative > ..."). 

## Example

Show the +/- 3 sigma iso-level mesh ("wire-frame") of a difference density map. 
    
    
    fetch 1ubq, diffmap, type=fofc, async=0
    isomesh diffmesh, diffmap, 3.0
    color green, diffmesh
    set mesh_negative_color, red
    set mesh_negative_visible
    

Same, but with a closed surface: 
    
    
    fetch 1ubq, diffmap, type=fofc, async=0
    isosurface diffsurf, diffmap, 3.0
    color green, diffsurf
    set surface_negative_color, red
    set surface_negative_visible
    

## Color Menu

  * [![Action \(A\) menu of a map object](/images/a/a5/A-menu-isomesh-negative.png)](/index.php/File:A-menu-isomesh-negative.png "Action \(A\) menu of a map object")

Action (A) menu of a map object 

  * [![Negative color \(C > negative\) menu of an isomesh object](/images/3/38/C-menu-isomesh-negative.png)](/index.php/File:C-menu-isomesh-negative.png "Negative color \(C > negative\) menu of an isomesh object")

Negative color (C > negative) menu of an isomesh object 




## See Also

  * [isomesh](/index.php/Isomesh "Isomesh")
  * [isosurface](/index.php/Isosurface "Isosurface")



Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_negative_visible&oldid=12451](https://pymolwiki.org/index.php?title=Mesh_negative_visible&oldid=12451)"


---

## Mesh quality

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Altering the **mesh_quality** setting adjusts how precisely the mesh represents the surface. The higher the value the more curved and precise the surface is defined. The default is good compromise for quality and [speed](/index.php/Category:Performance "Category:Performance") (mesh calculation time). 

## Syntax
    
    
    # The default is 2.
    set mesh_quality, <integer>
    

## Example

  * [![mesh_quality 0 \(not smooth\)](/images/8/8e/Mesh_quality_0.jpg)](/index.php/File:Mesh_quality_0.jpg "mesh_quality 0 \(not smooth\)")

mesh_quality 0 (not smooth) 

  * [![mesh_quality 1](/images/3/35/Mesh_quality_1.jpg)](/index.php/File:Mesh_quality_1.jpg "mesh_quality 1")

mesh_quality 1 

  * [![mesh_quality 2 \(default\)](/images/6/65/Min_mesh_spacing_default.jpg)](/index.php/File:Min_mesh_spacing_default.jpg "mesh_quality 2 \(default\)")

mesh_quality 2 (default) 




Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_quality&oldid=5510](https://pymolwiki.org/index.php?title=Mesh_quality&oldid=5510)"


---

## Mesh type

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set mesh_type adjusts the way a surface mesh is displayed  

    
    
    -1 nothing
    0  square grid (default)
    1  dots
    

At present (Pymol version 0.99rc6) type 1 does not ray trace correctly 

## Syntax
    
    
    set mesh_type, <integer>                          #default 0
    

## Example

  * [![mesh_type 0 square grid \(default\)](/images/6/65/Min_mesh_spacing_default.jpg)](/index.php/File:Min_mesh_spacing_default.jpg "mesh_type 0 square grid \(default\)")

mesh_type 0 square grid (default) 

  * [![mesh_type 1 dots](/images/b/b8/Mesh_type_1.png)](/index.php/File:Mesh_type_1.png "mesh_type 1 dots")

mesh_type 1 dots 

  * [![mesh_type 1 dots \(raytrace attempt\)](/images/7/7d/Mesh_type_1_ray.jpg)](/index.php/File:Mesh_type_1_ray.jpg "mesh_type 1 dots \(raytrace attempt\)")

mesh_type 1 dots (raytrace attempt) 




Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_type&oldid=4308](https://pymolwiki.org/index.php?title=Mesh_type&oldid=4308)"


---

## Mesh width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set mesh_width adjusts the thickness of the mesh lines 

## Syntax
    
    
    set mesh_width, <number>                          #default 0.5
    

## Example

  * [![mesh_width 0.5 \(default\)](/images/6/65/Min_mesh_spacing_default.jpg)](/index.php/File:Min_mesh_spacing_default.jpg "mesh_width 0.5 \(default\)")

mesh_width 0.5 (default) 

  * [![mesh_width 3](/images/1/1f/Mesh_width_3.jpg)](/index.php/File:Mesh_width_3.jpg "mesh_width 3")

mesh_width 3 




Retrieved from "[https://pymolwiki.org/index.php?title=Mesh_width&oldid=4309](https://pymolwiki.org/index.php?title=Mesh_width&oldid=4309)"


---

## Min mesh spacing

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The min_mesh_spacing setting controls the spacing between lines of the molecular [mesh](/index.php/Mesh "Mesh") representation. 

It does not affect [isomesh](/index.php/Isomesh "Isomesh") objects. Use [map_double](/index.php/Map_double "Map double") or [map_halve](/index.php?title=Map_halve&action=edit&redlink=1 "Map halve \(page does not exist\)") instead. 

## Syntax
    
    
    set min_mesh_spacing, <number>                      #default 0.6
    

## Example

  * [![min_mesh_spacing 0.6 \(default\)](/images/6/65/Min_mesh_spacing_default.jpg)](/index.php/File:Min_mesh_spacing_default.jpg "min_mesh_spacing 0.6 \(default\)")

min_mesh_spacing 0.6 (default) 

  * [![min_mesh_spacing 2](/images/7/70/Min_mesh_spacing_2.jpg)](/index.php/File:Min_mesh_spacing_2.jpg "min_mesh_spacing 2")

min_mesh_spacing 2 




Retrieved from "[https://pymolwiki.org/index.php?title=Min_mesh_spacing&oldid=12912](https://pymolwiki.org/index.php?title=Min_mesh_spacing&oldid=12912)"


---

## Movie fps

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The movie_fps setting only affects the playback speed within PyMOL and not the final exported movie. 

## See Also

  * [mplay](/index.php/Mplay "Mplay")
  * [movie.produce](/index.php/Movie.produce "Movie.produce")



Retrieved from "[https://pymolwiki.org/index.php?title=Movie_fps&oldid=10639](https://pymolwiki.org/index.php?title=Movie_fps&oldid=10639)"


---

## Normalize ccp4 maps

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

**Good default?** It has been suggested to change the default of this setting to **off**. If you agree or disagree, please leave a comment on the [discussion page](/index.php/Talk:Normalize_ccp4_maps "Talk:Normalize ccp4 maps"). 

The normalize_ccp4_maps settings (default: **on**) controls whether or not PyMOL normalizes the data from a ccp4 map file **upon loading**. The data is shifted and scaled so that mean=0.0 and stdev=1.0. 

Normalization is done on the raw data array, not across the unit cell. This means that if the raw data doesn't exactly cover an integral number of asymmetric units, the result will be different than for example when loading the same file into Coot, which normalizes across the cell. 

**Normalization should be turned off before loading, in cases like:**

  * the map file has already been normalized, for example with CCP4’s mapmask program
  * you want to specify mesh contour levels in raw units (e/A^3) instead of rmsd (sigma). This is recommended if you want to make figures that match the display in Coot.



## Usage
    
    
    # load map and normalize data
    set normalize_ccp4_maps, on
    load somemap.ccp4
    
    # load map as-is (without normalization)
    set normalize_ccp4_maps, off
    load somemap.ccp4
    

## See Also

  * [normalize_o_maps](/index.php?title=Normalize_o_maps&action=edit&redlink=1 "Normalize o maps \(page does not exist\)")
  * [normalize_grd_maps](/index.php?title=Normalize_grd_maps&action=edit&redlink=1 "Normalize grd maps \(page does not exist\)")
  * [load](/index.php/Load "Load")
  * [isomesh](/index.php/Isomesh "Isomesh")
  * [Display_CCP4_Maps](/index.php/Display_CCP4_Maps "Display CCP4 Maps")



Retrieved from "[https://pymolwiki.org/index.php?title=Normalize_ccp4_maps&oldid=11998](https://pymolwiki.org/index.php?title=Normalize_ccp4_maps&oldid=11998)"


---

## Cartoon nucleic acid mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Examples
  * 3 Syntax
  * 4 See Also



## Overview

This setting modifies how PyMol draws the sugar bb of nucleic acids. Some differences between the settings are subtle, however PyMol gives you the ability to get the representation you want. Click on each images for a closer look. 

## Examples

  * [![Set to 0](/images/0/0c/Cnam_0.png)](/index.php/File:Cnam_0.png "Set to 0")

Set to 0 

  * [![Set to 1](/images/7/7d/Cnam_1.png)](/index.php/File:Cnam_1.png "Set to 1")

Set to 1 

  * [![Set to 2](/images/7/73/Cnam_2.png)](/index.php/File:Cnam_2.png "Set to 2")

Set to 2 




## Syntax
    
    
    set cartoon_nucleic_acid_mode, 1  # try different natural numbers
    

## See Also

[Cartoon#Nucleic_Acid_Representation](/index.php/Cartoon#Nucleic_Acid_Representation "Cartoon")

Retrieved from "[https://pymolwiki.org/index.php?title=Cartoon_nucleic_acid_mode&oldid=4242](https://pymolwiki.org/index.php?title=Cartoon_nucleic_acid_mode&oldid=4242)"


---

## Opaque background

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting controls how PyMOL renders the background. If this opaque setting is **on** then the background is drawn as a solid color depending on your background color setting. If this is turned off, the background is drawn as transparent. Transparent backgrounds allow you to post your rendered image on a background of any color without having to first to background subtraction. This is great for posters or publications. It also works well when your presentation background is not simply black or white. In the PyMOL window a transparent background is shown as a checkerboard, but in your final image it will really be transparent. 

# Syntax
    
    
    # set it on or off
    
    # solid background
    set opaque_background, on
    
    # transparent background
    set opaque_background, off
    

# See Also

[Ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Opaque_background&oldid=6193](https://pymolwiki.org/index.php?title=Opaque_background&oldid=6193)"


---

## Pdb conect all

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When saving a PDB file, the pdb_conect_all settings controls whether PyMOL writes CONECT records for all bonds, or only for HETATM bonds. 

PDB files by convention do not contain bond information for standard polymer residues (ATOM records), because the connectivity is known for those. 

If a PDB file contains at least one CONECT record for two atoms from ATOM records, then PyMOL will set **pdb_conect_all=on** as an object-level setting, to again write CONECT records for all bonds when saving that object to a PDB file. 

## Values

  * **off** turns the feature off, CONECT records are written for HETATM bonds only
  * **on** turns the feature on, CONECT records are written for all bonds



## Example

The PDB file for [1rx1](http://pdbe.org/1rx1) contains 52 CONECT records (for the ligands). 
    
    
    fetch 1rx1, async=0
    set pdb_conect_all, on
    save 1rx1_conect_all.pdb
    

Now the file **1rx1_conect_all.pdb** contains 1316 CONECT records. 

## See Also

  * [connect_mode](/index.php/Connect_mode "Connect mode")
  * [pdb_ignore_conect](/index.php?title=Pdb_ignore_conect&action=edit&redlink=1 "Pdb ignore conect \(page does not exist\)")
  * [pdb_conect_nodup](/index.php/Pdb_conect_nodup "Pdb conect nodup")



Retrieved from "[https://pymolwiki.org/index.php?title=Pdb_conect_all&oldid=12209](https://pymolwiki.org/index.php?title=Pdb_conect_all&oldid=12209)"


---

## Pdb conect nodup

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Notes
    * 3.1 See Also



## Overview

The **pdb_conect_nodup** setting in pymol controls if duplicated connectivity record ([CONECT](http://www.wwpdb.org/documentation/file-format-content/format33/sect10.html#CONECT)) in saved PDB is used to stored [bond order](http://en.wikipedia.org/wiki/Bond_order). 

This is an unofficial PDB feature to store bond order and is supported by several applications that read PDB files, but may break PDB file loading in other applications which don't support it. PyMOL always wrote duplicated connect records (`pdb_conect_nodup=0`) and the setting allows you to switch that off (`pdb_conect_nodup=1`). 

## Syntax
    
    
    get pdb_conect_nodup     # get current value 
    set pdb_conect_nodup, 1  # no CONECT duplication is used to store bond order
    

## Notes

  * PyMOL user mailing list discussion on `pdb_conect_nodup`: <http://sourceforge.net/p/pymol/mailman/message/32735483/>
  * Jmol support for bond order: <http://wiki.jmol.org/index.php/Support_for_bond_orders>
  * Aromatic bonds are stored using four duplication of the same CONECT records, as specified in, for example, MOL format. See <http://wiki.jmol.org/index.php/Support_for_bond_orders>



### See Also

[Pdb_Conect_All](/index.php/Pdb_Conect_All "Pdb Conect All"), [Pdb_use_ter_records](/index.php/Pdb_use_ter_records "Pdb use ter records")

Retrieved from "[https://pymolwiki.org/index.php?title=Pdb_conect_nodup&oldid=12139](https://pymolwiki.org/index.php?title=Pdb_conect_nodup&oldid=12139)"


---

## Pdb echo tags

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

pdb_echo_tags sets the field(s) to be displayed in the overlay/console window upon loading a PDB file. 

# Usage
    
    
    # example usage, display HEADER,TITLE and COMPND fields
    # The syntax of the final field is a quoted, comma-separated list of PDB fields.
    set pdb_echo_tags, "HEADER,TITLE,COMPND"
    

Default value is "HEADER,TITLE,COMPND". See [PDB file format documentation](http://www.wwpdb.org/docs.html) for possible fields. 

Retrieved from "[https://pymolwiki.org/index.php?title=Pdb_echo_tags&oldid=9081](https://pymolwiki.org/index.php?title=Pdb_echo_tags&oldid=9081)"


---

## Pdb reformat names mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

pdb_reformat_names_mode sets atom naming for PDB files as it reads them. 

# Usage
    
    
    # example usage, set to amber compliant
    
    set pdb_reformat_names_mode, 2
    

Modes: 

**0**

    

    default, off

**1**

    

    PDB compliant, (eg. HH12 becomes 2HH1, etc.)

**2**

    

    amber compliant, amber compliant (eg., 2HH1 becomes HH12)

**3**

    

    pdb compliant, but use IUPAC within PyMOL

Retrieved from "[https://pymolwiki.org/index.php?title=Pdb_reformat_names_mode&oldid=9078](https://pymolwiki.org/index.php?title=Pdb_reformat_names_mode&oldid=9078)"


---

## Pdb use ter records

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When a molecule (e.g. a protein molecule) is saved in pdb format by PyMOL, a line of [TER](http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#TER) record is inserted wherever the residue id is not sequential. To suppress this feature, set `pdb_use_ter_records` value to 0 before saving protein molecules. 

## Contents

  * 1 Version History
  * 2 Example
  * 3 PyMOL API
  * 4 See Also



## Version History

  * 2.3.0 new implementation, [works correctly with gaps](https://github.com/schrodinger/pymol-testing/commit/82440d760b6b2b48b2cb1ca6c0bb5347870a4541)
  * 1.8.4 feature removed
  * <= 1.8.2 feature broken for various use cases (e.g. gaps)
  * <= 1.5.0.4 feature broken for HETATM polymer residues (e.g. MSE)



## Example
    
    
    get pdb_use_ter_records  # return current value of pdb_use_ter_records
    set pdb_use_ter_records, 0
    

## PyMOL API
    
    
    cmd.set('pdb_use_ter_records', 0)
    

## See Also

  * [pdb_conect_nodup](/index.php/Pdb_conect_nodup "Pdb conect nodup")
  * [pdb_conect_all](/index.php/Pdb_conect_all "Pdb conect all")



Retrieved from "[https://pymolwiki.org/index.php?title=Pdb_use_ter_records&oldid=12846](https://pymolwiki.org/index.php?title=Pdb_use_ter_records&oldid=12846)"


---

## Pickable

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The pickable setting is used to determine whether clicking atoms with the mouse can add or remove selections. 

## Syntax

To turn off click-to-select: 
    
    
    set pickable, 0
    

To turn on click-to-select (default): 
    
    
    set pickable, 1
    

Retrieved from "[https://pymolwiki.org/index.php?title=Pickable&oldid=6522](https://pymolwiki.org/index.php?title=Pickable&oldid=6522)"


---

## Pse binary dump

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The pse_binary_dump setting affects the PyMOL session file format (.pse). If set to **on** , session export will be significantly faster and produce smaller files. 

_New in PyMOL 1.8_

## Examples
    
    
    fetch 1ubq
    
    set pse_binary_dump, off
    save regular.pse
    
    set pse_binary_dump, on
    save binary.pse
    
    # compare file size
    print(os.path.getsize("regular.pse"))   # prints 215482
    print(os.path.getsize("binary.pse"))    # prints 118557
    

## Known Limitations

  * Before 2.4: [Atom Properties](https://pymol.org/dokuwiki/doku.php?id=properties#atom_level_properties) not supported



## See Also

  * [pse_export_version](/index.php/Pse_export_version "Pse export version")



Retrieved from "[https://pymolwiki.org/index.php?title=Pse_binary_dump&oldid=13270](https://pymolwiki.org/index.php?title=Pse_binary_dump&oldid=13270)"


---

## Pse export version

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The pse_export_version setting switches to a legacy format when saving PSEs, so far possible. The value is the floating point number representation of the targeted legacy PyMOL version, for example **1.74** to save a session for PyMOL 1.7.4. 

_New in PyMOL 1.7.6_

Session file loading is always backwards compatible, so any newer version of PyMOL can load old session files. 

## Examples
    
    
    # current version (default)
    set pse_export_version, 0
    save current.pse
    
    # compatible with PyMOL 1.7.4
    set pse_export_version, 1.74
    save 174compat.pse
    

## Known Limitations

The following objects are not or only partly supported: 

  * scenes before 1.7.6 (support added in 1.8.4)
  * volumes before 1.7.2
  * ramps
  * CGOs



## See Also

  * [save](/index.php/Save "Save")



Retrieved from "[https://pymolwiki.org/index.php?title=Pse_export_version&oldid=12455](https://pymolwiki.org/index.php?title=Pse_export_version&oldid=12455)"


---

## Ray blend blue

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

See [Ray_blend_colors](/index.php/Ray_blend_colors "Ray blend colors"). 

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_blend_blue&oldid=8997](https://pymolwiki.org/index.php?title=Ray_blend_blue&oldid=8997)"


---

## Ray blend colors

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When ray_blend_colors is set, it knocks out the RGB component of the [ray_blend_red](/index.php/Ray_blend_red "Ray blend red"), [ray_blend_green](/index.php/Ray_blend_green "Ray blend green"), [ray_blend_blue](/index.php/Ray_blend_blue "Ray blend blue"). For example, without blending a rainbow picture looks like: 

[![Ray blend off.png](/images/6/66/Ray_blend_off.png)](/index.php/File:Ray_blend_off.png)

but if, before ray tracing we set, 
    
    
    set ray_blend_colors, on
    set ray_blend_red, 1.0
    

as you can see PyMOL has knocked out all red coloring, 

[![Ray blend red.png](/images/b/bf/Ray_blend_red.png)](/index.php/File:Ray_blend_red.png)

The amount of the colored component left is: (1.0 - (ray_blend_COLOR)) * original_color 

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_blend_colors&oldid=8996](https://pymolwiki.org/index.php?title=Ray_blend_colors&oldid=8996)"


---

## Ray blend green

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

See [Ray_blend_colors](/index.php/Ray_blend_colors "Ray blend colors"). 

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_blend_green&oldid=8998](https://pymolwiki.org/index.php?title=Ray_blend_green&oldid=8998)"


---

## Ray blend red

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

See [Ray_blend_colors](/index.php/Ray_blend_colors "Ray blend colors"). 

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_blend_red&oldid=8999](https://pymolwiki.org/index.php?title=Ray_blend_red&oldid=8999)"


---

## Ray direct shade

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

This setting heavily hints the shading from the viewer's POV. 

For example, without this set: 

[![No shade.png](/images/3/3c/No_shade.png)](/index.php/File:No_shade.png)

and with this set, 

[![Shade.png](/images/d/dc/Shade.png)](/index.php/File:Shade.png)

# Usage
    
    
    # turn on direct shade
    set ray_direct_shade, on
    
    # turn off direct shade
    set ray_direct_shade, off
    

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_direct_shade&oldid=9002](https://pymolwiki.org/index.php?title=Ray_direct_shade&oldid=9002)"


---

## Ray opaque background

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting changes how PyMol treats the background. If this option is ON then the background is whatever you specify -- like black or white; however, if the setting is OFF, then the background will be treated as a transparent alpha channel. 

## Settings

Note: turning this setting **off** creates the transparent backgrounds. 
    
    
    # turn on transparent alpha channel
    set ray_opaque_background, off
    

## Examples

  * [![ray_opaque_background set to 0. The image from PyMOL is shown over a checkerboard.](/images/e/e7/Rob0.png)](/index.php/File:Rob0.png "ray_opaque_background set to 0. The image from PyMOL is shown over a checkerboard.")

ray_opaque_background set to 0. The image from PyMOL is shown over a checkerboard. 

  * [![ray_opaque_background set to 1. The image from PyMOL is shown over a checkerboard, however because the background is opaque, the checkerboard underneath does not show through.](/images/a/ab/Rob1.png)](/index.php/File:Rob1.png "ray_opaque_background set to 1. The image from PyMOL is shown over a checkerboard, however because the background is opaque, the checkerboard underneath does not show through.")

ray_opaque_background set to 1. The image from PyMOL is shown over a checkerboard, however because the background is opaque, the checkerboard underneath does not show through. 




Retrieved from "[https://pymolwiki.org/index.php?title=Ray_opaque_background&oldid=11598](https://pymolwiki.org/index.php?title=Ray_opaque_background&oldid=11598)"


---

## Ray orthoscopic

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting controls whether ray-traced images are rendered with or without perspective. Note that this can be in conflict with the setting "orthoscopic"; by default, images are rendered with the same orthoscopic setting as the viewport, unless "ray_orthoscopic" is deliberately set otherwise. 

## Settings
    
    
    set ray_orthoscopic, off   # render ray-traced images with perspective (2-4x SLOWER)
    set ray_orthoscopic, on    # render ray-traced images without perspective
    

## Examples

  * [![orthoscopic on \(perspective off\)](/images/e/ea/Orthoon2.png)](/index.php/File:Orthoon2.png "orthoscopic on \(perspective off\)")

orthoscopic on (perspective off) 

  * [![orthoscopic off \(perspective on\)](/images/f/f8/Orthooff2.png)](/index.php/File:Orthooff2.png "orthoscopic off \(perspective on\)")

orthoscopic off (perspective on) 

  * For many images, the difference is hardly visible.



  * [![orthoscopic on \(perspective off\)](/images/4/46/Orthoon.png)](/index.php/File:Orthoon.png "orthoscopic on \(perspective off\)")

orthoscopic on (perspective off) 

  * [![orthoscopic off \(perspective on\)](/images/3/3d/Orthooff.png)](/index.php/File:Orthooff.png "orthoscopic off \(perspective on\)")

orthoscopic off (perspective on) 

  * Especially when straight features parallel to the z axis are shown, the effect may be large.




Retrieved from "[https://pymolwiki.org/index.php?title=Ray_orthoscopic&oldid=8551](https://pymolwiki.org/index.php?title=Ray_orthoscopic&oldid=8551)"


---

## Ray shadow

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Turns on/off whether or not PyMOL ray traces shadows. 

## Syntax
    
    
    set ray_shadows,0   # turn off shadows
    set ray_shadows,1   # turn on shadows
    

Also accepted: 
    
    
    set ray_shadows,off   # turn off shadows
    set ray_shadows,on    # turn on shadows
    

## Examples

  * [![Ray Shadows Off](/images/3/3a/Rs0.png)](/index.php/File:Rs0.png "Ray Shadows Off")

Ray Shadows Off 

  * [![Ray Shadows On](/images/e/e1/Rs1.png)](/index.php/File:Rs1.png "Ray Shadows On")

Ray Shadows On 




Retrieved from "[https://pymolwiki.org/index.php?title=Ray_shadow&oldid=4272](https://pymolwiki.org/index.php?title=Ray_shadow&oldid=4272)"


---

## Ray Shadows

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Turns on/off whether or not PyMol ray traces shadows. LEGACY variant; superceded by "ray_shadow" 

## Syntax
    
    
    set ray_shadows,0   # turn off shadows
    set ray_shadows,1   # turn on shadows
    

Also accepted: 
    
    
    set ray_shadows,off   # turn off shadows
    set ray_shadows,on    # turn on shadows
    

## Examples

  * [![Ray Shadows Off](/images/3/3a/Rs0.png)](/index.php/File:Rs0.png "Ray Shadows Off")

Ray Shadows Off 

  * [![Ray Shadows On](/images/e/e1/Rs1.png)](/index.php/File:Rs1.png "Ray Shadows On")

Ray Shadows On 




Retrieved from "[https://pymolwiki.org/index.php?title=Ray_Shadows&oldid=4441](https://pymolwiki.org/index.php?title=Ray_Shadows&oldid=4441)"


---

## Ray shadows

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 OVERVIEW
  * 2 SYNTAX
  * 3 PYMOL API
  * 4 EXAMPLES



### OVERVIEW

The **ray_shadows** setting determines whether or not PyMOL will ray trace shadows. This is also related to the **ray_shadow** setting. 

### SYNTAX
    
    
    # turn on ray traced shadows
    set ray_shadows, on
    set ray_shadows, 1
    
    # turn off ray traced shadows
    set ray_shadows, off
    set ray_shadows, 0
    

### PYMOL API
    
    
    cmd.set('ray_shadows','on')
    cmd.set('ray_shadows','off')
    

### EXAMPLES

To illustrate the difference, look at the two following images: 

  * [![Ray_shadow turned off. Click on the image to see the full detail.](/images/3/3d/Ray_shadow_off.png)](/index.php/File:Ray_shadow_off.png "Ray_shadow turned off. Click on the image to see the full detail.")

Ray_shadow turned off. Click on the image to see the full detail. 

  * [![Ray_shadow turned on. Click on the image to see the full detail.](/images/6/60/Ray_shadow_on.png)](/index.php/File:Ray_shadow_on.png "Ray_shadow turned on. Click on the image to see the full detail.")

Ray_shadow turned on. Click on the image to see the full detail. 




Retrieved from "[https://pymolwiki.org/index.php?title=Ray_shadows&oldid=8246](https://pymolwiki.org/index.php?title=Ray_shadows&oldid=8246)"


---

## Ray texture

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Usage
  * 3 Examples
  * 4 See Also



## Overview

**ray_texture** sets what texture PyMol applies to the surface of your object during ray tracing. Options now consist of 

  1. None
  2. Matte 1
  3. Matte 2
  4. Swirl 1
  5. Swirl 2
  6. Fiber



## Usage
    
    
    set ray_texture, X      # where X = 0 ... 5
    

## Examples

The thumbnails are most likely too small to see the detail. Please click on each image to see a close up. 

  * [![Ray Texture 0 \(None\)](/images/6/68/Ray_texture_0.png)](/index.php/File:Ray_texture_0.png "Ray Texture 0 \(None\)")

Ray Texture 0 (None) 

  * [![Ray Texture 1 \(Matte 1\)](/images/d/d7/Ray_texture_1.png)](/index.php/File:Ray_texture_1.png "Ray Texture 1 \(Matte 1\)")

Ray Texture 1 (Matte 1) 

  * [![Ray Texture 2 \(Matte 2\)](/images/1/11/Ray_texture_2.png)](/index.php/File:Ray_texture_2.png "Ray Texture 2 \(Matte 2\)")

Ray Texture 2 (Matte 2) 

  * [![Ray Texture 3 \(Swirl1\)](/images/7/79/Ray_texture_3.png)](/index.php/File:Ray_texture_3.png "Ray Texture 3 \(Swirl1\)")

Ray Texture 3 (Swirl1) 

  * [![Ray Texture 4 \(Swirl 2\)](/images/7/79/Ray_texture_4.png)](/index.php/File:Ray_texture_4.png "Ray Texture 4 \(Swirl 2\)")

Ray Texture 4 (Swirl 2) 

  * [![Ray Texture 5 \(Fiber\)](/images/b/bb/Ray_texture_5.png)](/index.php/File:Ray_texture_5.png "Ray Texture 5 \(Fiber\)")

Ray Texture 5 (Fiber) 




# See Also

  * [ray_texture_settings](/index.php/Ray_texture_settings "Ray texture settings")



Retrieved from "[https://pymolwiki.org/index.php?title=Ray_texture&oldid=13562](https://pymolwiki.org/index.php?title=Ray_texture&oldid=13562)"


---

## Ray trace disco factor

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The ray_trace_disco_factor setting has an effect on [ray_trace_mode](/index.php/Ray_trace_mode "Ray trace mode") 1-3. With small values, outlines will appear at surface curvature discontinuities. 

set ray_trace_disco_factor, 0.0  | set ray_trace_disco_factor, 1.0   
---|---  
[![Ray trace disco factor-0-cartoon.png](/images/a/a7/Ray_trace_disco_factor-0-cartoon.png)](/index.php/File:Ray_trace_disco_factor-0-cartoon.png) | [![Ray trace disco factor-1-cartoon.png](/images/f/fc/Ray_trace_disco_factor-1-cartoon.png)](/index.php/File:Ray_trace_disco_factor-1-cartoon.png)  
[![Ray trace disco factor-0-spheres.png](/images/c/cf/Ray_trace_disco_factor-0-spheres.png)](/index.php/File:Ray_trace_disco_factor-0-spheres.png) | [![Ray trace disco factor-1-spheres.png](/images/d/d8/Ray_trace_disco_factor-1-spheres.png)](/index.php/File:Ray_trace_disco_factor-1-spheres.png)  
[![Ray trace disco factor-0-sticks.png](/images/8/87/Ray_trace_disco_factor-0-sticks.png)](/index.php/File:Ray_trace_disco_factor-0-sticks.png) | [![Ray trace disco factor-1-sticks.png](/images/1/1e/Ray_trace_disco_factor-1-sticks.png)](/index.php/File:Ray_trace_disco_factor-1-sticks.png)  
  
## Usage
    
    
    set ray_trace_mode
    set ray_trace_disco_factor, 1.0 ;# disable effect
    set ray_trace_disco_factor, 0.1 ;# some outlines
    set ray_trace_disco_factor, 0.0 ;# strong outlines
    

## Example
    
    
    # load a molecule
    fragment indane
    show_as spheres
    
    # white background makes black outlines
    bg_color white
    
    # enable outline mode
    set ray_trace_mode, 3
    set ray_trace_disco_factor, 0.0
    ray
    

## See Also

  * [ray_trace_slope_factor](/index.php?title=Ray_trace_slope_factor&action=edit&redlink=1 "Ray trace slope factor \(page does not exist\)")
  * [ray_trace_depth_factor](/index.php?title=Ray_trace_depth_factor&action=edit&redlink=1 "Ray trace depth factor \(page does not exist\)")
  * [ray_trace_gain](/index.php/Ray_trace_gain "Ray trace gain")
  * [ray_trace_mode](/index.php/Ray_trace_mode "Ray trace mode")



Retrieved from "[https://pymolwiki.org/index.php?title=Ray_trace_disco_factor&oldid=12861](https://pymolwiki.org/index.php?title=Ray_trace_disco_factor&oldid=12861)"


---

## Ray Trace Fog

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The **ray_trace_fog** setting forces PyMol to render **fog** over objects further from the camera. This adds a sense of depth and brings the attention to foreground objects. 

## Examples

  * [![Setting on](/images/e/e0/Fogon.png)](/index.php/File:Fogon.png "Setting on")

Setting on 

  * [![Setting off](/images/3/3b/Fogoff.png)](/index.php/File:Fogoff.png "Setting off")

Setting off 




## Syntax
    
    
    set ray_trace_fog, 0  # off
    set ray_trace_fog, 1  # on
    

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_Trace_Fog&oldid=11385](https://pymolwiki.org/index.php?title=Ray_Trace_Fog&oldid=11385)"


---

## Ray trace fog

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

Turn on or off the tracing of fog. Adding fog to the image assists in adding the feeling of depth. 

# See Also

[depth_cue](/index.php/Depth_cue "Depth cue")

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_trace_fog&oldid=4262](https://pymolwiki.org/index.php?title=Ray_trace_fog&oldid=4262)"


---

## Ray trace frames

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting determines whether or not PyMol [ray](/index.php/Ray "Ray") traces your images or just takes a screen capture of them. 

## Syntax
    
    
    set ray_trace_frames, 1  #on, 0=off
    

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_trace_frames&oldid=4263](https://pymolwiki.org/index.php?title=Ray_trace_frames&oldid=4263)"


---

## Ray trace gain

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

When ray tracing, PyMOL knows the Z-depth of coordinates. The ray_trace_gain setting darkens pixels based on their Z-depth. 

  * [![ray_trace_gain,0](/images/c/cd/Ray_trace_gain0.png)](/index.php/File:Ray_trace_gain0.png "ray_trace_gain,0")

ray_trace_gain,0 

  * [![ray_trace_gain,2](/images/7/76/Ray_trace_gain2.png)](/index.php/File:Ray_trace_gain2.png "ray_trace_gain,2")

ray_trace_gain,2 

  * [![ray_trace_gain,8](/images/3/33/Ray_trace_gain8.png)](/index.php/File:Ray_trace_gain8.png "ray_trace_gain,8")

ray_trace_gain,8 

  * [![ray_trace_gain,20](/images/3/31/Ray_trace_gain20.png)](/index.php/File:Ray_trace_gain20.png "ray_trace_gain,20")

ray_trace_gain,20 




# Syntax
    
    
    # set it
    set ray_trace_gain, 7
    
    # you need to ray trace the image to see it
    ray
    

# See Also

[ray](/index.php/Ray "Ray")

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_trace_gain&oldid=5362](https://pymolwiki.org/index.php?title=Ray_trace_gain&oldid=5362)"


---

## Ray Trace Gain

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Description

A setting that alters the outline thickness when using ray_trace_mode=1 (full color with black outline). 

  * [![ray_trace_gain set to default](/images/1/13/Rtgd.png)](/index.php/File:Rtgd.png "ray_trace_gain set to default")

ray_trace_gain set to default 

  * [![ray_trace_gain set to 0](/images/c/c8/Rtg0.png)](/index.php/File:Rtg0.png "ray_trace_gain set to 0")

ray_trace_gain set to 0 

  * [![ray_trace_gain set to 10, and zoomed in, to show thickness of gain](/images/d/da/Rtg10.png)](/index.php/File:Rtg10.png "ray_trace_gain set to 10, and zoomed in, to show thickness of gain")

ray_trace_gain set to 10, and zoomed in, to show thickness of gain 




# Usage
    
    
    set ray_trace_gain, 0.0
    

# See Also

Retrieved from "[https://pymolwiki.org/index.php?title=Ray_Trace_Gain&oldid=8522](https://pymolwiki.org/index.php?title=Ray_Trace_Gain&oldid=8522)"


---

## Reflect

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set reflect controls the amount of light reflection and the effect directional light therefore has on shadows and the general lighting of the scene. 

## Syntax
    
    
    set reflect, <number>                          #default 0.5
    

## Example

  * [![reflect -1](/images/1/17/Reflect_minus1.png)](/index.php/File:Reflect_minus1.png "reflect -1")

reflect -1 

  * [![reflect 0](/images/e/e3/Refelect_0.png)](/index.php/File:Refelect_0.png "reflect 0")

reflect 0 

  * [![reflect 0.5 \(default\)](/images/4/43/Reflect_0.5.png)](/index.php/File:Reflect_0.5.png "reflect 0.5 \(default\)")

reflect 0.5 (default) 

  * [![reflect 1](/images/2/22/Refelect_1.png)](/index.php/File:Refelect_1.png "reflect 1")

reflect 1 




Retrieved from "[https://pymolwiki.org/index.php?title=Reflect&oldid=4250](https://pymolwiki.org/index.php?title=Reflect&oldid=4250)"


---

## Pdb retain ids

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

If set, atoms and residues are not renumbered when exporting imported pdb. (Must be set before import) 

## Syntax
    
    
    set pdb_retain_ids,[0,1]
    set pdb_retain_ids #default 1
    

## Example

Loaded and exported with pdb_retain_ids set to 0 
    
    
    ATOM     21  CH2 TRP A   2      38.350  22.030  11.200  1.00  0.00           C
    ATOM     22  HH2 TRP A   2      38.620  21.010  10.930  1.00  0.00           H
    ATOM     23  C   TRP A   2      35.010  27.370  14.910  1.00  0.00           C
    ATOM     24  O   TRP A   2      35.380  27.560  16.060  1.00  0.00           O
    ATOM     25  CH3 ACE A   1      33.160  24.100  12.400  1.00  0.00           C
    ATOM     26 HH31 ACE A   1      32.860  24.830  11.650  1.00  0.00           H
    ATOM     27 HH32 ACE A   1      33.800  23.390  11.880  1.00  0.00           H
    ATOM     28 HH33 ACE A   1      32.250  23.650  12.770  1.00  0.00           H
    ATOM     29  C   ACE A   1      33.910  24.800  13.510  1.00  0.00           C
    ATOM     30  O   ACE A   1      33.760  24.560  14.700  1.00  0.00           O
    TER      31      ACE A   1 
    ATOM     32  N   TRP A   3      33.910  27.930  14.400  1.00  0.00           N
    ATOM     33  H   TRP A   3      33.670  27.680  13.450  1.00  0.00           H
    ATOM     34  CA  TRP A   3      33.080  28.950  15.030  1.00  0.00           C
    

Loaded and exported with pdb_retain_ids set to 1 
    
    
    ATOM     27  CH2 TRP A   2      38.350  22.030  11.200  1.00  0.00           C
    ATOM     28  HH2 TRP A   2      38.620  21.010  10.930  1.00  0.00           H
    ATOM     29  C   TRP A   2      35.010  27.370  14.910  1.00  0.00           C
    ATOM     30  O   TRP A   2      35.380  27.560  16.060  1.00  0.00           O
    ATOM      1  CH3 ACE A   1      33.160  24.100  12.400  1.00  0.00           C
    ATOM      2 HH31 ACE A   1      32.860  24.830  11.650  1.00  0.00           H
    ATOM      3 HH32 ACE A   1      33.800  23.390  11.880  1.00  0.00           H
    ATOM      4 HH33 ACE A   1      32.250  23.650  12.770  1.00  0.00           H
    ATOM      5  C   ACE A   1      33.910  24.800  13.510  1.00  0.00           C
    ATOM      6  O   ACE A   1      33.760  24.560  14.700  1.00  0.00           O
    TER       7      ACE A   1 
    ATOM     31  N   TRP A   3      33.910  27.930  14.400  1.00  0.00           N
    ATOM     32  H   TRP A   3      33.670  27.680  13.450  1.00  0.00           H
    ATOM     33  CA  TRP A   3      33.080  28.950  15.030  1.00  0.00           C
    

Retrieved from "[https://pymolwiki.org/index.php?title=Pdb_retain_ids&oldid=4908](https://pymolwiki.org/index.php?title=Pdb_retain_ids&oldid=4908)"


---

## Retain order

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The retain_order setting keeps atoms in the original order, instead of [sorting](/index.php/Sort "Sort") them by [atom identifiers](/index.php?title=Atom_identifiers&action=edit&redlink=1 "Atom identifiers \(page does not exist\)"). The "original order" is defined by the [rank](/index.php?title=Rank&action=edit&redlink=1 "Rank \(page does not exist\)") property, so technically with **retain_order=on** , PyMOL performs a rank-assisted sort. 

Setting **retain_order** _on_ or _off_ triggers resorting of atoms. 

## Contents

  * 1 File Format Specifics
  * 2 Syntax
  * 3 Example
  * 4 See Also



## File Format Specifics

  * **vdb** : Since PyMOL 1.8.6, **retain_order** is automatically set for [VIPERdb](http://viperdb.scripps.edu/) files, because they don't have unique chain identifiers
  * **cif** : When **retain_order=on** , then PyMOL will skip loading of missing residues, because it depends on sorting



## Syntax
    
    
    set retain_order,[0,1]
    

## Example

Original PDB: 
    
    
    ATOM      1  CH3 ACE A   1      33.160  24.100  12.400  1.00  0.00
    ATOM      2 HH31 ACE A   1      32.860  24.830  11.650  1.00  0.00
    ATOM      3 HH32 ACE A   1      33.800  23.390  11.880  1.00  0.00
    ATOM      4 HH33 ACE A   1      32.250  23.650  12.770  1.00  0.00
    ATOM      5  C   ACE A   1      33.910  24.800  13.510  1.00  0.00
    ATOM      6  O   ACE A   1      33.760  24.560  14.700  1.00  0.00
    

Loaded and exported with retain_order set to 0 
    
    
    ATOM      1  C   ACE A   1      33.910  24.800  13.510  1.00  0.00           C
    ATOM      2  O   ACE A   1      33.760  24.560  14.700  1.00  0.00           O
    ATOM      3  CH3 ACE A   1      33.160  24.100  12.400  1.00  0.00           C
    ATOM      4 HH31 ACE A   1      32.860  24.830  11.650  1.00  0.00           H
    ATOM      5 HH32 ACE A   1      33.800  23.390  11.880  1.00  0.00           H
    ATOM      6 HH33 ACE A   1      32.250  23.650  12.770  1.00  0.00           H
    

Loaded and exported with retain_order set to 1 
    
    
    ATOM      1  CH3 ACE A   1      33.160  24.100  12.400  1.00  0.00           C
    ATOM      2 HH31 ACE A   1      32.860  24.830  11.650  1.00  0.00           H
    ATOM      3 HH32 ACE A   1      33.800  23.390  11.880  1.00  0.00           H
    ATOM      4 HH33 ACE A   1      32.250  23.650  12.770  1.00  0.00           H
    ATOM      5  C   ACE A   1      33.910  24.800  13.510  1.00  0.00           C
    ATOM      6  O   ACE A   1      33.760  24.560  14.700  1.00  0.00           O
    

## See Also

  * [sort](/index.php/Sort "Sort")
  * [pdb_hetatm_sort](/index.php?title=Pdb_hetatm_sort&action=edit&redlink=1 "Pdb hetatm sort \(page does not exist\)")
  * [pdb_insertions_go_first](/index.php?title=Pdb_insertions_go_first&action=edit&redlink=1 "Pdb insertions go first \(page does not exist\)")
  * [pdb_retain_ids](/index.php/Pdb_retain_ids "Pdb retain ids")
  * [alter](/index.php/Alter "Alter") (can modify **rank**)



Retrieved from "[https://pymolwiki.org/index.php?title=Retain_order&oldid=12537](https://pymolwiki.org/index.php?title=Retain_order&oldid=12537)"


---

## Ribbon color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 Advanced details



## Overview

Ribbon_color allows one to explicitly state the color to be applied to a ribbon object. 

## Syntax
    
    
    # set it to a color
    set ribbon_color, color
    
    # examples:
    # default auto-cycles the color for each new object
    set ribbon_color, marine
    
    # apply green to the ribbon only for object obj01
    set ribbon_color, green /obj01
    
    # apply orange to the ribbon only for chain B of object obj01; NB: not fully enabled until version 1.0
    set ribbon_color, orange /obj01//B
    

## Examples

  * [![ribbon_color green](/images/1/17/Ribbon_color_green.png)](/index.php/File:Ribbon_color_green.png "ribbon_color green")

ribbon_color green 

  * [![ribbon_color marine](/images/0/0d/Ribbon_color_marine.png)](/index.php/File:Ribbon_color_marine.png "ribbon_color marine")

ribbon_color marine 

  * [![two chains in same object with different colors](/images/f/f1/Ribbon_color_chains.png)](/index.php/File:Ribbon_color_chains.png "two chains in same object with different colors")

two chains in same object with different colors 




  


## Advanced details

Some notes on general [Set](/index.php/Set "Set") syntax (From PyMOL "help set"): 

DESCRIPTION 
    
    
      "set" changes one of the PyMOL state variables,
    
    

USAGE 
    
    
      set name, [,value [,object-or-selection [,state ]]]
    
      set name = value  # (DEPRECATED)
    
    

PYMOL API 
    
    
      cmd.set ( string name, string value=1,
                string selection=_, int state=0,_
                 int updates=1, quiet=1)
    
    

NOTES 
    
    
      The default behavior (with a blank selection) changes the global
      settings database.  If the selection is 'all', then the settings
      database in all individual objects will be changed.  Likewise, for
      a given object, if state is zero, then the object database will be
      modified.  Otherwise, the settings database for the indicated state
      within the object will be modified.
    
      If a selection is provided, then all objects in the selection will
      be affected.
    

Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_color&oldid=6230](https://pymolwiki.org/index.php?title=Ribbon_color&oldid=6230)"


---

## Ribbon sampling

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This setting changes the number of segments that makes up a given length of ribbon. The default value is ten; larger values (more segments) provide a smoother curve while requiring greater amounts of memory and slowing [performance](/index.php/Category:Performance "Category:Performance"). 

## Settings
    
    
    set ribbon_sampling, 20   # increases the number of ribbon segments
    

## Examples

Open the images to actually see the details! 

  * [![ribbon sampling 5](/images/b/b4/Ribbon_05.png)](/index.php/File:Ribbon_05.png "ribbon sampling 5")

ribbon sampling 5 

  * [![ribbon sampling 10 \(default\)](/images/1/1a/Ribbon_10.png)](/index.php/File:Ribbon_10.png "ribbon sampling 10 \(default\)")

ribbon sampling 10 (default) 

  * [![ribbon sampling 20](/images/4/41/Ribbon_20.png)](/index.php/File:Ribbon_20.png "ribbon sampling 20")

ribbon sampling 20 

  * [![ribbon sampling 30](/images/0/04/Ribbon_30.png)](/index.php/File:Ribbon_30.png "ribbon sampling 30")

ribbon sampling 30 




Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_sampling&oldid=5502](https://pymolwiki.org/index.php?title=Ribbon_sampling&oldid=5502)"


---

## Ribbon side chain helper

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Enabling Ribbon_side_chain_helper will hide the [Lines](/index.php/Lines "Lines") and [Sticks](/index.php/Sticks "Sticks") representation of backbone atoms when the [Ribbon](/index.php/Ribbon "Ribbon") and Lines/Sticks representations are shown. The effect is to help simplify the visualization of the backbone structure with sidechains shown. 

  * [![Ribbon sidechain helper off](/images/f/f6/Rsch0.png)](/index.php/File:Rsch0.png "Ribbon sidechain helper off")

Ribbon sidechain helper off 

  * [![Ribbon sidechain helper on, notice the backbone sticks are hidden](/images/7/79/Rsch1.png)](/index.php/File:Rsch1.png "Ribbon sidechain helper on, notice the backbone sticks are hidden")

Ribbon sidechain helper on, notice the backbone sticks are hidden 




# Usage
    
    
    # turn on
    set ribbon_side_chain_helper, on
    
    # turn off
    set ribbon_side_chain_helper, off
    

# See Also

[Cartoon_side_chain_helper](/index.php/Cartoon_side_chain_helper "Cartoon side chain helper")

Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_side_chain_helper&oldid=8518](https://pymolwiki.org/index.php?title=Ribbon_side_chain_helper&oldid=8518)"


---

## Ribbon smooth

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

This PyMOL setting determines smoothly how PyMOL draws ribbons. Turning _on_ the setting smooths out the ribbons and make the image look cleaner. The setting is by default **off** , and yet, when turned **on** shows absolutely no decreased [performance](/index.php/Category:Performance "Category:Performance"). 

## Usage
    
    
    #show ribbons, first
    hide; show ribbon
    
    # turn on smoothing
    set ribbon_smooth, 1
    
    # turn off smoothing
    set ribbon_smooth, 0
    

Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_smooth&oldid=5504](https://pymolwiki.org/index.php?title=Ribbon_smooth&oldid=5504)"


---

## Ribbon transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

In version v1.4.1 of PyMOL, we can now render transparent ribbons. This setting controls the level of transparency for the ribbons. 

  * [![Example of transparent ribbons.](/images/d/d6/Tp_ribbon.png)](/index.php/File:Tp_ribbon.png "Example of transparent ribbons.")

Example of transparent ribbons. 




Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_transparency&oldid=9049](https://pymolwiki.org/index.php?title=Ribbon_transparency&oldid=9049)"


---

## Ribbon Width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Examples
  * 3 Syntax
  * 4 See Also



## Overview

**ribbon_width** controls how fat or skinny PyMol draws the ribbons, when using **show ribbons**. 

## Examples

  * [![Ribbon width set to 1](/images/f/f0/Ribbon_width1.png)](/index.php/File:Ribbon_width1.png "Ribbon width set to 1")

Ribbon width set to 1 

  * [![Ribbon width set to 8](/images/5/51/Ribbon_width8.png)](/index.php/File:Ribbon_width8.png "Ribbon width set to 8")

Ribbon width set to 8 




## Syntax
    
    
    set ribbon_width,8
    

## See Also

[Ribbon](/index.php/Ribbon "Ribbon")

Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_Width&oldid=6281](https://pymolwiki.org/index.php?title=Ribbon_Width&oldid=6281)"


---

## Ribbon width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Ribbon_width sets the width of the ribbons in the [ribbon](/index.php/Ribbon "Ribbon") representation. Default is 3. 

## Syntax
    
    
    # set ribbon width
    set ribbon_width, val
    

## Examples
    
    
    # fat ribbons
    set ribbon_width, 8
    

  * [![ribbon_width, 0.5](/images/f/f2/Rw1.png)](/index.php/File:Rw1.png "ribbon_width, 0.5")

ribbon_width, 0.5 

  * [![ribbon_width, 2](/images/d/df/Rw2.png)](/index.php/File:Rw2.png "ribbon_width, 2")

ribbon_width, 2 

  * [![ribbon_width, 8](/images/9/9f/Rw3.png)](/index.php/File:Rw3.png "ribbon_width, 8")

ribbon_width, 8 




Retrieved from "[https://pymolwiki.org/index.php?title=Ribbon_width&oldid=8539](https://pymolwiki.org/index.php?title=Ribbon_width&oldid=8539)"


---

## Save settings

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Type  | Python Module   
---|---  
Download  | [scripts/save_settings.py](https://raw.githubusercontent.com/Pymol-Scripts/Pymol-script-repo/master/scripts/save_settings.py)  
Author(s)  | [Thomas Holder](/index.php/User:Speleo3 "User:Speleo3")  
License  | [BSD-2-Clause](http://opensource.org/licenses/BSD-2-Clause)  
This code has been put under version control in the project [Pymol-script-repo](/index.php/Git_intro "Git intro")  
  
save_settings is a prototype implementation to save all settings with non-default values to a file. 

By default settings will be stored to **~/.pymolrc-settings.py** which is automatically recognised and loaded by PyMOL on startup. You have to manually delete the file if you no more want those settings to be loaded. 

This answers a [feature request on sourceforge](https://sourceforge.net/tracker/?func=detail&aid=1009951&group_id=4546&atid=354546). 

## Usage
    
    
    save_settings [ filename ]
    

## See Also

  * [pymolrc](/index.php/Pymolrc "Pymolrc")



Retrieved from "[https://pymolwiki.org/index.php?title=Save_settings&oldid=13926](https://pymolwiki.org/index.php?title=Save_settings&oldid=13926)"


---

## Scene buttons

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting controls whether or not PyMOL displays Scene_buttons or openGL buttons in the lower left hand corner that allows interaction with scenes. 

[![](/images/d/db/Sb1.png)](/index.php/File:Sb1.png)

[](/index.php/File:Sb1.png "Enlarge")

frame

# Syntax
    
    
    # turn on scene buttons
    set scene_buttons, 1
    
    # turn off scene buttons
    set scene_buttons, 0
    

# Usage

Once these are enabled you can interact with them by using the mouse: 

  * left-button will select a scene
  * middle-button will rove over the scenes (to see this, load a bunch of scenes and then middle-click on the first and then drag down)
  * right-button will move a scene in the stack (drag-and-drop).



Retrieved from "[https://pymolwiki.org/index.php?title=Scene_buttons&oldid=6623](https://pymolwiki.org/index.php?title=Scene_buttons&oldid=6623)"


---

## Selection overlay

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

selection_overlay changes how PyMOL draws the selection indicator dots. If enabled, the indicators are drawn above the atoms, else underneath. 

  * [![Overlay off](/images/b/bc/Selection_overaly0.png)](/index.php/File:Selection_overaly0.png "Overlay off")

Overlay off 

  * [![Overlay on](/images/9/95/Selection_overaly1.png)](/index.php/File:Selection_overaly1.png "Overlay on")

Overlay on 




Retrieved from "[https://pymolwiki.org/index.php?title=Selection_overlay&oldid=9042](https://pymolwiki.org/index.php?title=Selection_overlay&oldid=9042)"


---

## Selection round points

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

This setting controls the shape of PyMOL's indicator dots. If set, it'll round the edges, if not rectangular. 

  * [![Round points off](/images/9/95/Selection_overaly1.png)](/index.php/File:Selection_overaly1.png "Round points off")

Round points off 

  * [![Round points on](/images/b/b9/Selection_roundpts1.png)](/index.php/File:Selection_roundpts1.png "Round points on")

Round points on 




Retrieved from "[https://pymolwiki.org/index.php?title=Selection_round_points&oldid=9043](https://pymolwiki.org/index.php?title=Selection_round_points&oldid=9043)"


---

## Selection visible only

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

If this setting is turned on, then PyMOL will only show indicator dots for visible objects. 

Retrieved from "[https://pymolwiki.org/index.php?title=Selection_visible_only&oldid=9044](https://pymolwiki.org/index.php?title=Selection_visible_only&oldid=9044)"


---

## Selection width

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Controls the minimum width of selection indicators in pixels 

  


## Syntax

set selection_width, 3 (default) 

## Examples

  * [![The selection indicators are in pink](/images/f/fa/Selection_3.png)](/index.php/File:Selection_3.png "The selection indicators are in pink")

The selection indicators are in pink 




Retrieved from "[https://pymolwiki.org/index.php?title=Selection_width&oldid=4283](https://pymolwiki.org/index.php?title=Selection_width&oldid=4283)"


---

## Seq view

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The **seq_view** command turns the sequence viewer on and off. The sequence viewer is a very handy tool, you can use it for example to select residues very easily. When the sequence viewer is turned on, you can select individual residues or multiple residues by selecting residues, using the mouse, on the sequence viewer. 

_Note:_ The sequence viewer, seems to decrease PyMOL's [performance](/index.php/Category:Performance "Category:Performance") when very many proteins are loaded into PyMOL. Therefore, if you have many proteins loaded, turn off the sequence viewer to increase performance if you don't need the viewer. You can remedy this, by turning on [texture_fonts](/index.php/Texture_fonts "Texture fonts"). Simply do, 
    
    
    set texture_fonts, 1
    

## Syntax
    
    
    # to turn the sequence viewer on.
    set seq_view, 1
    
    # to turn the sequence viewer off.
    set seq_view, 0
    

## Related settings

  * [seq_view_color](/index.php/Seq_view_color "Seq view color")
  * [seq_view_discrete_by_state](/index.php/Seq_view_discrete_by_state "Seq view discrete by state")
  * [seq_view_fill_char](/index.php/Seq_view_fill_char "Seq view fill char")
  * [seq_view_fill_color](/index.php?title=Seq_view_fill_color&action=edit&redlink=1 "Seq view fill color \(page does not exist\)")
  * [seq_view_format](/index.php/Seq_view_format "Seq view format")
  * [seq_view_label_color](/index.php/Seq_view_label_color "Seq view label color")
  * [seq_view_label_mode](/index.php/Seq_view_label_mode "Seq view label mode")
  * [seq_view_label_spacing](/index.php/Seq_view_label_spacing "Seq view label spacing")
  * [seq_view_label_start](/index.php?title=Seq_view_label_start&action=edit&redlink=1 "Seq view label start \(page does not exist\)")
  * [seq_view_location](/index.php/Seq_view_location "Seq view location")
  * [seq_view_overlay](/index.php/Seq_view_overlay "Seq view overlay")
  * [seq_view_unaligned_color](/index.php/Seq_view_unaligned_color "Seq view unaligned color")
  * [seq_view_unaligned_mode](/index.php/Seq_view_unaligned_mode "Seq view unaligned mode")



Retrieved from "[https://pymolwiki.org/index.php?title=Seq_view&oldid=6507](https://pymolwiki.org/index.php?title=Seq_view&oldid=6507)"


---

## Seq view fill char

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overivew

This set the character used for filler. The default is '-'. 

# Syntax
    
    
    # set the fill character to some character, c
    set seq_view_fill_char, c
    
    # for example, set the fill char to X
    set seq_view_fill_char, X
    

# See Also

Retrieved from "[https://pymolwiki.org/index.php?title=Seq_view_fill_char&oldid=6279](https://pymolwiki.org/index.php?title=Seq_view_fill_char&oldid=6279)"


---

## Seq view format

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

### OVERVIEW

The **seq_view_format** setting controls how PyMOL displays the sequence viewer. (To turn on the sequence viewer, type, **set seq_view, on**.) The available formats are currently: 

  * 0 = Display residues as single letter amino acid names
  * 1 = Display residues as triple letter amino acid names
  * 2 = Displays all atoms in each residue based on their atom name
  * 3 = Displays each peptide chain



### SYNTAX
    
    
    # usage
    set seq_view_format, number
    
    # default
    set seq_view_format, 0
    
    # triple letter amino acids
    set seq_view_format, 1
    

### SEE ALSO

[Seq_view](/index.php/Seq_view "Seq view")

Retrieved from "[https://pymolwiki.org/index.php?title=Seq_view_format&oldid=6293](https://pymolwiki.org/index.php?title=Seq_view_format&oldid=6293)"


---

## Seq view gap mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The seq_view_gap_mode setting controls if gap indicators are displayed in the [sequence viewer](/index.php/Category:Seqeuence_Viewer "Category:Seqeuence Viewer"). 

_New in PyMOL 2.3_

## Values

  * 0: no gap indicator display
  * 1: number of dashes equals number of missing residues (based on residue numbers) {default}
  * 2: one dash per gap (independent of size)



## Example
    
    
    fetch 2xwu, type=pdb, async=0
    set seq_view_gap_mode, 1
    set seq_view
    

Scroll sequence viewer to chain B residue 152, it should display 3 dashes. 

## See Also

  * [seq_view](/index.php/Seq_view "Seq view")



Retrieved from "[https://pymolwiki.org/index.php?title=Seq_view_gap_mode&oldid=12863](https://pymolwiki.org/index.php?title=Seq_view_gap_mode&oldid=12863)"


---

## Seq view location

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

seq_view_location affects the location of the sequence viewer. 

## Syntax

set seq_view_location, _integer_

set _integer_ to 0 to get the default (the viewer is on top of the internal window) 

set _integer_ to 1 or higher to locate the sequence viewer just above the command line. 

  


## See Also

  * [seq_view](/index.php/Seq_view "Seq view")
  * [seq_view_color](/index.php/Seq_view_color "Seq view color")
  * [seq_view_discrete_by_state](/index.php/Seq_view_discrete_by_state "Seq view discrete by state")
  * [seq_view_fill_char](/index.php/Seq_view_fill_char "Seq view fill char")
  * [seq_view_fill_color](/index.php?title=Seq_view_fill_color&action=edit&redlink=1 "Seq view fill color \(page does not exist\)")
  * [seq_view_format](/index.php/Seq_view_format "Seq view format")
  * [seq_view_label_color](/index.php/Seq_view_label_color "Seq view label color")
  * [seq_view_label_mode](/index.php/Seq_view_label_mode "Seq view label mode")
  * [seq_view_label_spacing](/index.php/Seq_view_label_spacing "Seq view label spacing")
  * [seq_view_label_start](/index.php?title=Seq_view_label_start&action=edit&redlink=1 "Seq view label start \(page does not exist\)")
  * [seq_view_overlay](/index.php/Seq_view_overlay "Seq view overlay")
  * [seq_view_unaligned_color](/index.php/Seq_view_unaligned_color "Seq view unaligned color")
  * [seq_view_unaligned_mode](/index.php/Seq_view_unaligned_mode "Seq view unaligned mode")



Retrieved from "[https://pymolwiki.org/index.php?title=Seq_view_location&oldid=6172](https://pymolwiki.org/index.php?title=Seq_view_location&oldid=6172)"


---

## Seq view overlay

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

### OVERVIEW

The **seq_view_overaly** setting controls whether or not the background of the sequence viewer is opaque or not, allowing access to the underlying molecules. 

  


### USAGE
    
    
    # make the sequence viewer background transparent
    set seq_view_overlay, 1
    
    # make it opaque again
    set seq_view_overlay, 0
    

### SEE ALSO

[Seq_view](/index.php/Seq_view "Seq view")

Retrieved from "[https://pymolwiki.org/index.php?title=Seq_view_overlay&oldid=6173](https://pymolwiki.org/index.php?title=Seq_view_overlay&oldid=6173)"


---

## Set

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

set is one of the most utilized commands. PyMOL representations, states, options, etc. are changed with **set**. Briefly, set changes one of the PyMOL state variables. Currently there are over _600_ [PyMOL settings](/index.php/Category:Settings "Category:Settings")! 

## Contents

  * 1 USAGE
  * 2 PYMOL API
  * 3 EXAMPLES
  * 4 NOTES
  * 5 SEE ALSO



# USAGE
    
    
    # set '''name''' to '''value'''
    set name, [,value [,object-or-selection [,state ]]]
    
    # alternative way to do the above.
    set name = value  # (DEPRECATED)
    

# PYMOL API
    
    
    cmd.set ( string name, 
        string value=1,
        string selection='',
        int state=0,
        int updates=1,
        quiet=1)
    

# EXAMPLES
    
    
    set surface_color, red
    
    set ray_trace_mode, 3
    
    set ribbon_width, 4
    
    # set the label size to 2Ang.
    set label_size, -2
    

# NOTES

The default behavior (with a blank selection) changes the global settings database. If the selection is 'all', then the settings database in all individual objects will be changed. Likewise, for a given object, if state is zero, then the object database will be modified. Otherwise, the settings database for the indicated state within the object will be modified. 

If a selection is provided, then all objects in the selection will be affected. 

# SEE ALSO

[Get](/index.php/Get "Get")

Retrieved from "[https://pymolwiki.org/index.php?title=Set&oldid=7582](https://pymolwiki.org/index.php?title=Set&oldid=7582)"


---

## Set bond

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Set_bond sets properties on _bonds_. This is usually some atom-connecting property like [Line_width](/index.php/Line_width "Line width"), [line_color](/index.php/Line_color "Line color"), [Stick_radius](/index.php/Stick_radius "Stick radius"). 

[![](/images/b/ba/Lw2.png)](/index.php/File:Lw2.png)

[](/index.php/File:Lw2.png "Enlarge")

Line_width was set to 10 for residues 1-10

# Syntax
    
    
    # set settingName to value for object or selection objSel
    set_bond settingName, value, objSel
    

# Examples
    
    
    # Example
    load $TUT/1hpv.pdb
    remove het
    as lines
    color blue
    set_bond line_with, 5, i. 1-10
    

# See Also

  * [Set](/index.php/Set "Set")
  * [Line_width](/index.php/Line_width "Line width")
  * [Stick_radius](/index.php/Stick_radius "Stick radius")



Retrieved from "[https://pymolwiki.org/index.php?title=Set_bond&oldid=7570](https://pymolwiki.org/index.php?title=Set_bond&oldid=7570)"


---

## Roving byres

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Description

**roving_byres** determines if [roving_detail](/index.php/Roving_detail "Roving detail") selection is by residue or not. 

## Syntax
    
    
    # to turn on roving_detail byres
    set roving_byres, true
    
    # to turn off roving_detail byres
    set roving_byres, false
    

  


## Related settings

  * roving_byres
  * [roving_cartoon](/index.php?title=Roving_cartoon&action=edit&redlink=1 "Roving cartoon \(page does not exist\)")
  * [roving_delay](/index.php/Roving_delay "Roving delay")
  * [roving_detail](/index.php/Roving_detail "Roving detail")
  * [roving_isomesh](/index.php?title=Roving_isomesh&action=edit&redlink=1 "Roving isomesh \(page does not exist\)")
  * [roving_isosurface](/index.php?title=Roving_isosurface&action=edit&redlink=1 "Roving isosurface \(page does not exist\)")
  * [roving_labels](/index.php?title=Roving_labels&action=edit&redlink=1 "Roving labels \(page does not exist\)")
  * [roving_lines](/index.php?title=Roving_lines&action=edit&redlink=1 "Roving lines \(page does not exist\)")
  * [roving_map1_level](/index.php?title=Roving_map1_level&action=edit&redlink=1 "Roving map1 level \(page does not exist\)")
  * [roving_map1_name](/index.php?title=Roving_map1_name&action=edit&redlink=1 "Roving map1 name \(page does not exist\)")
  * [roving_map2_level](/index.php?title=Roving_map2_level&action=edit&redlink=1 "Roving map2 level \(page does not exist\)")
  * [roving_map2_name](/index.php?title=Roving_map2_name&action=edit&redlink=1 "Roving map2 name \(page does not exist\)")
  * [roving_map3_level](/index.php?title=Roving_map3_level&action=edit&redlink=1 "Roving map3 level \(page does not exist\)")
  * [roving_map3_name](/index.php?title=Roving_map3_name&action=edit&redlink=1 "Roving map3 name \(page does not exist\)")
  * [roving_nb_spheres](/index.php/Roving_nb_spheres "Roving nb spheres")
  * [roving_nonbonded](/index.php?title=Roving_nonbonded&action=edit&redlink=1 "Roving nonbonded \(page does not exist\)")
  * [roving_origin](/index.php?title=Roving_origin&action=edit&redlink=1 "Roving origin \(page does not exist\)")
  * [roving_origin_z](/index.php?title=Roving_origin_z&action=edit&redlink=1 "Roving origin z \(page does not exist\)")
  * [roving_origin_z_cushion](/index.php?title=Roving_origin_z_cushion&action=edit&redlink=1 "Roving origin z cushion \(page does not exist\)")
  * [roving_polar_contacts ](/index.php?title=Roving_polar_contacts&action=edit&redlink=1 "Roving polar contacts \(page does not exist\)")
  * [roving_polar_cutoff](/index.php?title=Roving_polar_cutoff&action=edit&redlink=1 "Roving polar cutoff \(page does not exist\)")
  * [roving_ribbon](/index.php?title=Roving_ribbon&action=edit&redlink=1 "Roving ribbon \(page does not exist\)")
  * [roving_selection](/index.php?title=Roving_selection&action=edit&redlink=1 "Roving selection \(page does not exist\)")
  * [roving_spheres](/index.php?title=Roving_spheres&action=edit&redlink=1 "Roving spheres \(page does not exist\)")
  * [roving_sticks](/index.php?title=Roving_sticks&action=edit&redlink=1 "Roving sticks \(page does not exist\)")



Retrieved from "[https://pymolwiki.org/index.php?title=Roving_byres&oldid=7266](https://pymolwiki.org/index.php?title=Roving_byres&oldid=7266)"


---

## Roving delay

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Description
  * 2 Syntax
  * 3 Usage
  * 4 Related settings



## Description

**roving_delay** determines the time before updating the scene. 

## Syntax
    
    
    # change roving_delay
    set roving_delay, number # number > 0
    

  


## Usage

If you are on a slow computer it might prove useful to set this value to e.g. 2 (seconds). This prevents the scene from getting updated before you reach your place of interest. 

## Related settings

  * [roving_byres](/index.php/Roving_byres "Roving byres")
  * [roving_cartoon](/index.php?title=Roving_cartoon&action=edit&redlink=1 "Roving cartoon \(page does not exist\)")
  * roving_delay
  * [roving_detail](/index.php/Roving_detail "Roving detail")
  * [roving_isomesh](/index.php?title=Roving_isomesh&action=edit&redlink=1 "Roving isomesh \(page does not exist\)")
  * [roving_isosurface](/index.php?title=Roving_isosurface&action=edit&redlink=1 "Roving isosurface \(page does not exist\)")
  * [roving_labels](/index.php?title=Roving_labels&action=edit&redlink=1 "Roving labels \(page does not exist\)")
  * [roving_lines](/index.php?title=Roving_lines&action=edit&redlink=1 "Roving lines \(page does not exist\)")
  * [roving_map1_level](/index.php?title=Roving_map1_level&action=edit&redlink=1 "Roving map1 level \(page does not exist\)")
  * [roving_map1_name](/index.php?title=Roving_map1_name&action=edit&redlink=1 "Roving map1 name \(page does not exist\)")
  * [roving_map2_level](/index.php?title=Roving_map2_level&action=edit&redlink=1 "Roving map2 level \(page does not exist\)")
  * [roving_map2_name](/index.php?title=Roving_map2_name&action=edit&redlink=1 "Roving map2 name \(page does not exist\)")
  * [roving_map3_level](/index.php?title=Roving_map3_level&action=edit&redlink=1 "Roving map3 level \(page does not exist\)")
  * [roving_map3_name](/index.php?title=Roving_map3_name&action=edit&redlink=1 "Roving map3 name \(page does not exist\)")
  * [roving_nb_spheres](/index.php/Roving_nb_spheres "Roving nb spheres")
  * [roving_nonbonded](/index.php?title=Roving_nonbonded&action=edit&redlink=1 "Roving nonbonded \(page does not exist\)")
  * [roving_origin](/index.php?title=Roving_origin&action=edit&redlink=1 "Roving origin \(page does not exist\)")
  * [roving_origin_z](/index.php?title=Roving_origin_z&action=edit&redlink=1 "Roving origin z \(page does not exist\)")
  * [roving_origin_z_cushion](/index.php?title=Roving_origin_z_cushion&action=edit&redlink=1 "Roving origin z cushion \(page does not exist\)")
  * [roving_polar_contacts ](/index.php?title=Roving_polar_contacts&action=edit&redlink=1 "Roving polar contacts \(page does not exist\)")
  * [roving_polar_cutoff](/index.php?title=Roving_polar_cutoff&action=edit&redlink=1 "Roving polar cutoff \(page does not exist\)")
  * [roving_ribbon](/index.php?title=Roving_ribbon&action=edit&redlink=1 "Roving ribbon \(page does not exist\)")
  * [roving_selection](/index.php?title=Roving_selection&action=edit&redlink=1 "Roving selection \(page does not exist\)")
  * [roving_spheres](/index.php?title=Roving_spheres&action=edit&redlink=1 "Roving spheres \(page does not exist\)")
  * [roving_sticks](/index.php?title=Roving_sticks&action=edit&redlink=1 "Roving sticks \(page does not exist\)")



Retrieved from "[https://pymolwiki.org/index.php?title=Roving_delay&oldid=8879](https://pymolwiki.org/index.php?title=Roving_delay&oldid=8879)"


---

## Roving detail

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Description
  * 2 Syntax
  * 3 Disclaimer
  * 4 Related settings



## Description

**Roving_detail** shows details of the environment centered on the screen. 

  


## Syntax
    
    
    # to turn on roving detail
    set roving_detail, true
    
    # to turn off roving detail
    set roving_detail, false
    

  


## Disclaimer

When turning on roving_detail you immediately get the effect on your screen effectively changing all your representations. 

## Related settings

  * [roving_byres](/index.php/Roving_byres "Roving byres")
  * [roving_cartoon](/index.php?title=Roving_cartoon&action=edit&redlink=1 "Roving cartoon \(page does not exist\)")
  * [roving_delay](/index.php/Roving_delay "Roving delay")
  * roving_detail
  * [roving_isomesh](/index.php?title=Roving_isomesh&action=edit&redlink=1 "Roving isomesh \(page does not exist\)")
  * [roving_isosurface](/index.php?title=Roving_isosurface&action=edit&redlink=1 "Roving isosurface \(page does not exist\)")
  * [roving_labels](/index.php?title=Roving_labels&action=edit&redlink=1 "Roving labels \(page does not exist\)")
  * [roving_lines](/index.php?title=Roving_lines&action=edit&redlink=1 "Roving lines \(page does not exist\)")
  * [roving_map1_level](/index.php?title=Roving_map1_level&action=edit&redlink=1 "Roving map1 level \(page does not exist\)")
  * [roving_map1_name](/index.php?title=Roving_map1_name&action=edit&redlink=1 "Roving map1 name \(page does not exist\)")
  * [roving_map2_level](/index.php?title=Roving_map2_level&action=edit&redlink=1 "Roving map2 level \(page does not exist\)")
  * [roving_map2_name](/index.php?title=Roving_map2_name&action=edit&redlink=1 "Roving map2 name \(page does not exist\)")
  * [roving_map3_level](/index.php?title=Roving_map3_level&action=edit&redlink=1 "Roving map3 level \(page does not exist\)")
  * [roving_map3_name](/index.php?title=Roving_map3_name&action=edit&redlink=1 "Roving map3 name \(page does not exist\)")
  * [roving_nb_spheres](/index.php/Roving_nb_spheres "Roving nb spheres")
  * [roving_nonbonded](/index.php?title=Roving_nonbonded&action=edit&redlink=1 "Roving nonbonded \(page does not exist\)")
  * [roving_origin](/index.php?title=Roving_origin&action=edit&redlink=1 "Roving origin \(page does not exist\)")
  * [roving_origin_z](/index.php?title=Roving_origin_z&action=edit&redlink=1 "Roving origin z \(page does not exist\)")
  * [roving_origin_z_cushion](/index.php?title=Roving_origin_z_cushion&action=edit&redlink=1 "Roving origin z cushion \(page does not exist\)")
  * [roving_polar_contacts ](/index.php?title=Roving_polar_contacts&action=edit&redlink=1 "Roving polar contacts \(page does not exist\)")
  * [roving_polar_cutoff](/index.php?title=Roving_polar_cutoff&action=edit&redlink=1 "Roving polar cutoff \(page does not exist\)")
  * [roving_ribbon](/index.php?title=Roving_ribbon&action=edit&redlink=1 "Roving ribbon \(page does not exist\)")
  * [roving_selection](/index.php?title=Roving_selection&action=edit&redlink=1 "Roving selection \(page does not exist\)")
  * [roving_spheres](/index.php?title=Roving_spheres&action=edit&redlink=1 "Roving spheres \(page does not exist\)")
  * [roving_sticks](/index.php?title=Roving_sticks&action=edit&redlink=1 "Roving sticks \(page does not exist\)")



Retrieved from "[https://pymolwiki.org/index.php?title=Roving_detail&oldid=7263](https://pymolwiki.org/index.php?title=Roving_detail&oldid=7263)"


---

## Roving nb spheres

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Description

**roving_nb_spheres** determines the radius within spheres are shown for non-bonded atoms (e.g. waters). 

  


## Syntax
    
    
    # set radius for non-bonded spheres while roving_detail is on
    set roving_nb_spheres, typeNumber # typeNumber = [1...], 0 to disable
    

## Related settings

  * [roving_byres](/index.php/Roving_byres "Roving byres")
  * [roving_cartoon](/index.php?title=Roving_cartoon&action=edit&redlink=1 "Roving cartoon \(page does not exist\)")
  * [roving_delay](/index.php/Roving_delay "Roving delay")
  * [roving_detail](/index.php/Roving_detail "Roving detail")
  * [roving_isomesh](/index.php?title=Roving_isomesh&action=edit&redlink=1 "Roving isomesh \(page does not exist\)")
  * [roving_isosurface](/index.php?title=Roving_isosurface&action=edit&redlink=1 "Roving isosurface \(page does not exist\)")
  * [roving_labels](/index.php?title=Roving_labels&action=edit&redlink=1 "Roving labels \(page does not exist\)")
  * [roving_lines](/index.php?title=Roving_lines&action=edit&redlink=1 "Roving lines \(page does not exist\)")
  * [roving_map1_level](/index.php?title=Roving_map1_level&action=edit&redlink=1 "Roving map1 level \(page does not exist\)")
  * [roving_map1_name](/index.php?title=Roving_map1_name&action=edit&redlink=1 "Roving map1 name \(page does not exist\)")
  * [roving_map2_level](/index.php?title=Roving_map2_level&action=edit&redlink=1 "Roving map2 level \(page does not exist\)")
  * [roving_map2_name](/index.php?title=Roving_map2_name&action=edit&redlink=1 "Roving map2 name \(page does not exist\)")
  * [roving_map3_level](/index.php?title=Roving_map3_level&action=edit&redlink=1 "Roving map3 level \(page does not exist\)")
  * [roving_map3_name](/index.php?title=Roving_map3_name&action=edit&redlink=1 "Roving map3 name \(page does not exist\)")
  * roving_nb_spheres
  * [roving_nonbonded](/index.php?title=Roving_nonbonded&action=edit&redlink=1 "Roving nonbonded \(page does not exist\)")
  * [roving_origin](/index.php?title=Roving_origin&action=edit&redlink=1 "Roving origin \(page does not exist\)")
  * [roving_origin_z](/index.php?title=Roving_origin_z&action=edit&redlink=1 "Roving origin z \(page does not exist\)")
  * [roving_origin_z_cushion](/index.php?title=Roving_origin_z_cushion&action=edit&redlink=1 "Roving origin z cushion \(page does not exist\)")
  * [roving_polar_contacts ](/index.php?title=Roving_polar_contacts&action=edit&redlink=1 "Roving polar contacts \(page does not exist\)")
  * [roving_polar_cutoff](/index.php?title=Roving_polar_cutoff&action=edit&redlink=1 "Roving polar cutoff \(page does not exist\)")
  * [roving_ribbon](/index.php?title=Roving_ribbon&action=edit&redlink=1 "Roving ribbon \(page does not exist\)")
  * [roving_selection](/index.php?title=Roving_selection&action=edit&redlink=1 "Roving selection \(page does not exist\)")
  * [roving_spheres](/index.php?title=Roving_spheres&action=edit&redlink=1 "Roving spheres \(page does not exist\)")
  * [roving_sticks](/index.php?title=Roving_sticks&action=edit&redlink=1 "Roving sticks \(page does not exist\)")



Retrieved from "[https://pymolwiki.org/index.php?title=Roving_nb_spheres&oldid=7265](https://pymolwiki.org/index.php?title=Roving_nb_spheres&oldid=7265)"


---

## Stereo Mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

[![](/images/9/93/Anaglyph1.png)](/index.php/File:Anaglyph1.png)

[](/index.php/File:Anaglyph1.png "Enlarge")

Example of Anaglyph 3D in PyMOL

[![](/images/6/63/Ana4_focal_ray.png)](/index.php/File:Ana4_focal_ray.png)

[](/index.php/File:Ana4_focal_ray.png "Enlarge")

Example of a ray traced, focal blurred, anaglyph image.

[![](/images/8/82/3D-B-DNA.png)](/index.php/File:3D-B-DNA.png)

[](/index.php/File:3D-B-DNA.png "Enlarge")

3D B-DNA Example

[![](/images/2/21/Amyloid.png)](/index.php/File:Amyloid.png)

[](/index.php/File:Amyloid.png "Enlarge")

Amyloid beta example

The [stereo_mode](/index.php/Stereo_mode "Stereo mode") setting sets the type of stereo mode, if the [stereo](/index.php/Stereo "Stereo") setting is enabled. 

_You can also control both settings with the[stereo](/index.php/Stereo "Stereo") command, which is more convenient!_

## Contents

  * 1 Syntax
  * 2 Supported Stereo Modes
  * 3 Notes
    * 3.1 Anaglyph Color Quality and Ghosting
  * 4 See Also



## Syntax
    
    
    set stereo_mode, integer
    

Valid values for the **integer** argument are listed in the following table. 

## Supported Stereo Modes

Corresponding keyword arguments (instead of numeric values) can be passed to the [stereo](/index.php/Stereo "Stereo") command. 

value | description   
---|---  
1 | **quad-buffered**  
2 | **cross-eyed**  
3 | **walleye**  
4 | **geowall**  
5 | **sidebyside**  
6 | **stencil by row** , Zalman   
7 | **stencil by col**  
8 | **stencil checkerboard**  
9 | **stencil custom** for developers   
10 | **anaglyph** (requires green/magenta glasses)   
11 | **dynamic polarization**  
12 | **clone dynamic**  
  
## Notes

### Anaglyph Color Quality and Ghosting

To test the quality of your glasses and coloring, you can test for "ghosting". Choose a part of the structure where the channels for the left and right eye are nicely separated in space. Hold one lens over the part of the structure and see you see one image or if there is a faint second image for the other channel. Then hold the other other lens over the image and see if there is a faint second image. 

## See Also

  * [Stereo](/index.php/Stereo "Stereo")
  * [Stereo_angle](/index.php/Stereo_angle "Stereo angle")



Retrieved from "[https://pymolwiki.org/index.php?title=Stereo_Mode&oldid=11520](https://pymolwiki.org/index.php?title=Stereo_Mode&oldid=11520)"


---

## Shininess

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

Set the shininess of objects in PyMol. 

## Examples

  * [![Shininess, 30](/images/3/3d/Shiny30.png)](/index.php/File:Shiny30.png "Shininess, 30")

Shininess, 30 

  * [![Shininess, 55](/images/c/cd/Shiny55.png)](/index.php/File:Shiny55.png "Shininess, 55")

Shininess, 55 

  * [![Shininess, 100](/images/1/1e/Shiny100.png)](/index.php/File:Shiny100.png "Shininess, 100")

Shininess, 100 




Retrieved from "[https://pymolwiki.org/index.php?title=Shininess&oldid=4278](https://pymolwiki.org/index.php?title=Shininess&oldid=4278)"


---

## Solvent radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This defines the solvent radius. Changing the solvent radius in PyMOL affects how PyMOL interprets [surfaces](/index.php/Surface_solvent "Surface solvent"), [surface area](/index.php/Displaying_Biochemical_Properties#Surface-Related "Displaying Biochemical Properties"). 

The default solvent radius is 1.4 Angstroms. 

# Syntax
    
    
    # set the solvent radius
    set solvent_radius, float
    
    # for example, set it to 2.0 Angstroms
    set solvent_radius, 2.0
    

# See Also

[Surface_solvent](/index.php/Surface_solvent "Surface solvent")

Retrieved from "[https://pymolwiki.org/index.php?title=Solvent_radius&oldid=6195](https://pymolwiki.org/index.php?title=Solvent_radius&oldid=6195)"


---

## Spec direct

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting controls how PyMOL renders the direct specular reflection from a representation. 

This image shows an increasing Spec_direct setting from 0 to 1.0 

[![](/images/7/76/Spec_direct.gif)](/index.php/File:Spec_direct.gif)

[](/index.php/File:Spec_direct.gif "Enlarge")

Increasing the spec_direct setting from 0 to 1.0.

# Syntax
    
    
    # set spec direct to real-valued number
    set spec_direct, float
    
    # turn up the spec_direct
    set spec_direct, 0.75
    

# See Also

[Specular Reflections Category](/index.php/Category:Specular_Reflections "Category:Specular Reflections")

Retrieved from "[https://pymolwiki.org/index.php?title=Spec_direct&oldid=6199](https://pymolwiki.org/index.php?title=Spec_direct&oldid=6199)"


---

## Spec power

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting controls the power of specular reflections. 

# Syntax
    
    
    # set the power
    set spec_power, float
    
    # turn up the reflections
    set spec_power, 200
    

# See Also

[Spec_reflect](/index.php/Spec_reflect "Spec reflect")

[Category:Specular_Reflections](/index.php/Category:Specular_Reflections "Category:Specular Reflections")

Retrieved from "[https://pymolwiki.org/index.php?title=Spec_power&oldid=6207](https://pymolwiki.org/index.php?title=Spec_power&oldid=6207)"


---

## Specular

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This controls whether or not specular reflections are shown unrendered, in the GUI. Turning this off only makes a slight improvement in speed. 

  * [![Specular set to 0](/images/6/64/Spec0.png)](/index.php/File:Spec0.png "Specular set to 0")

Specular set to 0 

  * [![Specular set to 1](/images/2/2c/Spec01.png)](/index.php/File:Spec01.png "Specular set to 1")

Specular set to 1 

  * [![Specular set to 5](/images/e/e1/Spec5.png)](/index.php/File:Spec5.png "Specular set to 5")

Specular set to 5 

  * [![Specular set to 10](/images/0/0a/Spec10.png)](/index.php/File:Spec10.png "Specular set to 10")

Specular set to 10 




Notice in the images above that the setting is controlling the amount of directly reflected light and not the shininess of the reflection. 

# Syntax
    
    
    # turn the setting on or off
    set specular, boolean
    
    # turn it off
    set specular, off
    
    # turn it on
    set specular, on
    

# See Also

[Pages on Specular Reflections](/index.php/Category:Specular_Reflections "Category:Specular Reflections")

Retrieved from "[https://pymolwiki.org/index.php?title=Specular&oldid=7156](https://pymolwiki.org/index.php?title=Specular&oldid=7156)"


---

## Specular intensity

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This controls how PyMOL displays specular reflections in the GUI (unrendered). This is analogous to [Spec_power](/index.php/Spec_power "Spec power") for ray traced images. 

Values range form [0, 200+]. 

# Syntax
    
    
    # set the specular intensity to some real valued number
    set specular_intensity, float
    
    # for example, set it to 200
    set specular_intensity, 200.0
    

# See Also

[Spec_power](/index.php/Spec_power "Spec power")

Retrieved from "[https://pymolwiki.org/index.php?title=Specular_intensity&oldid=6209](https://pymolwiki.org/index.php?title=Specular_intensity&oldid=6209)"


---

## Spec reflect

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



## Overview

This setting changes how PyMol handles specular reflections. Essentially, this setting, when combined with [spec_power](/index.php/Spec_power "Spec power") adjusts how sharp or diffuse the reflection is, giving you shiny looking surface or duller surfaces. 

## Syntax
    
    
    set spec_reflect, 2
    set spec_power, 1500
    

## Examples

  * [![Specular Reflection Example. Settings are spec_reflect=2.0; spec_power=1500](/images/b/b0/Spec.png)](/index.php/File:Spec.png "Specular Reflection Example. Settings are spec_reflect=2.0; spec_power=1500")

Specular Reflection Example. Settings are spec_reflect=2.0; spec_power=1500 

  * [![Specular Reflection Example. Settings are spec_reflect=0.0; spec_power=1500](/images/3/3d/Spec1.png)](/index.php/File:Spec1.png "Specular Reflection Example. Settings are spec_reflect=0.0; spec_power=1500")

Specular Reflection Example. Settings are spec_reflect=0.0; spec_power=1500 




## See Also

  * [spec_power](/index.php/Spec_power "Spec power")
  * [About Specular Reflections](http://en.wikipedia.org/wiki/Specular_reflection)



Retrieved from "[https://pymolwiki.org/index.php?title=Spec_reflect&oldid=8897](https://pymolwiki.org/index.php?title=Spec_reflect&oldid=8897)"


---

## Sphere color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

set sphere_color allows to color spheres independently from the rest of the representations. 

## Syntax
    
    
    set sphere_color, theColor
    

where _theColor_ can be : 

  * any usual colors (blue, yellow, grey50,...)
  * number-coded colors (1:black, 2:blue, 3:greenish, ...)
  * special code -1 to revert to original chameleon setting (_set sphere_color,-1_)



## Related settings

[cartoon_color](/index.php/Cartoon_color "Cartoon color")

Retrieved from "[https://pymolwiki.org/index.php?title=Sphere_color&oldid=6300](https://pymolwiki.org/index.php?title=Sphere_color&oldid=6300)"


---

## Sphere quality

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



## Overview

**sphere_quality** controls the rendering quality of sphere objects. This setting only affects sphere rendering when not using shaders. 

## Syntax
    
    
    # the default value is 1
    set sphere_quality, <integer>
    

Larger values of <integer> result in higher quality sphere rendering. Values >1 may result in poor [performance](/index.php/Category:Performance "Category:Performance") during real-time rotation or translation. 

**Note** : Selecting values larger than 2 with **stick_ball** = 1 (enabled) causes PyMol to crash in the Windows version. 

## Examples

Open the images to see rendering details. 

  * [![sphere_quality 0](/images/0/05/Sphere_quality_0.png)](/index.php/File:Sphere_quality_0.png "sphere_quality 0")

sphere_quality 0 

  * [![sphere_quality 1 \(default\)](/images/6/6e/Sphere_quality_1.png)](/index.php/File:Sphere_quality_1.png "sphere_quality 1 \(default\)")

sphere_quality 1 (default) 

  * [![sphere_quality 2](/images/2/2f/Sphere_quality_2.png)](/index.php/File:Sphere_quality_2.png "sphere_quality 2")

sphere_quality 2 




# See Also

[cgo_sphere_quality](/index.php?title=Cgo_sphere_quality&action=edit&redlink=1 "Cgo sphere quality \(page does not exist\)")

Retrieved from "[https://pymolwiki.org/index.php?title=Sphere_quality&oldid=13559](https://pymolwiki.org/index.php?title=Sphere_quality&oldid=13559)"


---

## Sphere scale

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 Related settings



## Overview

This setting affects the apparent radius of spheres in the sphere representation. Default scale is set to 1.0. 

## Syntax
    
    
    # set the sphere scale to size for selection.
    cmd.set ("sphere_scale", size=1, selection='', state=0, updates=1, log=0, quiet=1)
    # generally it's simpler to use the console form
    set sphere_scale, size, selection
    # you can print the current value for sphere_scale
    get sphere_scale
    

_size_ can be any floating point number, _selection_ is the name of a selection. 

## Examples

Using 0.25 gives a nice balls&sticks representation with both lines and spheres turned on. 
    
    
    set_bond stick_color, white, (all), (all)
    set_bond stick_radius, 0.14, (all), (all)
    set sphere_scale, 0.25, (all)
    show sticks
    show spheres
    

**set sphere_scale** by itself will revert to default. Here you'll get a simple VDW rapresentation. 
    
    
    set_bond stick_color, blue, (all), (all)
    set_bond stick_radius, 0.3, (all), (all)
    set sphere_transparency, 0.3
    set sphere_scale
    

## Related settings

  * [sphere_color](/index.php/Sphere_color "Sphere color")
  * [sphere_mode](/index.php/Sphere_mode "Sphere mode")
  * [sphere_point_max_size](/index.php/Sphere_point_max_size "Sphere point max size")
  * [sphere_point_size](/index.php/Sphere_point_size "Sphere point size")
  * [sphere_quality](/index.php/Sphere_quality "Sphere quality")
  * [sphere_solvent](/index.php/Sphere_solvent "Sphere solvent")
  * [sphere_transparency](/index.php/Sphere_transparency "Sphere transparency")



Retrieved from "[https://pymolwiki.org/index.php?title=Sphere_scale&oldid=8294](https://pymolwiki.org/index.php?title=Sphere_scale&oldid=8294)"


---

## Sphere transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

set sphere_transparency is used to adjust the transparency of spheres! 
    
    
    set sphere_transparency, 0.5
    

or 
    
    
    set sphere_transparency=0.5, selection
    

Where 1.0 is invisible and 0.0 completely solid 

# Examples

  * [![Example of transparent spheres](/images/c/c2/Sphere_transparency_ex1.png)](/index.php/File:Sphere_transparency_ex1.png "Example of transparent spheres")

Example of transparent spheres 

  * [![Example of transparent spheres](/images/6/65/Sphere_transparency_ex3.png)](/index.php/File:Sphere_transparency_ex3.png "Example of transparent spheres")

Example of transparent spheres 




These images were made through the following script: 
    
    
    fetch 1ifr; color wheat; hide; show spheres;
    color red, i. 506-509;
    color marine, byres all within 5 of color red;
    set sphere_transparency,0.5,color marine;
    

Retrieved from "[https://pymolwiki.org/index.php?title=Sphere_transparency&oldid=6306](https://pymolwiki.org/index.php?title=Sphere_transparency&oldid=6306)"


---

## Sphere mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The **sphere_mode** setting controls how PyMOL draws spheres in the OpenGL interface. It does not affect rendering, using [ray](/index.php/Ray "Ray"). 

Sphere mode set to **5** , which looks beautiful, requires have [OpenGL Shaders Installed](/index.php/Spheres#Enabling_Shaders "Spheres"). 

## Syntax
    
    
    set sphere_mode, VAL
    

where 

VAL = [1..5]

## Example

  * [![Sphere mode unset](/images/9/97/Sphere_mode_off.png)](/index.php/File:Sphere_mode_off.png "Sphere mode unset")

Sphere mode unset 

  * [![Sphere mode 1](/images/2/22/Sphere_mode_1.png)](/index.php/File:Sphere_mode_1.png "Sphere mode 1")

Sphere mode 1 

  * [![Sphere mode 2](/images/4/47/Sphere_mode_2.png)](/index.php/File:Sphere_mode_2.png "Sphere mode 2")

Sphere mode 2 

  * [![Sphere mode 3](/images/9/91/Sphere_mode_3.png)](/index.php/File:Sphere_mode_3.png "Sphere mode 3")

Sphere mode 3 

  * [![Sphere mode 4](/images/4/42/Sphere_mode_4.png)](/index.php/File:Sphere_mode_4.png "Sphere mode 4")

Sphere mode 4 

  * [![Sphere mode 5](/images/3/33/Sphere_mode_5.png)](/index.php/File:Sphere_mode_5.png "Sphere mode 5")

Sphere mode 5 




Retrieved from "[https://pymolwiki.org/index.php?title=Sphere_mode&oldid=6301](https://pymolwiki.org/index.php?title=Sphere_mode&oldid=6301)"


---

## State counter mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

This setting controls whether or not the object-specific state counter is shown and how it's shown: 

Values: 

**0, off**

    

    off

**1, on**

    

    on, fractional mode (eg. X/Y, means this object is in state X out of a possible Y)

**2**

    

    on, state mode, (eg. X, means state X)

Retrieved from "[https://pymolwiki.org/index.php?title=State_counter_mode&oldid=9045](https://pymolwiki.org/index.php?title=State_counter_mode&oldid=9045)"


---

## Static singletons

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

If set, static_singletons will always show single-state objects in the viewer, regardless of 'scene' or 'global' state. Turning this off, will hide single-state objects, when the global state counter is not 1. 

Retrieved from "[https://pymolwiki.org/index.php?title=Static_singletons&oldid=9037](https://pymolwiki.org/index.php?title=Static_singletons&oldid=9037)"


---

## Stereo

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The stereo command sets the current stereo mode. Stereo mode is a convenient way to "see" 3D with two images from slightly different angles. 

_There are corresponding**stereo** and [stereo_mode](/index.php/Stereo_mode "Stereo mode") settings which are controlled by the **stereo** command, so you don't need to set them directly._

## Usage
    
    
    stereo [ toggle ]
    

Valid values for the **toggle** argument are: on, swap, off, quadbuffer, crosseye, walleye, geowall, sidebyside, byrow, bycolumn, checkerboard, custom, anaglyph, dynamic, clonedynamic (see also [stereo_mode](/index.php/Stereo_mode "Stereo mode")) 

[![](/images/b/b8/Stereo_on.png)](/index.php/File:Stereo_on.png)

[](/index.php/File:Stereo_on.png "Enlarge")

Example of 1ESR shown in cross-eyed stereo

## Example
    
    
    fetch 1ESR, async=0
    as cartoon
    set cartoon_smooth_loops
    spectrum
    bg white
    stereo crosseye
    

## See Also

  * [stereo_mode](/index.php/Stereo_mode "Stereo mode")
  * [stereo_angle](/index.php/Stereo_angle "Stereo angle")
  * [stereo_shift](/index.php?title=Stereo_shift&action=edit&redlink=1 "Stereo shift \(page does not exist\)")
  * [stereo_ray](/index.php/Stereo_ray "Stereo ray")



Retrieved from "[https://pymolwiki.org/index.php?title=Stereo&oldid=10665](https://pymolwiki.org/index.php?title=Stereo&oldid=10665)"


---

## Stereo mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

[![](/images/9/93/Anaglyph1.png)](/index.php/File:Anaglyph1.png)

[](/index.php/File:Anaglyph1.png "Enlarge")

Example of Anaglyph 3D in PyMOL

[![](/images/6/63/Ana4_focal_ray.png)](/index.php/File:Ana4_focal_ray.png)

[](/index.php/File:Ana4_focal_ray.png "Enlarge")

Example of a ray traced, focal blurred, anaglyph image.

[![](/images/8/82/3D-B-DNA.png)](/index.php/File:3D-B-DNA.png)

[](/index.php/File:3D-B-DNA.png "Enlarge")

3D B-DNA Example

[![](/images/2/21/Amyloid.png)](/index.php/File:Amyloid.png)

[](/index.php/File:Amyloid.png "Enlarge")

Amyloid beta example

The [stereo_mode](/index.php/Stereo_mode "Stereo mode") setting sets the type of stereo mode, if the [stereo](/index.php/Stereo "Stereo") setting is enabled. 

_You can also control both settings with the[stereo](/index.php/Stereo "Stereo") command, which is more convenient!_

## Contents

  * 1 Syntax
  * 2 Supported Stereo Modes
  * 3 Notes
    * 3.1 Anaglyph Color Quality and Ghosting
  * 4 See Also



## Syntax
    
    
    set stereo_mode, integer
    

Valid values for the **integer** argument are listed in the following table. 

## Supported Stereo Modes

Corresponding keyword arguments (instead of numeric values) can be passed to the [stereo](/index.php/Stereo "Stereo") command. 

value | description   
---|---  
1 | **quad-buffered**  
2 | **cross-eyed**  
3 | **walleye**  
4 | **geowall**  
5 | **sidebyside**  
6 | **stencil by row** , Zalman   
7 | **stencil by col**  
8 | **stencil checkerboard**  
9 | **stencil custom** for developers   
10 | **anaglyph** (requires green/magenta glasses)   
11 | **dynamic polarization**  
12 | **clone dynamic**  
  
## Notes

### Anaglyph Color Quality and Ghosting

To test the quality of your glasses and coloring, you can test for "ghosting". Choose a part of the structure where the channels for the left and right eye are nicely separated in space. Hold one lens over the part of the structure and see you see one image or if there is a faint second image for the other channel. Then hold the other other lens over the image and see if there is a faint second image. 

## See Also

  * [Stereo](/index.php/Stereo "Stereo")
  * [Stereo_angle](/index.php/Stereo_angle "Stereo angle")



Retrieved from "[https://pymolwiki.org/index.php?title=Stereo_Mode&oldid=11520](https://pymolwiki.org/index.php?title=Stereo_Mode&oldid=11520)"


---

## Stereo angle

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

**Stereo_angle** sets the angle between the right and left frames for cross-eyed stereo. See [stereo](/index.php/Stereo "Stereo"). 

## Syntax
    
    
    set stereo_angle, float
    

## See Also

[Stereo](/index.php/Stereo "Stereo") [Stereo_mode](/index.php/Stereo_mode "Stereo mode")

Retrieved from "[https://pymolwiki.org/index.php?title=Stereo_angle&oldid=4535](https://pymolwiki.org/index.php?title=Stereo_angle&oldid=4535)"


---

## Stick ball

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

**This setting is deprecated in v1.5 and later as it is always enabled.**

The setting "stick_ball" controls whether bonded atoms are shown simply as joined sticks (set stick_ball, off) or as traditional "ball-and-stick" representation (set stick_ball, on). Note that simply setting stick_ball on will result in balls with the same radius as the sticks and so will appear only slightly different (the joins will be smoother). 

## Settings
    
    
    set stick_ball, on   # displays atoms as balls joined by sticks
    set stick_ball, off  # displays only connected sticks
    
    set stick_ball_ratio, 1.7 # change the radius of the balls
    

## Examples

Open the images to actually see the details! 

  * [![stick_ball, off](/images/c/cc/Stick_ball_off.png)](/index.php/File:Stick_ball_off.png "stick_ball, off")

stick_ball, off 

  * [![stick_ball, on](/images/0/00/Stick_ball_on.png)](/index.php/File:Stick_ball_on.png "stick_ball, on")

stick_ball, on 

  * [![stick_ball "on" with stick_ball_ratio at 1.5](/images/4/4d/Stick_ball_ratio_1.5.png)](/index.php/File:Stick_ball_ratio_1.5.png "stick_ball "on" with stick_ball_ratio at 1.5")

stick_ball "on" with stick_ball_ratio at 1.5 




Retrieved from "[https://pymolwiki.org/index.php?title=Stick_ball&oldid=10909](https://pymolwiki.org/index.php?title=Stick_ball&oldid=10909)"


---

## Stick ball ratio

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Settings
  * 3 Related
  * 4 Examples



## Overview

The setting "stick_ball_ratio" controls the relative ratio between the radius of sticks connecting bonded atoms and the radius of the atom spheres. Note that simply setting "stick_ball, on" will result in balls with the same radius as the sticks and so will appear only slightly different (the joins will be smoother). Changing the stick_ball_ratio without setting "stick_ball, on" will -- obviously -- have no apparent effect. 

## Settings
    
    
    set stick_ball_ratio, 1.5
    

## Related
    
    
    set stick_ball, on   # displays atoms as balls joined by sticks
    set stick_ball, off  # displays only connected sticks
    

## Examples

Open the images to actually see the details! 

  * [![stick_ball, off](/images/c/cc/Stick_ball_off.png)](/index.php/File:Stick_ball_off.png "stick_ball, off")

stick_ball, off 

  * [![stick_ball, on](/images/0/00/Stick_ball_on.png)](/index.php/File:Stick_ball_on.png "stick_ball, on")

stick_ball, on 

  * [![stick_ball "on" with stick_ball_ratio at 1.5](/images/4/4d/Stick_ball_ratio_1.5.png)](/index.php/File:Stick_ball_ratio_1.5.png "stick_ball "on" with stick_ball_ratio at 1.5")

stick_ball "on" with stick_ball_ratio at 1.5 




Retrieved from "[https://pymolwiki.org/index.php?title=Stick_ball_ratio&oldid=5270](https://pymolwiki.org/index.php?title=Stick_ball_ratio&oldid=5270)"


---

## Stick h scale

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

[![](/images/b/b6/Stick_h_scale.png)](/index.php/File:Stick_h_scale.png)

[](/index.php/File:Stick_h_scale.png "Enlarge")

Tryptophan fragment in stick representation with stick_h_scale=0.4

The stick_h_scale setting scales the [stick](/index.php/Sticks "Sticks") radius of bonds to hydrogens. 

_Default value changed in 1.8.4_

## Example
    
    
    fragment trp
    as sticks
    set stick_h_scale, 0.4
    
    # before 1.8.4 this was only in effect with a negative stick_radius
    set stick_radius, -0.25
    

## See Also

  * [sticks](/index.php/Sticks "Sticks") representation
  * [stick_radius](/index.php/Stick_radius "Stick radius")



Retrieved from "[https://pymolwiki.org/index.php?title=Stick_h_scale&oldid=12448](https://pymolwiki.org/index.php?title=Stick_h_scale&oldid=12448)"


---

## Stick nub

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

The stick_nub setting controls the height of the cones at the end of the sticks in stick representation. This only affects the simple representation: this setting doesn't have any effect for ray tracing. 

  * [![Default stick_nub setting.](/images/2/28/Sticknub_default.png)](/index.php/File:Sticknub_default.png "Default stick_nub setting.")

Default stick_nub setting. 

  * [![Stubby nubs--stick nub set to 0.](/images/5/55/Sticknub_0.png)](/index.php/File:Sticknub_0.png "Stubby nubs--stick nub set to 0.")

Stubby nubs--stick nub set to 0. 

  * [![Nub spears: stick_nub=2.0](/images/f/f4/Sticknub_2.png)](/index.php/File:Sticknub_2.png "Nub spears: stick_nub=2.0")

Nub spears: stick_nub=2.0 




## Syntax
    
    
    set stick_nub, ''float''
    

Where _float_ is a floating point number. The default value is 0.7. 

## Example
    
    
    set stick_nub, 0
    

Retrieved from "[https://pymolwiki.org/index.php?title=Stick_nub&oldid=5281](https://pymolwiki.org/index.php?title=Stick_nub&oldid=5281)"


---

## Stick radius

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 If the above commands do not work
  * 4 Related settings



## Overview

This setting affects the radius of sticks in the sticks representation. Default scale is set to 0.25. 

In newer versions of PyMOL, one may set the Stick_radius on a per-bond basis. So, you can set for example, the radius of only selected bonds if you want. This is done through the [Set_bond](/index.php/Set_bond "Set bond") command. 

  * [![stick_radius set to 0.05](/images/1/18/Stick_rad_0.05.png)](/index.php/File:Stick_rad_0.05.png "stick_radius set to 0.05")

stick_radius set to 0.05 

  * [![stick_radius set to the default 0.25](/images/d/de/Stick_radius_default.png)](/index.php/File:Stick_radius_default.png "stick_radius set to the default 0.25")

stick_radius set to the default 0.25 

  * [![stick_radius set to 0.85](/images/4/41/Stick_rad_0.85.png)](/index.php/File:Stick_rad_0.85.png "stick_radius set to 0.85")

stick_radius set to 0.85 




## Syntax
    
    
    set_bond stick_radius, ''size'', selection
    

where, 

  * _size_ can be any float number. Using 0.25 (default value) is usually appropriate for most representations, although 0.15 migh be preferred for comparing closely related structures, e.g., conformers.



_Note:_
    
    
    set_bond stick_radius
    

by itself will revert to 1.00. 

## If the above commands do not work

You can do something like below 
    
    
    To set on the entire object
    
    set stick_radius=0.12
    
    OR
    
    create myObj, <selection>
    
    Ex : create myObj, hetatm
    
    set stick_radius,0.2,myObj
    

## Related settings

  * [Set_bond](/index.php/Set_bond "Set bond")
  * [sphere_scale](/index.php/Sphere_scale "Sphere scale")
  * [stick_ball_ratio](/index.php/Stick_ball_ratio "Stick ball ratio")



Retrieved from "[https://pymolwiki.org/index.php?title=Stick_radius&oldid=7204](https://pymolwiki.org/index.php?title=Stick_radius&oldid=7204)"


---

## Stick transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Usage
  * 3 Examples
  * 4 Note



## Overview

The setting "stick_transparency" allows one to set the degree of transparency for stick objects, independent of all other objects. Allowable values range from 0 (fully opaque) to 1 (fully transparent, i.e. invisible). 

## Usage
    
    
    set stick_transparency, F, objectname
    set_bond stick_transparency, F, selection
    

where **F** is a floating point number in the range _[0.0 - 1.0]_ , and **objectname** is the object to apply the changes to. If ObjectName is omitted, then the transparency of all stick representations will be changed. 

To apply this setting to a selection, use "set_bond" syntax 
    
    
    set_bond stick_transparency, 0.7, */n+c+ca+o
    

For the value of _F_ =1.0 sticks will be completely transparent/invisible and for _F_ =0.0 sticks are solid/opaque. 

## Examples
    
    
    set stick_transparency, 0.50   # Makes sticks 50-percent transparent
    

Open the images to actually see the details! 

  * [![sticks with no transparency](/images/b/bf/Stick_trans_zero.png)](/index.php/File:Stick_trans_zero.png "sticks with no transparency")

sticks with no transparency 

  * [![sticks with 0.5 transparency](/images/9/99/Stick_trans_50.png)](/index.php/File:Stick_trans_50.png "sticks with 0.5 transparency")

sticks with 0.5 transparency 

  * [![sticks with 0.9 transparency](/images/e/e0/Stick_trans_90.png)](/index.php/File:Stick_trans_90.png "sticks with 0.9 transparency")

sticks with 0.9 transparency 




## Note

Stick transparency works best with "unilayer" transparency (Setting menu > transparency > Unilayer) rather than "multilayer", which leads to odd artifacts where the sticks join. 

  * [![sticks with 0.5 transparency and multilayer transparency](/images/a/a0/Stick_trans_50-multi.png)](/index.php/File:Stick_trans_50-multi.png "sticks with 0.5 transparency and multilayer transparency")

sticks with 0.5 transparency and multilayer transparency 




Retrieved from "[https://pymolwiki.org/index.php?title=Stick_transparency&oldid=12792](https://pymolwiki.org/index.php?title=Stick_transparency&oldid=12792)"


---

## Stick color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Example
  * 4 See Also



## Overview

_Stick_color_ controls the color of sticks for a selection or object. For example, one could color all atoms by their element as transparent spheres, but for a nice underlying structure, have all the sticks shown in marine. 

## Syntax
    
    
    set stick_color, color
    

## Example

As noted in the above text -- transparent spheres, where sticks are all one color: 

  * [![Example of color sticks one color](/images/3/33/Stick_color.png)](/index.php/File:Stick_color.png "Example of color sticks one color")

Example of color sticks one color 




## See Also

[Color](/index.php/Color "Color")

Retrieved from "[https://pymolwiki.org/index.php?title=Stick_color&oldid=5272](https://pymolwiki.org/index.php?title=Stick_color&oldid=5272)"


---

## Surface cavity mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Surface_cavity_mode changes how PyMOL represents cavities. 

  * [![surface_cavity_mode set to 0](/images/0/03/Scm0.png)](/index.php/File:Scm0.png "surface_cavity_mode set to 0")

surface_cavity_mode set to 0 

  * [![surface_cavity_mode set to 1](/images/6/6f/Scm1.png)](/index.php/File:Scm1.png "surface_cavity_mode set to 1")

surface_cavity_mode set to 1 




# Details

There are a few modes for displaying surface cavities. Normal (0), inverted (1) and more. 

# Usage
    
    
    # set the mode to (integer)
    set surface_cavity_mode, (integer)
    

# Examples
    
    
    # set the mode to inverted
    set surface_cavity_mode, 1
    

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_cavity_mode&oldid=7919](https://pymolwiki.org/index.php?title=Surface_cavity_mode&oldid=7919)"


---

## Surface color

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



# Overview

**surface_color** controls the color of surfaces as drawn in PyMOL. 

  * [![Usual surface coloring](/images/b/bc/Surface_color_ex1.png)](/index.php/File:Surface_color_ex1.png "Usual surface coloring")

Usual surface coloring 

  * [![Color the entire surface "marine".](/images/d/d3/Surface_color_ex2.png)](/index.php/File:Surface_color_ex2.png "Color the entire surface "marine".")

Color the entire surface "marine". 




# Syntax
    
    
    # color the surface
    set surface_color, (color), (selection)
    

# Examples
    
    
    # color the surface white
    set surface_color, white, *
    
    # return surface coloring to the default scheme
    set surface_color, default, *
    

# See Also

[Color](/index.php/Color "Color"), [Color_Values](/index.php/Color_Values "Color Values")

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_color&oldid=12614](https://pymolwiki.org/index.php?title=Surface_color&oldid=12614)"


---

## Surface mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Sets how PyMOL draws the surface. The default, surface_mode=0 does not include the heteroatoms within the surface; setting it to 1, does include them. See the example images. 

## Usage
    
    
    set surface_mode, int
    

where _int_ is one of the following values: 

  * 0: Default mode, surfacing with respect to [flags](/index.php/Flag "Flag").
  * 1: Surface everything, including HET and hydrogens
  * 2: Surface only heavy atoms
  * 3: Surface only visible
  * 4: Surface visible and heavy



## Examples

  * [![surface_mode set to 0, the default. The galactose \(blue\) is not considered part of the surface.](/images/6/6c/Sm0.png)](/index.php/File:Sm0.png "surface_mode set to 0, the default. The galactose \(blue\) is not considered part of the surface.")

surface_mode set to 0, the default. The galactose (blue) is not considered part of the surface. 

  * [![surface_mode set to 1 -- now including heteroatoms. The galactose and all heteroatoms \(blue\) are now considered part of the surface and colored blue.](/images/4/46/Sm1.png)](/index.php/File:Sm1.png "surface_mode set to 1 -- now including heteroatoms. The galactose and all heteroatoms \(blue\) are now considered part of the surface and colored blue.")

surface_mode set to 1 -- now including heteroatoms. The galactose and all heteroatoms (blue) are now considered part of the surface and colored blue. 



    
    
    # make the above images, or something like them
    fetch 2v72, async=0
    color wheat
    color marine, het
    show surface
    
    # default
    set surface_mode, 0
    
    # non default
    set surface_mode, 1
    

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_mode&oldid=11521](https://pymolwiki.org/index.php?title=Surface_mode&oldid=11521)"


---

## Surface proximity

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When showing the molecular [surface](/index.php/Surface "Surface") of a subset of atoms (not the entire molecule), then the surface_proximity setting controls whether triangles are shown which are between an included and an excluded atom. 

## Values

  * **surface_proximity=on** : (default) Show all triangles which have at least one vertex belonging to the atom selection. In this case colors from hidden atoms can leak into the open surface edge.
  * **surface_proximity=off** : Show only triangles where all three vertices belong to the atom selection.



## Example

This example demonstrates the color "leaking" from atoms which don't belong to the surface selection. 
    
    
    fetch 5hpr, async=0
    color red
    color yellow, chain B
    show surface, chain B
    set surface_proximity, off
    

[![Surface proximity.png](/images/a/ae/Surface_proximity.png)](/index.php/File:Surface_proximity.png)

## See Also

  * [surface_smooth_edges](/index.php/Surface_smooth_edges "Surface smooth edges")
  * [surface](/index.php/Surface "Surface") representation



Retrieved from "[https://pymolwiki.org/index.php?title=Surface_proximity&oldid=12556](https://pymolwiki.org/index.php?title=Surface_proximity&oldid=12556)"


---

## Surface quality

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This controls how well PyMOL draws surfaces. Lower values, like 0, are rough surface approximations. These low values are good for speed--especially for larger surfaces. For rendering of publication quality photos, and truer representations of the biological surface, set the value higher--to something like 2, 3 or 4. In practice typical values are 1, 2 and 3. 

This value, I believe, changes how finely the polygons are sampled for surface representation. Settings of 2 or 3 are far more computationally intensive to compute and show than are the default and low values. For example, using a value of 3 took my computer about 2 minutes just to prepare the surface for showing in the PyMOL GUI (this does not include any ray-tracing or rendering). Lastly, ray tracing surfaces with high quality settings takes much longer. 

  * [![Surface_quality set to its default](/images/d/d4/Sqd.png)](/index.php/File:Sqd.png "Surface_quality set to its default")

Surface_quality set to its default 

  * [![Surface_quality turned down](/images/3/3b/Sq0.png)](/index.php/File:Sq0.png "Surface_quality turned down")

Surface_quality turned down 

  * [![Surface_quality turned up high \(2\).](/images/f/fd/Sq2.png)](/index.php/File:Sq2.png "Surface_quality turned up high \(2\).")

Surface_quality turned up high (2). 




# Syntax
    
    
    # set to some positive integer
    set surface_quality, int
    
    # for example turn up the quality a bit
    set surface_quality, 2
    

### SEE ALSO

[Huge surfaces](/index.php/Huge_surfaces "Huge surfaces")

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_quality&oldid=12464](https://pymolwiki.org/index.php?title=Surface_quality&oldid=12464)"


---

## Surface ramp above mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The **surface_ramp_above_mode** setting is for coloring [molecular surfaces](/index.php/Surface "Surface") by electrostatic potential, using a [color ramp](/index.php/Ramp_new "Ramp new"). If **on** , the surface is colored by the potential on the [solvent accessible surface](/index.php/Surface_solvent "Surface solvent"), which is one solvent radius above the surface. 

Coloring with _surface_solvent=0_ and _surface_ramp_above_mode=0_ is less meaningful, since a solvent or ligand atom never reaches the molecular surface of the protein (only the accessible surface). Furthermore, _surface_solvent=1_ with _surface_ramp_above_mode=1_ isn't meaningful either. 

## Illustration

[![Surface ramp above mode.jpg](/images/c/cf/Surface_ramp_above_mode.jpg)](/index.php/File:Surface_ramp_above_mode.jpg)

## Example
    
    
    load molecule.pdb
    load electrostaticmap.dx
    
    ramp_new espramp, electrostaticmap, [ -3, 0, 3]
    
    show surface
    
    set surface_color, espramp
    set surface_ramp_above_mode
    

## See Also

  * [surface_solvent](/index.php/Surface_solvent "Surface solvent")
  * [ramp_new](/index.php/Ramp_new "Ramp new")



Retrieved from "[https://pymolwiki.org/index.php?title=Surface_ramp_above_mode&oldid=11536](https://pymolwiki.org/index.php?title=Surface_ramp_above_mode&oldid=11536)"


---

## Surface smooth edges

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When the surface representation is only partially visible (e.g. ligand binding site surface), the **surface_smooth_edges** setting controls appearance of the surface edges. 

_New in Incentive PyMOL 1.8.6_

## Values

  * **surface_smooth_edges=off** : (default) disables surface edge smoothing.
  * **surface_smooth_edges=on** : enables surface edge smoothing. Consecutive vertices of the surface edge are averaged resulting in smoother-looking edge.



## Examples
    
    
    set surface_smooth_edges, off 
    

[![Surface smooth off example.png](/images/2/21/Surface_smooth_off_example.png)](/index.php/File:Surface_smooth_off_example.png)
    
    
    set surface_smooth_edges, on
    

[![Surface smooth on example.png](/images/5/59/Surface_smooth_on_example.png)](/index.php/File:Surface_smooth_on_example.png)
    
    
    set surface_smooth_edges, off 
    

[![Surface smooth edges off.png](/images/5/5a/Surface_smooth_edges_off.png)](/index.php/File:Surface_smooth_edges_off.png)
    
    
    set surface_smooth_edges, on 
    

[![Surface smooth edges on.png](/images/0/09/Surface_smooth_edges_on.png)](/index.php/File:Surface_smooth_edges_on.png)

## See Also

  * [surface](/index.php/Surface "Surface") representation
  * [surface_proximity](/index.php/Surface_proximity "Surface proximity")
  * [surface_carve_selection](/index.php/Surface_carve_selection "Surface carve selection")



Retrieved from "[https://pymolwiki.org/index.php?title=Surface_smooth_edges&oldid=12557](https://pymolwiki.org/index.php?title=Surface_smooth_edges&oldid=12557)"


---

## Surface solvent

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

The **surface_solvent** setting controls whether the solvent accessible (on) or the solvent excluded (off) [surface](/index.php/Surface "Surface") is shown. 
    
    
    set surface_solvent, off
    

[![Surface solvent OFF.jpg](/images/6/69/Surface_solvent_OFF.jpg)](/index.php/File:Surface_solvent_OFF.jpg)
    
    
    set surface_solvent, on
    

[![Surface solvent ON.jpg](/images/3/38/Surface_solvent_ON.jpg)](/index.php/File:Surface_solvent_ON.jpg)

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_solvent&oldid=9431](https://pymolwiki.org/index.php?title=Surface_solvent&oldid=9431)"


---

## Surface carve normal cutoff

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



# Overview

**surface_carve_normal_cutoff** will hide all triangles in a selection not facing the carved selection. 

# Syntax
    
    
    # set the surface carving cutoff
    set surface_carve_normal_cutoff, value
    

# Examples
    
    
    # set the carve normal cutoff
    set surface_carve_normal_cutoff, -0.1
    

# See Also

[surface_carve_selection](/index.php/Surface_carve_selection "Surface carve selection") [surface_carve_cutoff](/index.php/Surface_carve_cutoff "Surface carve cutoff")

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_carve_normal_cutoff&oldid=5042](https://pymolwiki.org/index.php?title=Surface_carve_normal_cutoff&oldid=5042)"


---

## Surface carve selection

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

[![](/images/f/f6/Surface_carve_selection.gif)](/index.php/File:Surface_carve_selection.gif)

[](/index.php/File:Surface_carve_selection.gif "Enlarge")

Animation with a moving pseudoatom as the surface_carve_selection

With the **surface_carve_selection** and **surface_carve_cutoff** settings, the visibility of a molecular [surface](/index.php/Surface "Surface") can be limited to the proximity of an atom selection. 

Unlike when [showing](/index.php/Show "Show") [surface](/index.php/Surface "Surface") representation only for a subset of atoms, the visibility is controlled on the vertex (surface point) level, not the atom level. 

The value of **surface_carve_selection** must be a [named selection](/index.php/Named_Atom_Selections "Named Atom Selections") (created with [select](/index.php/Select "Select")), it **cannot** be a selection expression (e.g. "chain A" won't work). 

## Example

Visibility control on the vertex level. Hide surface points which are more than 5 Angstrom away from **carvesele** : 
    
    
    fetch 1rx1, async=0
    as cartoon
    select carvesele, organic
    set surface_carve_selection, carvesele
    set surface_carve_cutoff, 5
    show surface
    show sticks, carvesele
    

Compare to visibility control on the atom level. Hide surface for atoms which are more than 5 Angstrom away from **carvesele**. 
    
    
    fetch 1rx1, async=0
    as cartoon
    select carvesele, organic
    show surface, all within 5 of carvesele
    show sticks, carvesele
    

## See Also

  * [surface_proximity](/index.php/Surface_proximity "Surface proximity")
  * [surface_smooth_edges](/index.php/Surface_smooth_edges "Surface smooth edges")



Retrieved from "[https://pymolwiki.org/index.php?title=Surface_carve_selection&oldid=12533](https://pymolwiki.org/index.php?title=Surface_carve_selection&oldid=12533)"


---

## Surface type

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Syntax
  * 3 Examples
  * 4 See Also



# Overview

**Surface_type** sets the type (dots, filled triangles, empty triangles) of the surface that PyMOL will render. 

# Syntax
    
    
    set surface_type, typeNumber  # typeNumber = [1..3]
    

# Examples
    
    
    set surface_type, 2
    

  * [![surface_type, 0 # usual](/images/1/18/St0.png)](/index.php/File:St0.png "surface_type, 0 # usual")

surface_type, 0 # usual 

  * [![surface_type, 1 # dots](/images/2/2f/St1.png)](/index.php/File:St1.png "surface_type, 1 # dots")

surface_type, 1 # dots 

  * [![surface_type, 2 # triangles](/images/1/19/St2.png)](/index.php/File:St2.png "surface_type, 2 # triangles")

surface_type, 2 # triangles 

  * [![surface_type, 3 # is this the same as surface_type 0?](/images/f/f2/St3.png)](/index.php/File:St3.png "surface_type, 3 # is this the same as surface_type 0?")

surface_type, 3 # is this the same as surface_type 0? 




# See Also

[surface](/index.php/Surface "Surface"), [as](/index.php/As "As")

Retrieved from "[https://pymolwiki.org/index.php?title=Surface_type&oldid=5049](https://pymolwiki.org/index.php?title=Surface_type&oldid=5049)"


---

## Suspend updates

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

When set, Suspend_updates stops PyMOL from updating the GUI whenever changes are made. This can be used to mask multiple operations from the user to make it appear as if only one operation occurred. See examples. 

# Example
    
    
    load $TUT/1hpv.pdb
    set suspend_updates
    # remove the waters
    remove resn HOH
    # the waters are still visible
    # until we unset suspend_updates
    #
    # Now, to get UI responsiveness type
    #
    # unset suspend_updates
    

### PYMOL API
    
    
    cmd.set('suspend_updates', 'on')
    cmd.set('suspend_updates', 'off')
    

### SEE ALSO

[Show](/index.php/Show "Show"), [Hide](/index.php/Hide "Hide"), [Enable](/index.php/Enable "Enable"), [Disable](/index.php/Disable "Disable")

Retrieved from "[https://pymolwiki.org/index.php?title=Suspend_updates&oldid=11566](https://pymolwiki.org/index.php?title=Suspend_updates&oldid=11566)"


---

## Texture fonts

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overivew

Not sure exactly what this does. The reason this page exists is to let the users know that this setting can affect rendering performance in PyMOL. 

To test how this affects your PyMOL. Turn on the frames per second (FPS) meter. Load a complex scene turn on the sequence viewer ([Seq_View](/index.php?title=Seq_View&action=edit&redlink=1 "Seq View \(page does not exist\)")) and then rotate the molecule. While doing this look at the maximum frame rate. Then turn on Texture_fonts and rotate the scene while watching the frame rate. Here are my example numbers: 

    

    **texture_fonts off** max = 142 FPS
    **texture_fonts on** max = 363 FPS

That's a huge difference. 

# Syntax
    
    
    # turn on texture fonts
    set texture_fonts, 1
    

Retrieved from "[https://pymolwiki.org/index.php?title=Texture_fonts&oldid=6901](https://pymolwiki.org/index.php?title=Texture_fonts&oldid=6901)"


---

## Transparency

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Usage
  * 3 Examples
    * 3.1 Whole Surface
    * 3.2 Selected Surface Elements



## Overview

**Transparency** is used to adjust the transparency of **[Surfaces](/index.php/Surface "Surface")** and **[Slices](/index.php/Slice "Slice")**. (For other transparencies in PyMOL, see [Cartoon Transparency](/index.php/Cartoon_transparency "Cartoon transparency"), [Sphere Transparency](/index.php/Sphere_transparency "Sphere transparency"), and [Stick Transparency](/index.php/Stick_transparency "Stick transparency"). 

## Usage
    
    
    set transparency, F, selection
    

where **F** is a floating point number in the range _[0.0 - 1.0]_ , where **selection** is the selected surface to apply the change to (for examples, see below). 

For the value of _F_ , 1.0 will be an invisible and 0.0 a completely solid surface. 

  


## Examples

### Whole Surface

Change the transparency of the whole surface to 50%. 
    
    
    # show all surfaces with 50% transparency.
    set transparency, 0.5
    

  


  * [![Image showing partial surface transparencies](/images/2/2e/Surf065.png)](/index.php/File:Surf065.png "Image showing partial surface transparencies")

Image showing partial surface transparencies 

  * [![Image showing 100% of the surface with 65% transparency.](/images/d/d6/Surfall_065.png)](/index.php/File:Surfall_065.png "Image showing 100% of the surface with 65% transparency.")

Image showing 100% of the surface with 65% transparency. 




### Selected Surface Elements

Simple example showing how to do partial surface transparency. This allows different selections to have different transparencies on the same object (or also on different objects). 
    
    
    # load a random protein
    fetch 1rty
    
    # set the partial transparency for the selected residues
    set transparency, 0.65, i. 1-100
    

  * [![Different transparency settings on different objects.](/images/f/f5/Diffy_transp.png)](/index.php/File:Diffy_transp.png "Different transparency settings on different objects.")

Different transparency settings on different objects. 

  * [![Different transparency settings on the same object.](/images/f/f3/Diffy_transp2.png)](/index.php/File:Diffy_transp2.png "Different transparency settings on the same object.")

Different transparency settings on the same object. 




Retrieved from "[https://pymolwiki.org/index.php?title=Transparency&oldid=10608](https://pymolwiki.org/index.php?title=Transparency&oldid=10608)"


---

## Transparency mode

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Transparency_mode controls how PyMOL handles transparent objects 

# Example Images

  * [![Unilayer transparency](/images/f/f9/Unilayer.png)](/index.php/File:Unilayer.png "Unilayer transparency")

Unilayer transparency 

  * [![Multilayer transparency. Notice that PyMOL adds other transparent objects beneath each other](/images/5/55/Multilayer.png)](/index.php/File:Multilayer.png "Multilayer transparency. Notice that PyMOL adds other transparent objects beneath each other")

Multilayer transparency. Notice that PyMOL adds other transparent objects beneath each other 




# Settings

  * **1** : Multi-layer transparency
  * **2** : Single-layer transparency (default)
  * **3** : Multi-layer [weighted blended order independent transparency](http://jcgt.org/published/0002/02/09/) for real-time rendering



Retrieved from "[https://pymolwiki.org/index.php?title=Transparency_mode&oldid=12423](https://pymolwiki.org/index.php?title=Transparency_mode&oldid=12423)"


---

## Two sided lighting

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Overview

**Two_sided_lighting** when turned on illuminates both the inside and outside of surfaces; when off, just the outside. Notice the light on the inside of the surface in the images below. 

## Examples

  * [![Two Sided Lighting Off](/images/d/dd/Tsl_off.png)](/index.php/File:Tsl_off.png "Two Sided Lighting Off")

Two Sided Lighting Off 

  * [![Two Sided Lighting On](/images/f/fb/Tsl_on.png)](/index.php/File:Tsl_on.png "Two Sided Lighting On")

Two Sided Lighting On 




## Syntax
    
    
    set two_sided_lighting, off   # enable
    set two_sided_lighting, on    # disable
    

Retrieved from "[https://pymolwiki.org/index.php?title=Two_sided_lighting&oldid=4298](https://pymolwiki.org/index.php?title=Two_sided_lighting&oldid=4298)"


---

## Format bonds

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

Type  | Python Module   
---|---  
Download  | [scripts/format_bonds.py](https://raw.githubusercontent.com/Pymol-Scripts/Pymol-script-repo/master/scripts/format_bonds.py)  
Author(s)  | [Andreas Warnecke](/index.php/User:Andwar "User:Andwar")  
License  | [BSD-2-Clause](http://opensource.org/licenses/BSD-2-Clause)  
This code has been put under version control in the project [Pymol-script-repo](/index.php/Git_intro "Git intro")  
  
The script **format_bonds** will automatically format bonds in amino acids. 

## Contents

  * 1 Usage
  * 2 Examples
  * 3 Notes
  * 4 SEE ALSO



## Usage
    
    
    format_bonds [ selection [, bonds ]]
    

## Examples

  * [![format_bonds bonds=1](/images/1/10/PHE_valence_0.png)](/index.php/File:PHE_valence_0.png "format_bonds bonds=1")

format_bonds bonds=1 

  * [![format_bonds bonds=2](/images/9/9f/PHE_valence_1_mode_1.png)](/index.php/File:PHE_valence_1_mode_1.png "format_bonds bonds=2")

format_bonds bonds=2 

  * [![format_bonds](/images/c/ce/PHE_delocalized.png)](/index.php/File:PHE_delocalized.png "format_bonds")

format_bonds 



    
    
    import format_bonds
    
    frag PHE
    format_bonds
    
    format_bonds bonds=2
    

  


## Notes

  * Remember to correctly configure plugin import (see: [Git intro](/index.php/Git_intro "Git intro"))
  * **format_bonds** will introduce delocalized bonds by default or when _bonds_ is larger than 2.  

  * Setting _bonds=1_ will simply disable valence display (globally!)
  * The _selection_ argument is 'all' by default and can be used to restrict editing to selected residues.  

  * Note that **format_bonds** will also format acidic residues, the C-terminus, arginine and nitro groups.
  * Tip: press the TAB key after entering format_bonds to get argument suggestions



  


# SEE ALSO

[Git intro](/index.php/Git_intro "Git intro"), [Valence](/index.php/Valence "Valence"), [Modeling and Editing Structures](/index.php/Modeling_and_Editing_Structures "Modeling and Editing Structures")

Retrieved from "[https://pymolwiki.org/index.php?title=Format_bonds&oldid=13941](https://pymolwiki.org/index.php?title=Format_bonds&oldid=13941)"


---

## Valence

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 Overview
  * 2 Examples for the settings: valence and valence_mode
    * 2.1 Syntax
  * 3 The valence command
  * 4 Editing bonds
  * 5 Automatic editing of bonds
  * 6 SEE ALSO



## Overview

Turning on the **valence** setting will enable the display of double bonds.  
Toggling **valence_mode** alters the positioning of double bonds (for representation as [Lines](/index.php/Lines "Lines"))  
**valence_size** alters the distance of double bonds.  
Note that bonds can be edited to be delocalized using [Unbond](/index.php/Unbond "Unbond") and [Bond](/index.php/Bond "Bond").  
There is also a command called **valence**.  


  


## Examples for the settings: valence and valence_mode

  * [![set valence, 0 #\(no double bonds\)](/images/1/10/PHE_valence_0.png)](/index.php/File:PHE_valence_0.png "set valence, 0 #\(no double bonds\)")

set valence, 0  
#(no double bonds) 

  * [![set valence, 1 set valence_mode, 1 #bonds inside](/images/9/9f/PHE_valence_1_mode_1.png)](/index.php/File:PHE_valence_1_mode_1.png "set valence, 1 set valence_mode, 1 #bonds inside")

set valence, 1  
set valence_mode, 1  
#bonds inside 

  * [![set valence, 1 set valence_mode, 0 #bonds centered](/images/8/8e/PHE_valence_1_mode_0.png)](/index.php/File:PHE_valence_1_mode_0.png "set valence, 1 set valence_mode, 0 #bonds centered")

set valence, 1  
set valence_mode, 0  
#bonds centered 

  * [![set valence, 1 #delocalized bonds #see section "Editing bonds" below or try command "valence guess, all"](/images/c/ce/PHE_delocalized.png)](/index.php/File:PHE_delocalized.png "set valence, 1 #delocalized bonds #see section "Editing bonds" below or try command "valence guess, all"")

set valence, 1  
#delocalized bonds  
#see section "Editing bonds" below or try command "valence guess, all" 




  
**valence_size** alters the distance of double bonds, but behaves slightly different depending on valence_mode  


valence_size  valence_size with valence_mode 1  
inside | valence_size with valence_mode 0   
centered   
---|---  
[![Valence size mode1.gif](/images/b/b1/Valence_size_mode1.gif)](/index.php/File:Valence_size_mode1.gif) | [![Valence size mode0.gif](/images/d/d9/Valence_size_mode0.gif)](/index.php/File:Valence_size_mode0.gif)  
  
### Syntax
    
    
    set valence, 0 # off
    set valence, 1 # on
    
    set valence_mode, 0 # centered
    set valence_mode, 1 # inside
    
    set valence_size, 0.1 # default: 0.06 # range 0 - ~0.5
    

## The valence command

The **valence** command automatically formats existing bonds and can even guess the bonds for standard amino acids.  

    
    
    #USAGE:
    valence order, selection1 [, selection2 [, source [, target_state [, source_state [, reset [, quiet ]]]]]]
    order can be either: 1, 2, 3, 4, aromatic, copy, guess
    
    #make PyMOL guess/autoformat bonds in proteins
    valence guess, all
    

  


## Editing bonds
    
    
    # In editing mode: select the bond using Ctrl-right-click, then enter:
    unbond pk1,pk2
    bond pk1,pk2,4
    # 1: single bond, 2: double bond, 3:triple bond, 4:delocalized
    

  


## Automatic editing of bonds

Try using the **valence** command first.  
Secondly, [Format_bonds](/index.php/Format_bonds "Format bonds") is a script that automatically formats valence in all amino acids and has additional options. 

  


## SEE ALSO

[Bond](/index.php/Bond "Bond"), [Unbond](/index.php/Unbond "Unbond")

Retrieved from "[https://pymolwiki.org/index.php?title=Valence&oldid=13567](https://pymolwiki.org/index.php?title=Valence&oldid=13567)"


---

## Virtual trackball

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

## Contents

  * 1 DESCRIPTION
  * 2 USAGE
  * 3 PYMOL API
  * 4 EXAMPLES
  * 5 NOTES
  * 6 SEE ALSO



### DESCRIPTION

The **virtual_trackbal** setting allows you to toggle on or off the virtual trackball feature. 

### USAGE
    
    
    set virtual_trackball, <state>
    

### PYMOL API
    
    
     cmd.set("virtual_trackball", <state>)
    

### EXAMPLES
    
    
    set virtual_trackball, off
    set virtual_trackball, 0                                        #off
    
    set virtual_trackball, on
    set virtual_trackball, 1                                        #on
    

### NOTES

The virtual trackball is a less precise, more 'intuitive' way of manipulating a 3-dimensional scene. 

### SEE ALSO

[Set](/index.php/Set "Set")

Retrieved from "[https://pymolwiki.org/index.php?title=Virtual_trackball&oldid=4661](https://pymolwiki.org/index.php?title=Virtual_trackball&oldid=4661)"


---

## Wrap output

The printable version is no longer supported and may have rendering errors. Please update your browser bookmarks and please use the default browser print function instead.

# Overview

This setting controls the width of the output in the GUI. 

# Syntax
    
    
    # set it on or off
    set wrap_output, boolean
    
    # for example, turn on wrapping?
    set wrap_output, 1
    

# See Also

[Category:GUI](/index.php/Category:GUI "Category:GUI"), [Internal_gui](/index.php/Internal_gui "Internal gui")

Retrieved from "[https://pymolwiki.org/index.php?title=Wrap_output&oldid=6161](https://pymolwiki.org/index.php?title=Wrap_output&oldid=6161)"


---

